// @aws-sdk/client-iot/dist-es/models/enums.js
export const AbortAction = {
    CANCEL: "CANCEL",
};
export const JobExecutionFailureType = {
    ALL: "ALL",
    FAILED: "FAILED",
    REJECTED: "REJECTED",
    TIMED_OUT: "TIMED_OUT",
};
export const DynamoKeyType = {
    NUMBER: "NUMBER",
    STRING: "STRING",
};
export const CannedAccessControlList = {
    AuthenticatedRead: "authenticated-read",
    AwsExecRead: "aws-exec-read",
    BucketOwnerFullControl: "bucket-owner-full-control",
    BucketOwnerRead: "bucket-owner-read",
    LogDeliveryWrite: "log-delivery-write",
    Private: "private",
    PublicRead: "public-read",
    PublicReadWrite: "public-read-write",
};
export const MessageFormat = {
    JSON: "JSON",
    RAW: "RAW",
};
export const ActionType = {
    CONNECT: "CONNECT",
    PUBLISH: "PUBLISH",
    RECEIVE: "RECEIVE",
    SUBSCRIBE: "SUBSCRIBE",
};
export const ComparisonOperator = {
    GREATER_THAN: "greater-than",
    GREATER_THAN_EQUALS: "greater-than-equals",
    IN_CIDR_SET: "in-cidr-set",
    IN_PORT_SET: "in-port-set",
    IN_SET: "in-set",
    LESS_THAN: "less-than",
    LESS_THAN_EQUALS: "less-than-equals",
    NOT_IN_CIDR_SET: "not-in-cidr-set",
    NOT_IN_PORT_SET: "not-in-port-set",
    NOT_IN_SET: "not-in-set",
};
export const ConfidenceLevel = {
    HIGH: "HIGH",
    LOW: "LOW",
    MEDIUM: "MEDIUM",
};
export const DimensionValueOperator = {
    IN: "IN",
    NOT_IN: "NOT_IN",
};
export const VerificationState = {
    BENIGN_POSITIVE: "BENIGN_POSITIVE",
    FALSE_POSITIVE: "FALSE_POSITIVE",
    TRUE_POSITIVE: "TRUE_POSITIVE",
    UNKNOWN: "UNKNOWN",
};
export const AggregationTypeName = {
    CARDINALITY: "Cardinality",
    PERCENTILES: "Percentiles",
    STATISTICS: "Statistics",
};
export const AlertTargetType = {
    SNS: "SNS",
};
export const ApplicationProtocol = {
    DEFAULT: "DEFAULT",
    HTTPS: "HTTPS",
    MQTT_WSS: "MQTT_WSS",
    SECURE_MQTT: "SECURE_MQTT",
};
export const SbomValidationStatus = {
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    SUCCEEDED: "SUCCEEDED",
};
export const ThingPrincipalType = {
    EXCLUSIVE_THING: "EXCLUSIVE_THING",
    NON_EXCLUSIVE_THING: "NON_EXCLUSIVE_THING",
};
export const ConfigName = {
    CERT_AGE_THRESHOLD_IN_DAYS: "CERT_AGE_THRESHOLD_IN_DAYS",
    CERT_EXPIRATION_THRESHOLD_IN_DAYS: "CERT_EXPIRATION_THRESHOLD_IN_DAYS",
};
export const AuditCheckRunStatus = {
    CANCELED: "CANCELED",
    COMPLETED_COMPLIANT: "COMPLETED_COMPLIANT",
    COMPLETED_NON_COMPLIANT: "COMPLETED_NON_COMPLIANT",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    WAITING_FOR_DATA_COLLECTION: "WAITING_FOR_DATA_COLLECTION",
};
export const ResourceType = {
    ACCOUNT_SETTINGS: "ACCOUNT_SETTINGS",
    CA_CERTIFICATE: "CA_CERTIFICATE",
    CLIENT_ID: "CLIENT_ID",
    COGNITO_IDENTITY_POOL: "COGNITO_IDENTITY_POOL",
    DEVICE_CERTIFICATE: "DEVICE_CERTIFICATE",
    IAM_ROLE: "IAM_ROLE",
    IOT_POLICY: "IOT_POLICY",
    ISSUER_CERTIFICATE: "ISSUER_CERTIFICATE",
    ROLE_ALIAS: "ROLE_ALIAS",
};
export const AuditFindingSeverity = {
    CRITICAL: "CRITICAL",
    HIGH: "HIGH",
    LOW: "LOW",
    MEDIUM: "MEDIUM",
};
export const AuditFrequency = {
    BIWEEKLY: "BIWEEKLY",
    DAILY: "DAILY",
    MONTHLY: "MONTHLY",
    WEEKLY: "WEEKLY",
};
export const AuditMitigationActionsExecutionStatus = {
    CANCELED: "CANCELED",
    COMPLETED: "COMPLETED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    PENDING: "PENDING",
    SKIPPED: "SKIPPED",
};
export const AuditMitigationActionsTaskStatus = {
    CANCELED: "CANCELED",
    COMPLETED: "COMPLETED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
};
export const AuditNotificationType = {
    SNS: "SNS",
};
export const AuditTaskStatus = {
    CANCELED: "CANCELED",
    COMPLETED: "COMPLETED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
};
export const AuditTaskType = {
    ON_DEMAND_AUDIT_TASK: "ON_DEMAND_AUDIT_TASK",
    SCHEDULED_AUDIT_TASK: "SCHEDULED_AUDIT_TASK",
};
export const AuthDecision = {
    ALLOWED: "ALLOWED",
    EXPLICIT_DENY: "EXPLICIT_DENY",
    IMPLICIT_DENY: "IMPLICIT_DENY",
};
export const AuthenticationType = {
    AWS_SIGV4: "AWS_SIGV4",
    AWS_X509: "AWS_X509",
    CUSTOM_AUTH: "CUSTOM_AUTH",
    CUSTOM_AUTH_X509: "CUSTOM_AUTH_X509",
    DEFAULT: "DEFAULT",
};
export const AuthorizerStatus = {
    ACTIVE: "ACTIVE",
    INACTIVE: "INACTIVE",
};
export const AutoRegistrationStatus = {
    DISABLE: "DISABLE",
    ENABLE: "ENABLE",
};
export const CertificateProviderOperation = {
    CreateCertificateFromCsr: "CreateCertificateFromCsr",
};
export const CommandParameterType = {
    BINARY: "BINARY",
    BOOLEAN: "BOOLEAN",
    DOUBLE: "DOUBLE",
    INTEGER: "INTEGER",
    LONG: "LONG",
    STRING: "STRING",
    UNSIGNEDLONG: "UNSIGNEDLONG",
};
export const CommandParameterValueComparisonOperator = {
    EQUALS: "EQUALS",
    GREATER_THAN: "GREATER_THAN",
    GREATER_THAN_EQUALS: "GREATER_THAN_EQUALS",
    IN_RANGE: "IN_RANGE",
    IN_SET: "IN_SET",
    LESS_THAN: "LESS_THAN",
    LESS_THAN_EQUALS: "LESS_THAN_EQUALS",
    NOT_EQUALS: "NOT_EQUALS",
    NOT_IN_RANGE: "NOT_IN_RANGE",
    NOT_IN_SET: "NOT_IN_SET",
};
export const CommandNamespace = {
    AWSIoT: "AWS-IoT",
    AWSIoTFleetWise: "AWS-IoT-FleetWise",
};
export const OutputFormat = {
    CBOR: "CBOR",
    JSON: "JSON",
};
export const CustomMetricType = {
    IP_ADDRESS_LIST: "ip-address-list",
    NUMBER: "number",
    NUMBER_LIST: "number-list",
    STRING_LIST: "string-list",
};
export const DimensionType = {
    TOPIC_FILTER: "TOPIC_FILTER",
};
export const ServiceType = {
    CREDENTIAL_PROVIDER: "CREDENTIAL_PROVIDER",
    DATA: "DATA",
    JOBS: "JOBS",
};
export const FleetMetricUnit = {
    Bits: "Bits",
    BitsSecond: "Bits/Second",
    Bytes: "Bytes",
    BytesSecond: "Bytes/Second",
    Count: "Count",
    CountSecond: "Count/Second",
    Gigabits: "Gigabits",
    GigabitsSecond: "Gigabits/Second",
    Gigabytes: "Gigabytes",
    GigabytesSecond: "Gigabytes/Second",
    Kilobits: "Kilobits",
    KilobitsSecond: "Kilobits/Second",
    Kilobytes: "Kilobytes",
    KilobytesSecond: "Kilobytes/Second",
    Megabits: "Megabits",
    MegabitsSecond: "Megabits/Second",
    Megabytes: "Megabytes",
    MegabytesSecond: "Megabytes/Second",
    Microseconds: "Microseconds",
    Milliseconds: "Milliseconds",
    None: "None",
    Percent: "Percent",
    Seconds: "Seconds",
    Terabits: "Terabits",
    TerabitsSecond: "Terabits/Second",
    Terabytes: "Terabytes",
    TerabytesSecond: "Terabytes/Second",
};
export const RetryableFailureType = {
    ALL: "ALL",
    FAILED: "FAILED",
    TIMED_OUT: "TIMED_OUT",
};
export const JobEndBehavior = {
    CANCEL: "CANCEL",
    FORCE_CANCEL: "FORCE_CANCEL",
    STOP_ROLLOUT: "STOP_ROLLOUT",
};
export const TargetSelection = {
    CONTINUOUS: "CONTINUOUS",
    SNAPSHOT: "SNAPSHOT",
};
export const LogLevel = {
    DEBUG: "DEBUG",
    DISABLED: "DISABLED",
    ERROR: "ERROR",
    INFO: "INFO",
    WARN: "WARN",
};
export const PolicyTemplateName = {
    BLANK_POLICY: "BLANK_POLICY",
};
export const CACertificateUpdateAction = {
    DEACTIVATE: "DEACTIVATE",
};
export const DeviceCertificateUpdateAction = {
    DEACTIVATE: "DEACTIVATE",
};
export const AwsJobAbortCriteriaAbortAction = {
    CANCEL: "CANCEL",
};
export const AwsJobAbortCriteriaFailureType = {
    ALL: "ALL",
    FAILED: "FAILED",
    REJECTED: "REJECTED",
    TIMED_OUT: "TIMED_OUT",
};
export const Protocol = {
    HTTP: "HTTP",
    MQTT: "MQTT",
};
export const OTAUpdateStatus = {
    CREATE_COMPLETE: "CREATE_COMPLETE",
    CREATE_FAILED: "CREATE_FAILED",
    CREATE_IN_PROGRESS: "CREATE_IN_PROGRESS",
    CREATE_PENDING: "CREATE_PENDING",
    DELETE_FAILED: "DELETE_FAILED",
    DELETE_IN_PROGRESS: "DELETE_IN_PROGRESS",
};
export const PackageVersionStatus = {
    DEPRECATED: "DEPRECATED",
    DRAFT: "DRAFT",
    PUBLISHED: "PUBLISHED",
};
export const TemplateType = {
    FLEET_PROVISIONING: "FLEET_PROVISIONING",
    JITP: "JITP",
};
export const DayOfWeek = {
    FRI: "FRI",
    MON: "MON",
    SAT: "SAT",
    SUN: "SUN",
    THU: "THU",
    TUE: "TUE",
    WED: "WED",
};
export const TopicRuleDestinationStatus = {
    DELETING: "DELETING",
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
    ERROR: "ERROR",
    IN_PROGRESS: "IN_PROGRESS",
};
export const LogTargetType = {
    CLIENT_ID: "CLIENT_ID",
    DEFAULT: "DEFAULT",
    PRINCIPAL_ID: "PRINCIPAL_ID",
    SOURCE_IP: "SOURCE_IP",
    THING_GROUP: "THING_GROUP",
};
export const CertificateMode = {
    DEFAULT: "DEFAULT",
    SNI_ONLY: "SNI_ONLY",
};
export const CACertificateStatus = {
    ACTIVE: "ACTIVE",
    INACTIVE: "INACTIVE",
};
export const CertificateStatus = {
    ACTIVE: "ACTIVE",
    INACTIVE: "INACTIVE",
    PENDING_ACTIVATION: "PENDING_ACTIVATION",
    PENDING_TRANSFER: "PENDING_TRANSFER",
    REGISTER_INACTIVE: "REGISTER_INACTIVE",
    REVOKED: "REVOKED",
};
export const DetectMitigationActionsTaskStatus = {
    CANCELED: "CANCELED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    SUCCESSFUL: "SUCCESSFUL",
};
export const DomainConfigurationStatus = {
    DISABLED: "DISABLED",
    ENABLED: "ENABLED",
};
export const DomainType = {
    AWS_MANAGED: "AWS_MANAGED",
    CUSTOMER_MANAGED: "CUSTOMER_MANAGED",
    ENDPOINT: "ENDPOINT",
};
export const ServerCertificateStatus = {
    INVALID: "INVALID",
    VALID: "VALID",
};
export const ConfigurationStatus = {
    HEALTHY: "HEALTHY",
    UNHEALTHY: "UNHEALTHY",
};
export const EncryptionType = {
    AWS_OWNED_KMS_KEY: "AWS_OWNED_KMS_KEY",
    CUSTOMER_MANAGED_KMS_KEY: "CUSTOMER_MANAGED_KMS_KEY",
};
export const EventType = {
    CA_CERTIFICATE: "CA_CERTIFICATE",
    CERTIFICATE: "CERTIFICATE",
    JOB: "JOB",
    JOB_EXECUTION: "JOB_EXECUTION",
    POLICY: "POLICY",
    THING: "THING",
    THING_GROUP: "THING_GROUP",
    THING_GROUP_HIERARCHY: "THING_GROUP_HIERARCHY",
    THING_GROUP_MEMBERSHIP: "THING_GROUP_MEMBERSHIP",
    THING_TYPE: "THING_TYPE",
    THING_TYPE_ASSOCIATION: "THING_TYPE_ASSOCIATION",
};
export const IndexStatus = {
    ACTIVE: "ACTIVE",
    BUILDING: "BUILDING",
    REBUILDING: "REBUILDING",
};
export const JobStatus = {
    CANCELED: "CANCELED",
    COMPLETED: "COMPLETED",
    DELETION_IN_PROGRESS: "DELETION_IN_PROGRESS",
    IN_PROGRESS: "IN_PROGRESS",
    SCHEDULED: "SCHEDULED",
};
export const JobExecutionStatus = {
    CANCELED: "CANCELED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    QUEUED: "QUEUED",
    REJECTED: "REJECTED",
    REMOVED: "REMOVED",
    SUCCEEDED: "SUCCEEDED",
    TIMED_OUT: "TIMED_OUT",
};
export const MitigationActionType = {
    ADD_THINGS_TO_THING_GROUP: "ADD_THINGS_TO_THING_GROUP",
    ENABLE_IOT_LOGGING: "ENABLE_IOT_LOGGING",
    PUBLISH_FINDING_TO_SNS: "PUBLISH_FINDING_TO_SNS",
    REPLACE_DEFAULT_POLICY_VERSION: "REPLACE_DEFAULT_POLICY_VERSION",
    UPDATE_CA_CERTIFICATE: "UPDATE_CA_CERTIFICATE",
    UPDATE_DEVICE_CERTIFICATE: "UPDATE_DEVICE_CERTIFICATE",
};
export const DynamicGroupStatus = {
    ACTIVE: "ACTIVE",
    BUILDING: "BUILDING",
    REBUILDING: "REBUILDING",
};
export const Status = {
    Cancelled: "Cancelled",
    Cancelling: "Cancelling",
    Completed: "Completed",
    Failed: "Failed",
    InProgress: "InProgress",
};
export const ModelStatus = {
    ACTIVE: "ACTIVE",
    EXPIRED: "EXPIRED",
    PENDING_BUILD: "PENDING_BUILD",
};
export const CommandExecutionStatus = {
    CREATED: "CREATED",
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    REJECTED: "REJECTED",
    SUCCEEDED: "SUCCEEDED",
    TIMED_OUT: "TIMED_OUT",
};
export const FieldType = {
    BOOLEAN: "Boolean",
    NUMBER: "Number",
    STRING: "String",
};
export const ThingGroupIndexingMode = {
    OFF: "OFF",
    ON: "ON",
};
export const DeviceDefenderIndexingMode = {
    OFF: "OFF",
    VIOLATIONS: "VIOLATIONS",
};
export const FleetIndexingApi = {
    GET_THING_CONNECTIVITY_DATA: "GET_THING_CONNECTIVITY_DATA",
};
export const TargetFieldOrder = {
    LatLon: "LatLon",
    LonLat: "LonLat",
};
export const NamedShadowIndexingMode = {
    OFF: "OFF",
    ON: "ON",
};
export const ThingConnectivityIndexingMode = {
    OFF: "OFF",
    STATUS: "STATUS",
};
export const ThingIndexingMode = {
    OFF: "OFF",
    REGISTRY: "REGISTRY",
    REGISTRY_AND_SHADOW: "REGISTRY_AND_SHADOW",
};
export const DisconnectReasonValue = {
    API_INITIATED_DISCONNECT: "API_INITIATED_DISCONNECT",
    AUTH_ERROR: "AUTH_ERROR",
    CLIENT_ERROR: "CLIENT_ERROR",
    CLIENT_INITIATED_DISCONNECT: "CLIENT_INITIATED_DISCONNECT",
    CONNECTION_LOST: "CONNECTION_LOST",
    CUSTOMAUTH_TTL_EXPIRATION: "CUSTOMAUTH_TTL_EXPIRATION",
    DUPLICATE_CLIENTID: "DUPLICATE_CLIENTID",
    FORBIDDEN_ACCESS: "FORBIDDEN_ACCESS",
    MQTT_KEEP_ALIVE_TIMEOUT: "MQTT_KEEP_ALIVE_TIMEOUT",
    NONE: "NONE",
    SERVER_ERROR: "SERVER_ERROR",
    SERVER_INITIATED_DISCONNECT: "SERVER_INITIATED_DISCONNECT",
    THROTTLED: "THROTTLED",
    UNKNOWN: "UNKNOWN",
    WEBSOCKET_TTL_EXPIRATION: "WEBSOCKET_TTL_EXPIRATION",
};
export const BehaviorCriteriaType = {
    MACHINE_LEARNING: "MACHINE_LEARNING",
    STATIC: "STATIC",
    STATISTICAL: "STATISTICAL",
};
export const SortOrder = {
    Ascending: "ASCENDING",
    Descending: "DESCENDING",
};
export const DetectMitigationActionExecutionStatus = {
    FAILED: "FAILED",
    IN_PROGRESS: "IN_PROGRESS",
    SKIPPED: "SKIPPED",
    SUCCESSFUL: "SUCCESSFUL",
};
export const SbomValidationResult = {
    FAILED: "FAILED",
    SUCCEEDED: "SUCCEEDED",
};
export const SbomValidationErrorCode = {
    FILE_SIZE_LIMIT_EXCEEDED: "FILE_SIZE_LIMIT_EXCEEDED",
    INCOMPATIBLE_FORMAT: "INCOMPATIBLE_FORMAT",
};
export const ReportType = {
    ERRORS: "ERRORS",
    RESULTS: "RESULTS",
};
export const ViolationEventType = {
    ALARM_CLEARED: "alarm-cleared",
    ALARM_INVALIDATED: "alarm-invalidated",
    IN_ALARM: "in-alarm",
};
export const PackageVersionAction = {
    DEPRECATE: "DEPRECATE",
    PUBLISH: "PUBLISH",
};
