// @aws-sdk/client-iot/dist-es/models/errors.js
import { IoTServiceException as __BaseException } from "./IoTServiceException";
export class InternalFailureException extends __BaseException {
    name = "InternalFailureException";
    $fault = "server";
    constructor(opts) {
        super({
            name: "InternalFailureException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalFailureException.prototype);
    }
}
export class InvalidRequestException extends __BaseException {
    name = "InvalidRequestException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidRequestException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidRequestException.prototype);
    }
}
export class ResourceNotFoundException extends __BaseException {
    name = "ResourceNotFoundException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ResourceNotFoundException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceNotFoundException.prototype);
    }
}
export class ServiceUnavailableException extends __BaseException {
    name = "ServiceUnavailableException";
    $fault = "server";
    constructor(opts) {
        super({
            name: "ServiceUnavailableException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceUnavailableException.prototype);
    }
}
export class ThrottlingException extends __BaseException {
    name = "ThrottlingException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ThrottlingException.prototype);
    }
}
export class TransferAlreadyCompletedException extends __BaseException {
    name = "TransferAlreadyCompletedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TransferAlreadyCompletedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TransferAlreadyCompletedException.prototype);
    }
}
export class UnauthorizedException extends __BaseException {
    name = "UnauthorizedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "UnauthorizedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, UnauthorizedException.prototype);
    }
}
export class ConflictException extends __BaseException {
    name = "ConflictException";
    $fault = "client";
    resourceId;
    constructor(opts) {
        super({
            name: "ConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictException.prototype);
        this.resourceId = opts.resourceId;
    }
}
export class InternalServerException extends __BaseException {
    name = "InternalServerException";
    $fault = "server";
    constructor(opts) {
        super({
            name: "InternalServerException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalServerException.prototype);
    }
}
export class ServiceQuotaExceededException extends __BaseException {
    name = "ServiceQuotaExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ServiceQuotaExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ServiceQuotaExceededException.prototype);
    }
}
export class ValidationException extends __BaseException {
    name = "ValidationException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ValidationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ValidationException.prototype);
    }
}
export class LimitExceededException extends __BaseException {
    name = "LimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "LimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LimitExceededException.prototype);
    }
}
export class VersionConflictException extends __BaseException {
    name = "VersionConflictException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "VersionConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, VersionConflictException.prototype);
    }
}
export class InvalidStateTransitionException extends __BaseException {
    name = "InvalidStateTransitionException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidStateTransitionException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidStateTransitionException.prototype);
    }
}
export class ConflictingResourceUpdateException extends __BaseException {
    name = "ConflictingResourceUpdateException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ConflictingResourceUpdateException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictingResourceUpdateException.prototype);
    }
}
export class InternalException extends __BaseException {
    name = "InternalException";
    $fault = "server";
    constructor(opts) {
        super({
            name: "InternalException",
            $fault: "server",
            ...opts,
        });
        Object.setPrototypeOf(this, InternalException.prototype);
    }
}
export class ResourceAlreadyExistsException extends __BaseException {
    name = "ResourceAlreadyExistsException";
    $fault = "client";
    resourceId;
    resourceArn;
    constructor(opts) {
        super({
            name: "ResourceAlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceAlreadyExistsException.prototype);
        this.resourceId = opts.resourceId;
        this.resourceArn = opts.resourceArn;
    }
}
export class CertificateValidationException extends __BaseException {
    name = "CertificateValidationException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "CertificateValidationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CertificateValidationException.prototype);
    }
}
export class InvalidQueryException extends __BaseException {
    name = "InvalidQueryException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidQueryException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidQueryException.prototype);
    }
}
export class IndexNotReadyException extends __BaseException {
    name = "IndexNotReadyException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "IndexNotReadyException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IndexNotReadyException.prototype);
    }
}
export class InvalidAggregationException extends __BaseException {
    name = "InvalidAggregationException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidAggregationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidAggregationException.prototype);
    }
}
export class MalformedPolicyException extends __BaseException {
    name = "MalformedPolicyException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "MalformedPolicyException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, MalformedPolicyException.prototype);
    }
}
export class VersionsLimitExceededException extends __BaseException {
    name = "VersionsLimitExceededException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "VersionsLimitExceededException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, VersionsLimitExceededException.prototype);
    }
}
export class SqlParseException extends __BaseException {
    name = "SqlParseException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "SqlParseException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, SqlParseException.prototype);
    }
}
export class DeleteConflictException extends __BaseException {
    name = "DeleteConflictException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DeleteConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DeleteConflictException.prototype);
    }
}
export class CertificateStateException extends __BaseException {
    name = "CertificateStateException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "CertificateStateException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CertificateStateException.prototype);
    }
}
export class NotConfiguredException extends __BaseException {
    name = "NotConfiguredException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NotConfiguredException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NotConfiguredException.prototype);
    }
}
export class RegistrationCodeValidationException extends __BaseException {
    name = "RegistrationCodeValidationException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "RegistrationCodeValidationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, RegistrationCodeValidationException.prototype);
    }
}
export class CertificateConflictException extends __BaseException {
    name = "CertificateConflictException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "CertificateConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CertificateConflictException.prototype);
    }
}
export class ResourceRegistrationFailureException extends __BaseException {
    name = "ResourceRegistrationFailureException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ResourceRegistrationFailureException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ResourceRegistrationFailureException.prototype);
    }
}
export class TaskAlreadyExistsException extends __BaseException {
    name = "TaskAlreadyExistsException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TaskAlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TaskAlreadyExistsException.prototype);
    }
}
export class InvalidResponseException extends __BaseException {
    name = "InvalidResponseException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidResponseException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidResponseException.prototype);
    }
}
export class TransferConflictException extends __BaseException {
    name = "TransferConflictException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TransferConflictException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TransferConflictException.prototype);
    }
}
