// @smithy/middleware-compression/dist-cjs/index.js
const { booleanSelector, SelectorType, numberSelector } = require("@smithy/core/config");
const { setFeature } = require("@smithy/core");
const { HttpRequest } = require("@smithy/core/protocols");
const { compressStream } = require("./compressStream");
const { compressString } = require("./compressString");
const { isArrayBuffer } = require("@smithy/core/serde");
const { normalizeProvider } = require("@smithy/core/client");

const NODE_DISABLE_REQUEST_COMPRESSION_ENV_NAME = "AWS_DISABLE_REQUEST_COMPRESSION";
const NODE_DISABLE_REQUEST_COMPRESSION_INI_NAME = "disable_request_compression";
const DEFAULT_DISABLE_REQUEST_COMPRESSION = false;
const NODE_DISABLE_REQUEST_COMPRESSION_CONFIG_OPTIONS = {
    environmentVariableSelector: (env) => booleanSelector(env, NODE_DISABLE_REQUEST_COMPRESSION_ENV_NAME, SelectorType.ENV),
    configFileSelector: (profile) => booleanSelector(profile, NODE_DISABLE_REQUEST_COMPRESSION_INI_NAME, SelectorType.CONFIG),
    default: DEFAULT_DISABLE_REQUEST_COMPRESSION,
};

const NODE_REQUEST_MIN_COMPRESSION_SIZE_BYTES_ENV_NAME = "AWS_REQUEST_MIN_COMPRESSION_SIZE_BYTES";
const NODE_REQUEST_MIN_COMPRESSION_SIZE_BYTES_INI_NAME = "request_min_compression_size_bytes";
const DEFAULT_NODE_REQUEST_MIN_COMPRESSION_SIZE_BYTES = 10240;
const NODE_REQUEST_MIN_COMPRESSION_SIZE_BYTES_CONFIG_OPTIONS = {
    environmentVariableSelector: (env) => numberSelector(env, NODE_REQUEST_MIN_COMPRESSION_SIZE_BYTES_ENV_NAME, SelectorType.ENV),
    configFileSelector: (profile) => numberSelector(profile, NODE_REQUEST_MIN_COMPRESSION_SIZE_BYTES_INI_NAME, SelectorType.CONFIG),
    default: DEFAULT_NODE_REQUEST_MIN_COMPRESSION_SIZE_BYTES,
};

var CompressionAlgorithm;
(function (CompressionAlgorithm) {
    CompressionAlgorithm["GZIP"] = "gzip";
})(CompressionAlgorithm || (CompressionAlgorithm = {}));
const CLIENT_SUPPORTED_ALGORITHMS = [CompressionAlgorithm.GZIP];

const isStreaming = (body) => body !== undefined && typeof body !== "string" && !ArrayBuffer.isView(body) && !isArrayBuffer(body);

const compressionMiddleware = (config, middlewareConfig) => (next, context) => async (args) => {
    if (!HttpRequest.isInstance(args.request)) {
        return next(args);
    }
    const disableRequestCompression = await config.disableRequestCompression();
    if (disableRequestCompression) {
        return next(args);
    }
    const { request } = args;
    const { body, headers } = request;
    const { encodings, streamRequiresLength } = middlewareConfig;
    let updatedBody = body;
    let updatedHeaders = headers;
    for (const algorithm of encodings) {
        if (CLIENT_SUPPORTED_ALGORITHMS.includes(algorithm)) {
            let isRequestCompressed = false;
            if (isStreaming(body)) {
                if (!streamRequiresLength) {
                    updatedBody = await compressStream(body);
                    isRequestCompressed = true;
                }
                else {
                    throw new Error("Compression is not supported for streaming blobs that require a length.");
                }
            }
            else {
                const bodyLength = config.bodyLengthChecker(body);
                const requestMinCompressionSizeBytes = await config.requestMinCompressionSizeBytes();
                if (bodyLength && bodyLength >= requestMinCompressionSizeBytes) {
                    updatedBody = await compressString(body);
                    isRequestCompressed = true;
                }
            }
            if (isRequestCompressed) {
                if (headers["content-encoding"]) {
                    updatedHeaders = {
                        ...headers,
                        "content-encoding": `${headers["content-encoding"]}, ${algorithm}`,
                    };
                }
                else {
                    updatedHeaders = { ...headers, "content-encoding": algorithm };
                }
                if (updatedHeaders["content-encoding"].includes("gzip")) {
                    setFeature(context, "GZIP_REQUEST_COMPRESSION", "L");
                }
                break;
            }
        }
    }
    return next({
        ...args,
        request: {
            ...request,
            body: updatedBody,
            headers: updatedHeaders,
        },
    });
};
const compressionMiddlewareOptions = {
    name: "compressionMiddleware",
    step: "build",
    tags: ["REQUEST_BODY_COMPRESSION", "GZIP"],
    override: true,
    priority: "high",
};

const getCompressionPlugin = (config, middlewareConfig) => ({
    applyToStack: (clientStack) => {
        clientStack.add(compressionMiddleware(config, middlewareConfig), compressionMiddlewareOptions);
    },
});

const resolveCompressionConfig = (input) => {
    const { disableRequestCompression, requestMinCompressionSizeBytes: _requestMinCompressionSizeBytes } = input;
    return Object.assign(input, {
        disableRequestCompression: normalizeProvider(disableRequestCompression),
        requestMinCompressionSizeBytes: async () => {
            const requestMinCompressionSizeBytes = await normalizeProvider(_requestMinCompressionSizeBytes)();
            if (requestMinCompressionSizeBytes < 0 || requestMinCompressionSizeBytes > 10485760) {
                throw new RangeError("The value for requestMinCompressionSizeBytes must be between 0 and 10485760 inclusive. " +
                    `The provided value ${requestMinCompressionSizeBytes} is outside this range."`);
            }
            return requestMinCompressionSizeBytes;
        },
    });
};

exports.DEFAULT_DISABLE_REQUEST_COMPRESSION = DEFAULT_DISABLE_REQUEST_COMPRESSION;
exports.DEFAULT_NODE_REQUEST_MIN_COMPRESSION_SIZE_BYTES = DEFAULT_NODE_REQUEST_MIN_COMPRESSION_SIZE_BYTES;
exports.NODE_DISABLE_REQUEST_COMPRESSION_CONFIG_OPTIONS = NODE_DISABLE_REQUEST_COMPRESSION_CONFIG_OPTIONS;
exports.NODE_DISABLE_REQUEST_COMPRESSION_ENV_NAME = NODE_DISABLE_REQUEST_COMPRESSION_ENV_NAME;
exports.NODE_DISABLE_REQUEST_COMPRESSION_INI_NAME = NODE_DISABLE_REQUEST_COMPRESSION_INI_NAME;
exports.NODE_REQUEST_MIN_COMPRESSION_SIZE_BYTES_CONFIG_OPTIONS = NODE_REQUEST_MIN_COMPRESSION_SIZE_BYTES_CONFIG_OPTIONS;
exports.NODE_REQUEST_MIN_COMPRESSION_SIZE_BYTES_ENV_NAME = NODE_REQUEST_MIN_COMPRESSION_SIZE_BYTES_ENV_NAME;
exports.NODE_REQUEST_MIN_COMPRESSION_SIZE_BYTES_INI_NAME = NODE_REQUEST_MIN_COMPRESSION_SIZE_BYTES_INI_NAME;
exports.compressionMiddleware = compressionMiddleware;
exports.compressionMiddlewareOptions = compressionMiddlewareOptions;
exports.getCompressionPlugin = getCompressionPlugin;
exports.resolveCompressionConfig = resolveCompressionConfig;
