// @aws-sdk/client-route-53/dist-cjs/index.js
const { awsEndpointFunctions, emitWarningIfUnsupportedVersion: emitWarningIfUnsupportedVersion$1, createDefaultUserAgentProvider, NODE_APP_ID_CONFIG_OPTIONS, getAwsRegionExtensionConfiguration, resolveAwsRegionExtensionConfiguration, resolveUserAgentConfig, resolveHostHeaderConfig, getUserAgentPlugin, getHostHeaderPlugin, getLoggerPlugin, getRecursionDetectionPlugin } = require("@aws-sdk/core/client");
const { getHttpAuthSchemeEndpointRuleSetPlugin, DefaultIdentityProviderConfig, getHttpSigningPlugin, createPaginator } = require("@smithy/core");
const { normalizeProvider, getSmithyContext, ServiceException, NoOpLogger, emitWarningIfUnsupportedVersion, loadConfigsForDefaultMode, getDefaultExtensionConfiguration, resolveDefaultRuntimeConfig, Client, makeBuilder, createWaiter, checkExceptions, WaiterState, createAggregatedClient } = require("@smithy/core/client");
const { Command: $Command } = require("@smithy/core/client");
exports.$Command = $Command;
exports.__Client = Client;
const { resolveDefaultsModeConfig, loadConfig, NODE_USE_FIPS_ENDPOINT_CONFIG_OPTIONS, NODE_USE_DUALSTACK_ENDPOINT_CONFIG_OPTIONS, NODE_REGION_CONFIG_OPTIONS, NODE_REGION_CONFIG_FILE_OPTIONS, resolveRegionConfig } = require("@smithy/core/config");
const { BinaryDecisionDiagram, EndpointCache, decideEndpoint, customEndpointFunctions, resolveEndpointConfig, getEndpointPlugin } = require("@smithy/core/endpoints");
const { parseUrl, getHttpHandlerExtensionConfiguration, resolveHttpHandlerRuntimeConfig, getContentLengthPlugin } = require("@smithy/core/protocols");
const { DEFAULT_RETRY_MODE, NODE_RETRY_MODE_CONFIG_OPTIONS, NODE_MAX_ATTEMPT_CONFIG_OPTIONS, resolveRetryConfig, getRetryPlugin } = require("@smithy/core/retry");
const { TypeRegistry, getSchemaSerdePlugin } = require("@smithy/core/schema");
const { resolveAwsSdkSigV4Config, AwsSdkSigV4Signer, NODE_AUTH_SCHEME_PREFERENCE_OPTIONS } = require("@aws-sdk/core/httpAuthSchemes");
const { defaultProvider } = require("@aws-sdk/credential-provider-node");
const { toUtf8, fromUtf8, toBase64, fromBase64, calculateBodyLength } = require("@smithy/core/serde");
const { streamCollector, NodeHttpHandler } = require("@smithy/node-http-handler");
const { AwsRestXmlProtocol } = require("@aws-sdk/core/protocols");
const { Sha256 } = require("@smithy/core/checksum");
const { getIdNormalizerPlugin, getChangeResourceRecordSetsPlugin } = require("@aws-sdk/middleware-sdk-route53");

const defaultRoute53HttpAuthSchemeParametersProvider = async (config, context, input) => {
    return {
        operation: getSmithyContext(context).operation,
        region: await normalizeProvider(config.region)() || (() => {
            throw new Error("expected `region` to be configured for `aws.auth#sigv4`");
        })(),
    };
};
function createAwsAuthSigv4HttpAuthOption(authParameters) {
    return {
        schemeId: "aws.auth#sigv4",
        signingProperties: {
            name: "route53",
            region: authParameters.region,
        },
        propertiesExtractor: (config, context) => ({
            signingProperties: {
                config,
                context,
            },
        }),
    };
}
const defaultRoute53HttpAuthSchemeProvider = (authParameters) => {
    const options = [];
    switch (authParameters.operation) {
        default: {
            options.push(createAwsAuthSigv4HttpAuthOption(authParameters));
        }
    }
    return options;
};
const resolveHttpAuthSchemeConfig = (config) => {
    const config_0 = resolveAwsSdkSigV4Config(config);
    return Object.assign(config_0, {
        authSchemePreference: normalizeProvider(config.authSchemePreference ?? []),
    });
};

const resolveClientEndpointParameters = (options) => {
    return Object.assign(options, {
        useDualstackEndpoint: options.useDualstackEndpoint ?? false,
        useFipsEndpoint: options.useFipsEndpoint ?? false,
        defaultSigningName: "route53",
    });
};
const commonParams = {
    UseFIPS: { type: "builtInParams", name: "useFipsEndpoint" },
    Endpoint: { type: "builtInParams", name: "endpoint" },
    Region: { type: "builtInParams", name: "region" },
    UseDualStack: { type: "builtInParams", name: "useDualstackEndpoint" },
};

var version = "3.1080.0";
var packageInfo = {
	version: version};

const s = "ref", t = "authSchemes", u = "name", v = "signingRegion";
const a = -1, b = true, c = "isSet", d = "PartitionResult", e = "booleanEquals", f = "stringEquals", g = "getAttr", h = "sigv4", i = { [s]: "Endpoint" }, j = { "fn": g, "argv": [{ [s]: d }, "name"] }, k = { [s]: d }, l = { [t]: [{ [u]: h, [v]: "us-east-1" }] }, m = { [t]: [{ [u]: h, [v]: "cn-northwest-1" }] }, n = { [t]: [{ [u]: h, [v]: "us-gov-west-1" }] }, o = { [t]: [{ [u]: h, [v]: "eusc-de-east-1" }] }, p = { [t]: [{ [u]: h, [v]: "{PartitionResult#implicitGlobalRegion}" }] }, q = [{ [s]: "Region" }];
const _data = {
    conditions: [
        [c, [i]],
        [c, q],
        ["aws.partition", q, d],
        [e, [{ [s]: "UseFIPS" }, b]],
        [f, [j, "aws"]],
        [f, [j, "aws-cn"]],
        [f, [j, "aws-us-gov"]],
        [e, [{ [s]: "UseDualStack" }, b]],
        [e, [{ fn: g, argv: [k, "supportsFIPS"] }, b]],
        [f, [j, "aws-iso"]],
        [f, [j, "aws-iso-b"]],
        [f, [j, "aws-iso-e"]],
        [f, [j, "aws-iso-f"]],
        [f, [j, "aws-eusc"]],
        [e, [{ fn: g, argv: [k, "supportsDualStack"] }, b]]
    ],
    results: [
        [a],
        [a, "Invalid Configuration: FIPS and custom endpoint are not supported"],
        [a, "Invalid Configuration: Dualstack and custom endpoint are not supported"],
        [i, {}],
        ["https://route53.amazonaws.com", l],
        ["https://route53-fips.amazonaws.com", l],
        ["https://route53.global.api.aws", l],
        ["https://route53-fips.global.api.aws", l],
        ["https://route53.amazonaws.com.cn", m],
        ["https://route53.global.api.amazonwebservices.com.cn", m],
        ["https://route53.us-gov.amazonaws.com", n],
        ["https://route53.us-gov.api.aws", n],
        ["https://route53.c2s.ic.gov", { [t]: [{ [u]: h, [v]: "us-iso-east-1" }] }],
        ["https://route53.sc2s.sgov.gov", { [t]: [{ [u]: h, [v]: "us-isob-east-1" }] }],
        ["https://route53.cloud.adc-e.uk", { [t]: [{ [u]: h, [v]: "eu-isoe-west-1" }] }],
        ["https://route53.csp.hci.ic.gov", { [t]: [{ [u]: h, [v]: "us-isof-south-1" }] }],
        ["https://route53.amazonaws.eu", o],
        ["https://route53.global.api.amazonwebservices.eu", o],
        ["https://route53-fips.{PartitionResult#dualStackDnsSuffix}", p],
        [a, "FIPS and DualStack are enabled, but this partition does not support one or both"],
        ["https://route53-fips.{PartitionResult#dnsSuffix}", p],
        [a, "FIPS is enabled but this partition does not support FIPS"],
        ["https://route53.{PartitionResult#dualStackDnsSuffix}", p],
        [a, "DualStack is enabled but this partition does not support DualStack"],
        ["https://route53.{PartitionResult#dnsSuffix}", p],
        [a, "Invalid Configuration: Missing Region"]
    ]
};
const root = 2;
const r = 100_000_000;
const nodes = new Int32Array([
    -1, 1, -1,
    0, 27, 3,
    1, 4, r + 25,
    2, 5, r + 25,
    3, 19, 6,
    4, 18, 7,
    5, 17, 8,
    6, 25, 9,
    7, 15, 10,
    9, r + 12, 11,
    10, r + 13, 12,
    11, r + 14, 13,
    12, r + 15, 14,
    13, r + 16, r + 24,
    13, r + 17, 16,
    14, r + 22, r + 23,
    7, r + 9, r + 8,
    7, r + 6, r + 4,
    4, 26, 20,
    6, 25, 21,
    7, 23, 22,
    8, r + 20, r + 21,
    8, 24, r + 19,
    14, r + 18, r + 19,
    7, r + 11, r + 10,
    7, r + 7, r + 5,
    3, r + 1, 28,
    7, r + 2, r + 3,
]);
const bdd = BinaryDecisionDiagram.from(nodes, root, _data.conditions, _data.results);

const cache = new EndpointCache({
    size: 50,
    params: ["Endpoint", "Region", "UseDualStack", "UseFIPS"],
});
const defaultEndpointResolver = (endpointParams, context = {}) => {
    return cache.get(endpointParams, () => decideEndpoint(bdd, {
        endpointParams: endpointParams,
        logger: context.logger,
    }));
};
customEndpointFunctions.aws = awsEndpointFunctions;

class Route53ServiceException extends ServiceException {
    constructor(options) {
        super(options);
        Object.setPrototypeOf(this, Route53ServiceException.prototype);
    }
}

class ConcurrentModification extends Route53ServiceException {
    name = "ConcurrentModification";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ConcurrentModification",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConcurrentModification.prototype);
    }
}
class InvalidInput extends Route53ServiceException {
    name = "InvalidInput";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidInput",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidInput.prototype);
    }
}
class InvalidKeySigningKeyStatus extends Route53ServiceException {
    name = "InvalidKeySigningKeyStatus";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidKeySigningKeyStatus",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidKeySigningKeyStatus.prototype);
    }
}
class InvalidKMSArn extends Route53ServiceException {
    name = "InvalidKMSArn";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidKMSArn",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidKMSArn.prototype);
    }
}
class InvalidSigningStatus extends Route53ServiceException {
    name = "InvalidSigningStatus";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidSigningStatus",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidSigningStatus.prototype);
    }
}
class NoSuchKeySigningKey extends Route53ServiceException {
    name = "NoSuchKeySigningKey";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NoSuchKeySigningKey",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchKeySigningKey.prototype);
    }
}
class ConflictingDomainExists extends Route53ServiceException {
    name = "ConflictingDomainExists";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ConflictingDomainExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictingDomainExists.prototype);
    }
}
class InvalidVPCId extends Route53ServiceException {
    name = "InvalidVPCId";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidVPCId",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidVPCId.prototype);
    }
}
class LimitsExceeded extends Route53ServiceException {
    name = "LimitsExceeded";
    $fault = "client";
    constructor(opts) {
        super({
            name: "LimitsExceeded",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LimitsExceeded.prototype);
    }
}
class NoSuchHostedZone extends Route53ServiceException {
    name = "NoSuchHostedZone";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NoSuchHostedZone",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchHostedZone.prototype);
    }
}
class NotAuthorizedException extends Route53ServiceException {
    name = "NotAuthorizedException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NotAuthorizedException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NotAuthorizedException.prototype);
    }
}
class PriorRequestNotComplete extends Route53ServiceException {
    name = "PriorRequestNotComplete";
    $fault = "client";
    constructor(opts) {
        super({
            name: "PriorRequestNotComplete",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, PriorRequestNotComplete.prototype);
    }
}
class PublicZoneVPCAssociation extends Route53ServiceException {
    name = "PublicZoneVPCAssociation";
    $fault = "client";
    constructor(opts) {
        super({
            name: "PublicZoneVPCAssociation",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, PublicZoneVPCAssociation.prototype);
    }
}
class CidrBlockInUseException extends Route53ServiceException {
    name = "CidrBlockInUseException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CidrBlockInUseException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CidrBlockInUseException.prototype);
        this.Message = opts.Message;
    }
}
class CidrCollectionVersionMismatchException extends Route53ServiceException {
    name = "CidrCollectionVersionMismatchException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CidrCollectionVersionMismatchException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CidrCollectionVersionMismatchException.prototype);
        this.Message = opts.Message;
    }
}
class NoSuchCidrCollectionException extends Route53ServiceException {
    name = "NoSuchCidrCollectionException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchCidrCollectionException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchCidrCollectionException.prototype);
        this.Message = opts.Message;
    }
}
class InvalidChangeBatch extends Route53ServiceException {
    name = "InvalidChangeBatch";
    $fault = "client";
    messages;
    constructor(opts) {
        super({
            name: "InvalidChangeBatch",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidChangeBatch.prototype);
        this.messages = opts.messages;
    }
}
class NoSuchHealthCheck extends Route53ServiceException {
    name = "NoSuchHealthCheck";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NoSuchHealthCheck",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchHealthCheck.prototype);
    }
}
class ThrottlingException extends Route53ServiceException {
    name = "ThrottlingException";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ThrottlingException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ThrottlingException.prototype);
    }
}
class CidrCollectionAlreadyExistsException extends Route53ServiceException {
    name = "CidrCollectionAlreadyExistsException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CidrCollectionAlreadyExistsException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CidrCollectionAlreadyExistsException.prototype);
        this.Message = opts.Message;
    }
}
class HealthCheckAlreadyExists extends Route53ServiceException {
    name = "HealthCheckAlreadyExists";
    $fault = "client";
    constructor(opts) {
        super({
            name: "HealthCheckAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, HealthCheckAlreadyExists.prototype);
    }
}
class TooManyHealthChecks extends Route53ServiceException {
    name = "TooManyHealthChecks";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TooManyHealthChecks",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyHealthChecks.prototype);
    }
}
class DelegationSetNotAvailable extends Route53ServiceException {
    name = "DelegationSetNotAvailable";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DelegationSetNotAvailable",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DelegationSetNotAvailable.prototype);
    }
}
class DelegationSetNotReusable extends Route53ServiceException {
    name = "DelegationSetNotReusable";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DelegationSetNotReusable",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DelegationSetNotReusable.prototype);
    }
}
class HostedZoneAlreadyExists extends Route53ServiceException {
    name = "HostedZoneAlreadyExists";
    $fault = "client";
    constructor(opts) {
        super({
            name: "HostedZoneAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, HostedZoneAlreadyExists.prototype);
    }
}
class InvalidDomainName extends Route53ServiceException {
    name = "InvalidDomainName";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidDomainName",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidDomainName.prototype);
    }
}
class NoSuchDelegationSet extends Route53ServiceException {
    name = "NoSuchDelegationSet";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NoSuchDelegationSet",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchDelegationSet.prototype);
    }
}
class TooManyHostedZones extends Route53ServiceException {
    name = "TooManyHostedZones";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TooManyHostedZones",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyHostedZones.prototype);
    }
}
class InvalidArgument extends Route53ServiceException {
    name = "InvalidArgument";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidArgument",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidArgument.prototype);
    }
}
class InvalidKeySigningKeyName extends Route53ServiceException {
    name = "InvalidKeySigningKeyName";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidKeySigningKeyName",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidKeySigningKeyName.prototype);
    }
}
class KeySigningKeyAlreadyExists extends Route53ServiceException {
    name = "KeySigningKeyAlreadyExists";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KeySigningKeyAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KeySigningKeyAlreadyExists.prototype);
    }
}
class TooManyKeySigningKeys extends Route53ServiceException {
    name = "TooManyKeySigningKeys";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TooManyKeySigningKeys",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyKeySigningKeys.prototype);
    }
}
class InsufficientCloudWatchLogsResourcePolicy extends Route53ServiceException {
    name = "InsufficientCloudWatchLogsResourcePolicy";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InsufficientCloudWatchLogsResourcePolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InsufficientCloudWatchLogsResourcePolicy.prototype);
    }
}
class NoSuchCloudWatchLogsLogGroup extends Route53ServiceException {
    name = "NoSuchCloudWatchLogsLogGroup";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NoSuchCloudWatchLogsLogGroup",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchCloudWatchLogsLogGroup.prototype);
    }
}
class QueryLoggingConfigAlreadyExists extends Route53ServiceException {
    name = "QueryLoggingConfigAlreadyExists";
    $fault = "client";
    constructor(opts) {
        super({
            name: "QueryLoggingConfigAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, QueryLoggingConfigAlreadyExists.prototype);
    }
}
class DelegationSetAlreadyCreated extends Route53ServiceException {
    name = "DelegationSetAlreadyCreated";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DelegationSetAlreadyCreated",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DelegationSetAlreadyCreated.prototype);
    }
}
class DelegationSetAlreadyReusable extends Route53ServiceException {
    name = "DelegationSetAlreadyReusable";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DelegationSetAlreadyReusable",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DelegationSetAlreadyReusable.prototype);
    }
}
class HostedZoneNotFound extends Route53ServiceException {
    name = "HostedZoneNotFound";
    $fault = "client";
    constructor(opts) {
        super({
            name: "HostedZoneNotFound",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, HostedZoneNotFound.prototype);
    }
}
class InvalidTrafficPolicyDocument extends Route53ServiceException {
    name = "InvalidTrafficPolicyDocument";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidTrafficPolicyDocument",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidTrafficPolicyDocument.prototype);
    }
}
class TooManyTrafficPolicies extends Route53ServiceException {
    name = "TooManyTrafficPolicies";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TooManyTrafficPolicies",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyTrafficPolicies.prototype);
    }
}
class TrafficPolicyAlreadyExists extends Route53ServiceException {
    name = "TrafficPolicyAlreadyExists";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TrafficPolicyAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TrafficPolicyAlreadyExists.prototype);
    }
}
class NoSuchTrafficPolicy extends Route53ServiceException {
    name = "NoSuchTrafficPolicy";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NoSuchTrafficPolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchTrafficPolicy.prototype);
    }
}
class TooManyTrafficPolicyInstances extends Route53ServiceException {
    name = "TooManyTrafficPolicyInstances";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TooManyTrafficPolicyInstances",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyTrafficPolicyInstances.prototype);
    }
}
class TrafficPolicyInstanceAlreadyExists extends Route53ServiceException {
    name = "TrafficPolicyInstanceAlreadyExists";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TrafficPolicyInstanceAlreadyExists",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TrafficPolicyInstanceAlreadyExists.prototype);
    }
}
class TooManyTrafficPolicyVersionsForCurrentPolicy extends Route53ServiceException {
    name = "TooManyTrafficPolicyVersionsForCurrentPolicy";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TooManyTrafficPolicyVersionsForCurrentPolicy",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyTrafficPolicyVersionsForCurrentPolicy.prototype);
    }
}
class TooManyVPCAssociationAuthorizations extends Route53ServiceException {
    name = "TooManyVPCAssociationAuthorizations";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TooManyVPCAssociationAuthorizations",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TooManyVPCAssociationAuthorizations.prototype);
    }
}
class KeySigningKeyInParentDSRecord extends Route53ServiceException {
    name = "KeySigningKeyInParentDSRecord";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KeySigningKeyInParentDSRecord",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KeySigningKeyInParentDSRecord.prototype);
    }
}
class KeySigningKeyInUse extends Route53ServiceException {
    name = "KeySigningKeyInUse";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KeySigningKeyInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KeySigningKeyInUse.prototype);
    }
}
class CidrCollectionInUseException extends Route53ServiceException {
    name = "CidrCollectionInUseException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "CidrCollectionInUseException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, CidrCollectionInUseException.prototype);
        this.Message = opts.Message;
    }
}
class HealthCheckInUse extends Route53ServiceException {
    name = "HealthCheckInUse";
    $fault = "client";
    constructor(opts) {
        super({
            name: "HealthCheckInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, HealthCheckInUse.prototype);
    }
}
class HostedZoneNotEmpty extends Route53ServiceException {
    name = "HostedZoneNotEmpty";
    $fault = "client";
    constructor(opts) {
        super({
            name: "HostedZoneNotEmpty",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, HostedZoneNotEmpty.prototype);
    }
}
class NoSuchQueryLoggingConfig extends Route53ServiceException {
    name = "NoSuchQueryLoggingConfig";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NoSuchQueryLoggingConfig",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchQueryLoggingConfig.prototype);
    }
}
class DelegationSetInUse extends Route53ServiceException {
    name = "DelegationSetInUse";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DelegationSetInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DelegationSetInUse.prototype);
    }
}
class TrafficPolicyInUse extends Route53ServiceException {
    name = "TrafficPolicyInUse";
    $fault = "client";
    constructor(opts) {
        super({
            name: "TrafficPolicyInUse",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, TrafficPolicyInUse.prototype);
    }
}
class NoSuchTrafficPolicyInstance extends Route53ServiceException {
    name = "NoSuchTrafficPolicyInstance";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NoSuchTrafficPolicyInstance",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchTrafficPolicyInstance.prototype);
    }
}
class VPCAssociationAuthorizationNotFound extends Route53ServiceException {
    name = "VPCAssociationAuthorizationNotFound";
    $fault = "client";
    constructor(opts) {
        super({
            name: "VPCAssociationAuthorizationNotFound",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, VPCAssociationAuthorizationNotFound.prototype);
    }
}
class DNSSECNotFound extends Route53ServiceException {
    name = "DNSSECNotFound";
    $fault = "client";
    constructor(opts) {
        super({
            name: "DNSSECNotFound",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, DNSSECNotFound.prototype);
    }
}
class LastVPCAssociation extends Route53ServiceException {
    name = "LastVPCAssociation";
    $fault = "client";
    constructor(opts) {
        super({
            name: "LastVPCAssociation",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, LastVPCAssociation.prototype);
    }
}
class VPCAssociationNotFound extends Route53ServiceException {
    name = "VPCAssociationNotFound";
    $fault = "client";
    constructor(opts) {
        super({
            name: "VPCAssociationNotFound",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, VPCAssociationNotFound.prototype);
    }
}
class HostedZonePartiallyDelegated extends Route53ServiceException {
    name = "HostedZonePartiallyDelegated";
    $fault = "client";
    constructor(opts) {
        super({
            name: "HostedZonePartiallyDelegated",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, HostedZonePartiallyDelegated.prototype);
    }
}
class KeySigningKeyWithActiveStatusNotFound extends Route53ServiceException {
    name = "KeySigningKeyWithActiveStatusNotFound";
    $fault = "client";
    constructor(opts) {
        super({
            name: "KeySigningKeyWithActiveStatusNotFound",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, KeySigningKeyWithActiveStatusNotFound.prototype);
    }
}
class NoSuchChange extends Route53ServiceException {
    name = "NoSuchChange";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NoSuchChange",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchChange.prototype);
    }
}
class NoSuchGeoLocation extends Route53ServiceException {
    name = "NoSuchGeoLocation";
    $fault = "client";
    constructor(opts) {
        super({
            name: "NoSuchGeoLocation",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchGeoLocation.prototype);
    }
}
class IncompatibleVersion extends Route53ServiceException {
    name = "IncompatibleVersion";
    $fault = "client";
    constructor(opts) {
        super({
            name: "IncompatibleVersion",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, IncompatibleVersion.prototype);
    }
}
class HostedZoneNotPrivate extends Route53ServiceException {
    name = "HostedZoneNotPrivate";
    $fault = "client";
    constructor(opts) {
        super({
            name: "HostedZoneNotPrivate",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, HostedZoneNotPrivate.prototype);
    }
}
class NoSuchCidrLocationException extends Route53ServiceException {
    name = "NoSuchCidrLocationException";
    $fault = "client";
    Message;
    constructor(opts) {
        super({
            name: "NoSuchCidrLocationException",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, NoSuchCidrLocationException.prototype);
        this.Message = opts.Message;
    }
}
class InvalidPaginationToken extends Route53ServiceException {
    name = "InvalidPaginationToken";
    $fault = "client";
    constructor(opts) {
        super({
            name: "InvalidPaginationToken",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, InvalidPaginationToken.prototype);
    }
}
class HealthCheckVersionMismatch extends Route53ServiceException {
    name = "HealthCheckVersionMismatch";
    $fault = "client";
    constructor(opts) {
        super({
            name: "HealthCheckVersionMismatch",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, HealthCheckVersionMismatch.prototype);
    }
}
class ConflictingTypes extends Route53ServiceException {
    name = "ConflictingTypes";
    $fault = "client";
    constructor(opts) {
        super({
            name: "ConflictingTypes",
            $fault: "client",
            ...opts,
        });
        Object.setPrototypeOf(this, ConflictingTypes.prototype);
    }
}

const _A = "Action";
const _AI = "AlarmIdentifier";
const _AKSK = "ActivateKeySigningKey";
const _AKSKR = "ActivateKeySigningKeyRequest";
const _AKSKRc = "ActivateKeySigningKeyResponse";
const _AL = "AccountLimit";
const _AR = "AcceleratedRecovery";
const _ARS = "AcceleratedRecoveryStatus";
const _AT = "AliasTarget";
const _ATd = "AddTags";
const _AVPCWHZ = "AssociateVPCWithHostedZone";
const _AVPCWHZR = "AssociateVPCWithHostedZoneRequest";
const _AVPCWHZRs = "AssociateVPCWithHostedZoneResponse";
const _AWSR = "AWSRegion";
const _Ar = "Arn";
const _B = "Bias";
const _C = "Comment";
const _CB = "ChangeBatch";
const _CBIUE = "CidrBlockInUseException";
const _CBS = "CidrBlockSummary";
const _CBSi = "CidrBlockSummaries";
const _CBi = "CidrBlock";
const _CBid = "CidrBlocks";
const _CC = "CidrCollection";
const _CCAEE = "CidrCollectionAlreadyExistsException";
const _CCC = "CidrCollectionChange";
const _CCCR = "ChangeCidrCollectionRequest";
const _CCCRh = "ChangeCidrCollectionResponse";
const _CCCRr = "CreateCidrCollectionRequest";
const _CCCRre = "CreateCidrCollectionResponse";
const _CCCh = "ChangeCidrCollection";
const _CCCi = "CidrCollectionChanges";
const _CCCr = "CreateCidrCollection";
const _CCIUE = "CidrCollectionInUseException";
const _CCVME = "CidrCollectionVersionMismatchException";
const _CCi = "CidrCollections";
const _CCo = "ContinentCode";
const _CCou = "CountryCode";
const _CD = "CreatedDate";
const _CDE = "ConflictingDomainExists";
const _CHC = "ChildHealthChecks";
const _CHCL = "ChildHealthCheckList";
const _CHCR = "CreateHealthCheckRequest";
const _CHCRr = "CreateHealthCheckResponse";
const _CHCh = "ChildHealthCheck";
const _CHCr = "CreateHealthCheck";
const _CHZ = "CreateHostedZone";
const _CHZR = "CreateHostedZoneRequest";
const _CHZRr = "CreateHostedZoneResponse";
const _CI = "ChangeInfo";
const _CIR = "CheckerIpRanges";
const _CIo = "CollectionId";
const _CKSK = "CreateKeySigningKey";
const _CKSKR = "CreateKeySigningKeyRequest";
const _CKSKRr = "CreateKeySigningKeyResponse";
const _CL = "CidrList";
const _CLi = "CidrLocations";
const _CM = "ConcurrentModification";
const _CN = "ContinentName";
const _CNo = "CountryName";
const _CO = "ComparisonOperator";
const _CQLC = "CreateQueryLoggingConfig";
const _CQLCR = "CreateQueryLoggingConfigRequest";
const _CQLCRr = "CreateQueryLoggingConfigResponse";
const _CR = "CallerReference";
const _CRC = "CidrRoutingConfig";
const _CRDS = "CreateReusableDelegationSet";
const _CRDSR = "CreateReusableDelegationSetRequest";
const _CRDSRr = "CreateReusableDelegationSetResponse";
const _CRRS = "ChangeResourceRecordSets";
const _CRRSR = "ChangeResourceRecordSetsRequest";
const _CRRSRh = "ChangeResourceRecordSetsResponse";
const _CS = "CollectionSummary";
const _CSo = "CollectionSummaries";
const _CT = "ConflictingTypes";
const _CTFR = "ChangeTagsForResource";
const _CTFRR = "ChangeTagsForResourceRequest";
const _CTFRRh = "ChangeTagsForResourceResponse";
const _CTP = "CreateTrafficPolicy";
const _CTPI = "CreateTrafficPolicyInstance";
const _CTPIR = "CreateTrafficPolicyInstanceRequest";
const _CTPIRr = "CreateTrafficPolicyInstanceResponse";
const _CTPR = "CreateTrafficPolicyRequest";
const _CTPRr = "CreateTrafficPolicyResponse";
const _CTPV = "CreateTrafficPolicyVersion";
const _CTPVR = "CreateTrafficPolicyVersionRequest";
const _CTPVRr = "CreateTrafficPolicyVersionResponse";
const _CTh = "CheckedTime";
const _CV = "CollectionVersion";
const _CVPCAA = "CreateVPCAssociationAuthorization";
const _CVPCAAR = "CreateVPCAssociationAuthorizationRequest";
const _CVPCAARr = "CreateVPCAssociationAuthorizationResponse";
const _CWAC = "CloudWatchAlarmConfiguration";
const _CWLLGA = "CloudWatchLogsLogGroupArn";
const _Ch = "Change";
const _Cha = "Changes";
const _Ci = "Cidr";
const _Co = "Coordinates";
const _Col = "Collection";
const _Con = "Config";
const _Cou = "Count";
const _D = "Dimensions";
const _DAM = "DigestAlgorithmMnemonic";
const _DAT = "DigestAlgorithmType";
const _DCC = "DeleteCidrCollection";
const _DCCR = "DeleteCidrCollectionRequest";
const _DCCRe = "DeleteCidrCollectionResponse";
const _DHC = "DeleteHealthCheck";
const _DHCR = "DeleteHealthCheckRequest";
const _DHCRe = "DeleteHealthCheckResponse";
const _DHZ = "DeleteHostedZone";
const _DHZDNSSEC = "DisableHostedZoneDNSSEC";
const _DHZDNSSECR = "DisableHostedZoneDNSSECRequest";
const _DHZDNSSECRi = "DisableHostedZoneDNSSECResponse";
const _DHZR = "DeleteHostedZoneRequest";
const _DHZRe = "DeleteHostedZoneResponse";
const _DKSK = "DeactivateKeySigningKey";
const _DKSKR = "DeactivateKeySigningKeyRequest";
const _DKSKRe = "DeactivateKeySigningKeyResponse";
const _DKSKRel = "DeleteKeySigningKeyRequest";
const _DKSKRele = "DeleteKeySigningKeyResponse";
const _DKSKe = "DeleteKeySigningKey";
const _DL = "DimensionList";
const _DNSKEYR = "DNSKEYRecord";
const _DNSN = "DNSName";
const _DNSSECNF = "DNSSECNotFound";
const _DNSSECS = "DNSSECStatus";
const _DQLC = "DeleteQueryLoggingConfig";
const _DQLCR = "DeleteQueryLoggingConfigRequest";
const _DQLCRe = "DeleteQueryLoggingConfigResponse";
const _DRDS = "DeleteReusableDelegationSet";
const _DRDSR = "DeleteReusableDelegationSetRequest";
const _DRDSRe = "DeleteReusableDelegationSetResponse";
const _DS = "DelegationSet";
const _DSAC = "DelegationSetAlreadyCreated";
const _DSAR = "DelegationSetAlreadyReusable";
const _DSI = "DelegationSetId";
const _DSIU = "DelegationSetInUse";
const _DSNA = "DelegationSetNotAvailable";
const _DSNR = "DelegationSetNotReusable";
const _DSNS = "DelegationSetNameServers";
const _DSR = "DSRecord";
const _DSe = "DelegationSets";
const _DTP = "DeleteTrafficPolicy";
const _DTPI = "DeleteTrafficPolicyInstance";
const _DTPIR = "DeleteTrafficPolicyInstanceRequest";
const _DTPIRe = "DeleteTrafficPolicyInstanceResponse";
const _DTPR = "DeleteTrafficPolicyRequest";
const _DTPRe = "DeleteTrafficPolicyResponse";
const _DV = "DigestValue";
const _DVPCAA = "DeleteVPCAssociationAuthorization";
const _DVPCAAR = "DeleteVPCAssociationAuthorizationRequest";
const _DVPCAARe = "DeleteVPCAssociationAuthorizationResponse";
const _DVPCFHZ = "DisassociateVPCFromHostedZone";
const _DVPCFHZR = "DisassociateVPCFromHostedZoneRequest";
const _DVPCFHZRi = "DisassociateVPCFromHostedZoneResponse";
const _De = "Description";
const _Di = "Dimension";
const _Dis = "Disabled";
const _Do = "Document";
const _EAR = "EnableAcceleratedRecovery";
const _EDNSCSIP = "EDNS0ClientSubnetIP";
const _EDNSCSM = "EDNS0ClientSubnetMask";
const _EHZDNSSEC = "EnableHostedZoneDNSSEC";
const _EHZDNSSECR = "EnableHostedZoneDNSSECRequest";
const _EHZDNSSECRn = "EnableHostedZoneDNSSECResponse";
const _EM = "ErrorMessages";
const _EP = "EvaluationPeriods";
const _ESNI = "EnableSNI";
const _ETH = "EvaluateTargetHealth";
const _F = "Features";
const _FQDN = "FullyQualifiedDomainName";
const _FR = "FailureReasons";
const _FT = "FailureThreshold";
const _Fa = "Failover";
const _Fl = "Flag";
const _GAL = "GetAccountLimit";
const _GALR = "GetAccountLimitRequest";
const _GALRe = "GetAccountLimitResponse";
const _GC = "GetChange";
const _GCIR = "GetCheckerIpRanges";
const _GCIRR = "GetCheckerIpRangesRequest";
const _GCIRRe = "GetCheckerIpRangesResponse";
const _GCR = "GetChangeRequest";
const _GCRe = "GetChangeResponse";
const _GDNSSEC = "GetDNSSEC";
const _GDNSSECR = "GetDNSSECRequest";
const _GDNSSECRe = "GetDNSSECResponse";
const _GGL = "GetGeoLocation";
const _GGLR = "GetGeoLocationRequest";
const _GGLRe = "GetGeoLocationResponse";
const _GHC = "GetHealthCheck";
const _GHCC = "GetHealthCheckCount";
const _GHCCR = "GetHealthCheckCountRequest";
const _GHCCRe = "GetHealthCheckCountResponse";
const _GHCLFR = "GetHealthCheckLastFailureReason";
const _GHCLFRR = "GetHealthCheckLastFailureReasonRequest";
const _GHCLFRRe = "GetHealthCheckLastFailureReasonResponse";
const _GHCR = "GetHealthCheckRequest";
const _GHCRe = "GetHealthCheckResponse";
const _GHCS = "GetHealthCheckStatus";
const _GHCSR = "GetHealthCheckStatusRequest";
const _GHCSRe = "GetHealthCheckStatusResponse";
const _GHZ = "GetHostedZone";
const _GHZC = "GetHostedZoneCount";
const _GHZCR = "GetHostedZoneCountRequest";
const _GHZCRe = "GetHostedZoneCountResponse";
const _GHZL = "GetHostedZoneLimit";
const _GHZLR = "GetHostedZoneLimitRequest";
const _GHZLRe = "GetHostedZoneLimitResponse";
const _GHZR = "GetHostedZoneRequest";
const _GHZRe = "GetHostedZoneResponse";
const _GL = "GeoLocation";
const _GLD = "GeoLocationDetails";
const _GLDL = "GeoLocationDetailsList";
const _GPL = "GeoProximityLocation";
const _GQLC = "GetQueryLoggingConfig";
const _GQLCR = "GetQueryLoggingConfigRequest";
const _GQLCRe = "GetQueryLoggingConfigResponse";
const _GRDS = "GetReusableDelegationSet";
const _GRDSL = "GetReusableDelegationSetLimit";
const _GRDSLR = "GetReusableDelegationSetLimitRequest";
const _GRDSLRe = "GetReusableDelegationSetLimitResponse";
const _GRDSR = "GetReusableDelegationSetRequest";
const _GRDSRe = "GetReusableDelegationSetResponse";
const _GTP = "GetTrafficPolicy";
const _GTPI = "GetTrafficPolicyInstance";
const _GTPIC = "GetTrafficPolicyInstanceCount";
const _GTPICR = "GetTrafficPolicyInstanceCountRequest";
const _GTPICRe = "GetTrafficPolicyInstanceCountResponse";
const _GTPIR = "GetTrafficPolicyInstanceRequest";
const _GTPIRe = "GetTrafficPolicyInstanceResponse";
const _GTPR = "GetTrafficPolicyRequest";
const _GTPRe = "GetTrafficPolicyResponse";
const _HC = "HealthCheck";
const _HCAE = "HealthCheckAlreadyExists";
const _HCC = "HealthCheckConfig";
const _HCCe = "HealthCheckCount";
const _HCI = "HealthCheckId";
const _HCIU = "HealthCheckInUse";
const _HCO = "HealthCheckObservations";
const _HCOe = "HealthCheckObservation";
const _HCRL = "HealthCheckRegionList";
const _HCV = "HealthCheckVersion";
const _HCVM = "HealthCheckVersionMismatch";
const _HCe = "HealthChecks";
const _HT = "HealthThreshold";
const _HZ = "HostedZone";
const _HZAE = "HostedZoneAlreadyExists";
const _HZC = "HostedZoneConfig";
const _HZCo = "HostedZoneCount";
const _HZF = "HostedZoneFeatures";
const _HZFR = "HostedZoneFailureReasons";
const _HZI = "HostedZoneId";
const _HZIM = "HostedZoneIdMarker";
const _HZL = "HostedZoneLimit";
const _HZNE = "HostedZoneNotEmpty";
const _HZNF = "HostedZoneNotFound";
const _HZNP = "HostedZoneNotPrivate";
const _HZO = "HostedZoneOwner";
const _HZPD = "HostedZonePartiallyDelegated";
const _HZS = "HostedZoneSummary";
const _HZSo = "HostedZoneSummaries";
const _HZT = "HostedZoneType";
const _HZo = "HostedZones";
const _I = "Id";
const _IA = "InvalidArgument";
const _ICB = "InvalidChangeBatch";
const _ICWLRP = "InsufficientCloudWatchLogsResourcePolicy";
const _IDHS = "InsufficientDataHealthStatus";
const _IDN = "InvalidDomainName";
const _II = "InvalidInput";
const _IKMSA = "InvalidKMSArn";
const _IKSKN = "InvalidKeySigningKeyName";
const _IKSKS = "InvalidKeySigningKeyStatus";
const _IPA = "IPAddress";
const _IPT = "InvalidPaginationToken";
const _ISS = "InvalidSigningStatus";
const _IT = "IsTruncated";
const _ITPD = "InvalidTrafficPolicyDocument";
const _IV = "IncompatibleVersion";
const _IVPCI = "InvalidVPCId";
const _In = "Inverted";
const _K = "Key";
const _KA = "KmsArn";
const _KMSA = "KeyManagementServiceArn";
const _KSK = "KeySigningKey";
const _KSKAE = "KeySigningKeyAlreadyExists";
const _KSKIPDSR = "KeySigningKeyInParentDSRecord";
const _KSKIU = "KeySigningKeyInUse";
const _KSKWASNF = "KeySigningKeyWithActiveStatusNotFound";
const _KSKe = "KeySigningKeys";
const _KT = "KeyTag";
const _L = "Latitude";
const _LCB = "ListCidrBlocks";
const _LCBR = "ListCidrBlocksRequest";
const _LCBRi = "ListCidrBlocksResponse";
const _LCC = "ListCidrCollections";
const _LCCR = "ListCidrCollectionsRequest";
const _LCCRi = "ListCidrCollectionsResponse";
const _LCL = "ListCidrLocations";
const _LCLR = "ListCidrLocationsRequest";
const _LCLRi = "ListCidrLocationsResponse";
const _LE = "LimitsExceeded";
const _LGL = "ListGeoLocations";
const _LGLR = "ListGeoLocationsRequest";
const _LGLRi = "ListGeoLocationsResponse";
const _LHC = "ListHealthChecks";
const _LHCR = "ListHealthChecksRequest";
const _LHCRi = "ListHealthChecksResponse";
const _LHZ = "ListHostedZones";
const _LHZBN = "ListHostedZonesByName";
const _LHZBNR = "ListHostedZonesByNameRequest";
const _LHZBNRi = "ListHostedZonesByNameResponse";
const _LHZBVPC = "ListHostedZonesByVPC";
const _LHZBVPCR = "ListHostedZonesByVPCRequest";
const _LHZBVPCRi = "ListHostedZonesByVPCResponse";
const _LHZR = "ListHostedZonesRequest";
const _LHZRi = "ListHostedZonesResponse";
const _LMD = "LastModifiedDate";
const _LN = "LocationName";
const _LQLC = "ListQueryLoggingConfigs";
const _LQLCR = "ListQueryLoggingConfigsRequest";
const _LQLCRi = "ListQueryLoggingConfigsResponse";
const _LRDS = "ListReusableDelegationSets";
const _LRDSR = "ListReusableDelegationSetsRequest";
const _LRDSRi = "ListReusableDelegationSetsResponse";
const _LRRS = "ListResourceRecordSets";
const _LRRSR = "ListResourceRecordSetsRequest";
const _LRRSRi = "ListResourceRecordSetsResponse";
const _LS = "LinkedService";
const _LSo = "LocationSummary";
const _LSoc = "LocationSummaries";
const _LTFR = "ListTagsForResource";
const _LTFRR = "ListTagsForResourceRequest";
const _LTFRRi = "ListTagsForResourceResponse";
const _LTFRRis = "ListTagsForResourcesRequest";
const _LTFRRist = "ListTagsForResourcesResponse";
const _LTFRi = "ListTagsForResources";
const _LTP = "ListTrafficPolicies";
const _LTPI = "ListTrafficPolicyInstances";
const _LTPIBHZ = "ListTrafficPolicyInstancesByHostedZone";
const _LTPIBHZR = "ListTrafficPolicyInstancesByHostedZoneRequest";
const _LTPIBHZRi = "ListTrafficPolicyInstancesByHostedZoneResponse";
const _LTPIBP = "ListTrafficPolicyInstancesByPolicy";
const _LTPIBPR = "ListTrafficPolicyInstancesByPolicyRequest";
const _LTPIBPRi = "ListTrafficPolicyInstancesByPolicyResponse";
const _LTPIR = "ListTrafficPolicyInstancesRequest";
const _LTPIRi = "ListTrafficPolicyInstancesResponse";
const _LTPR = "ListTrafficPoliciesRequest";
const _LTPRi = "ListTrafficPoliciesResponse";
const _LTPV = "ListTrafficPolicyVersions";
const _LTPVR = "ListTrafficPolicyVersionsRequest";
const _LTPVRi = "ListTrafficPolicyVersionsResponse";
const _LV = "LatestVersion";
const _LVPCA = "LastVPCAssociation";
const _LVPCAA = "ListVPCAssociationAuthorizations";
const _LVPCAAR = "ListVPCAssociationAuthorizationsRequest";
const _LVPCAARi = "ListVPCAssociationAuthorizationsResponse";
const _LZG = "LocalZoneGroup";
const _Li = "Limit";
const _Lo = "Longitude";
const _Loc = "Location";
const _M = "Message";
const _MI = "MaxItems";
const _ML = "MeasureLatency";
const _MN = "MetricName";
const _MR = "MaxResults";
const _MVA = "MultiValueAnswer";
const _Ma = "Marker";
const _N = "Name";
const _NAE = "NotAuthorizedException";
const _NCC = "NextContinentCode";
const _NCCe = "NextCountryCode";
const _NDNSN = "NextDNSName";
const _NHZI = "NextHostedZoneId";
const _NM = "NextMarker";
const _NRI = "NextRecordIdentifier";
const _NRN = "NextRecordName";
const _NRT = "NextRecordType";
const _NS = "NameServers";
const _NSC = "NoSuchChange";
const _NSCCE = "NoSuchCidrCollectionException";
const _NSCLE = "NoSuchCidrLocationException";
const _NSCWLLG = "NoSuchCloudWatchLogsLogGroup";
const _NSCe = "NextSubdivisionCode";
const _NSDS = "NoSuchDelegationSet";
const _NSGL = "NoSuchGeoLocation";
const _NSHC = "NoSuchHealthCheck";
const _NSHZ = "NoSuchHostedZone";
const _NSKSK = "NoSuchKeySigningKey";
const _NSQLC = "NoSuchQueryLoggingConfig";
const _NSTP = "NoSuchTrafficPolicy";
const _NSTPI = "NoSuchTrafficPolicyInstance";
const _NSa = "NameServer";
const _NT = "NextToken";
const _Na = "Namespace";
const _Nam = "Nameserver";
const _O = "Owner";
const _OA = "OwningAccount";
const _OS = "OwningService";
const _P = "Period";
const _PK = "PublicKey";
const _PRNC = "PriorRequestNotComplete";
const _PZ = "PrivateZone";
const _PZVPCA = "PublicZoneVPCAssociation";
const _Po = "Port";
const _Pr = "Protocol";
const _QLC = "QueryLoggingConfig";
const _QLCAE = "QueryLoggingConfigAlreadyExists";
const _QLCu = "QueryLoggingConfigs";
const _R = "Region";
const _RC = "ResponseCode";
const _RCA = "RoutingControlArn";
const _RD = "RecordData";
const _RDE = "RecordDataEntry";
const _RDSL = "ReusableDelegationSetLimit";
const _RE = "ResetElements";
const _REN = "ResettableElementName";
const _RENL = "ResettableElementNameList";
const _RI = "ResourceId";
const _RIP = "ResolverIP";
const _RIe = "RequestInterval";
const _RIes = "ResourceIds";
const _RN = "RecordName";
const _RP = "ResourcePath";
const _RR = "ResourceRecord";
const _RRS = "ResourceRecordSet";
const _RRSC = "ResourceRecordSetCount";
const _RRSe = "ResourceRecordSets";
const _RRe = "ResourceRecords";
const _RT = "ResourceType";
const _RTK = "RemoveTagKeys";
const _RTS = "ResourceTagSet";
const _RTSL = "ResourceTagSetList";
const _RTSe = "ResourceTagSets";
const _RTe = "RecordType";
const _Re = "Regions";
const _S = "Status";
const _SA = "SubmittedAt";
const _SAM = "SigningAlgorithmMnemonic";
const _SAT = "SigningAlgorithmType";
const _SC = "SubdivisionCode";
const _SCC = "StartContinentCode";
const _SCCt = "StartCountryCode";
const _SI = "SetIdentifier";
const _SM = "StatusMessage";
const _SN = "SubdivisionName";
const _SP = "ServicePrincipal";
const _SR = "StatusReport";
const _SRI = "StartRecordIdentifier";
const _SRN = "StartRecordName";
const _SRT = "StartRecordType";
const _SS = "ServeSignature";
const _SSC = "StartSubdivisionCode";
const _SSe = "SearchString";
const _St = "Statistic";
const _Sta = "State";
const _T = "Type";
const _TDNSA = "TestDNSAnswer";
const _TDNSAR = "TestDNSAnswerRequest";
const _TDNSARe = "TestDNSAnswerResponse";
const _TE = "ThrottlingException";
const _TKL = "TagKeyList";
const _TL = "TagList";
const _TMHC = "TooManyHealthChecks";
const _TMHZ = "TooManyHostedZones";
const _TMKSK = "TooManyKeySigningKeys";
const _TMTP = "TooManyTrafficPolicies";
const _TMTPI = "TooManyTrafficPolicyInstances";
const _TMTPVFCP = "TooManyTrafficPolicyVersionsForCurrentPolicy";
const _TMVPCAA = "TooManyVPCAssociationAuthorizations";
const _TP = "TrafficPolicy";
const _TPAE = "TrafficPolicyAlreadyExists";
const _TPC = "TrafficPolicyCount";
const _TPI = "TrafficPolicyId";
const _TPIAE = "TrafficPolicyInstanceAlreadyExists";
const _TPIC = "TrafficPolicyInstanceCount";
const _TPII = "TrafficPolicyInstanceId";
const _TPIM = "TrafficPolicyIdMarker";
const _TPINM = "TrafficPolicyInstanceNameMarker";
const _TPITM = "TrafficPolicyInstanceTypeMarker";
const _TPIU = "TrafficPolicyInUse";
const _TPIr = "TrafficPolicyInstance";
const _TPIra = "TrafficPolicyInstances";
const _TPS = "TrafficPolicySummaries";
const _TPSr = "TrafficPolicySummary";
const _TPT = "TrafficPolicyType";
const _TPV = "TrafficPolicyVersion";
const _TPVM = "TrafficPolicyVersionMarker";
const _TPr = "TrafficPolicies";
const _TRIL = "TagResourceIdList";
const _TTL = "TTL";
const _Ta = "Tags";
const _Tag = "Tag";
const _Th = "Threshold";
const _UHC = "UpdateHealthCheck";
const _UHCR = "UpdateHealthCheckRequest";
const _UHCRp = "UpdateHealthCheckResponse";
const _UHZC = "UpdateHostedZoneComment";
const _UHZCR = "UpdateHostedZoneCommentRequest";
const _UHZCRp = "UpdateHostedZoneCommentResponse";
const _UHZF = "UpdateHostedZoneFeatures";
const _UHZFR = "UpdateHostedZoneFeaturesRequest";
const _UHZFRp = "UpdateHostedZoneFeaturesResponse";
const _UTPC = "UpdateTrafficPolicyComment";
const _UTPCR = "UpdateTrafficPolicyCommentRequest";
const _UTPCRp = "UpdateTrafficPolicyCommentResponse";
const _UTPI = "UpdateTrafficPolicyInstance";
const _UTPIR = "UpdateTrafficPolicyInstanceRequest";
const _UTPIRp = "UpdateTrafficPolicyInstanceResponse";
const _V = "Value";
const _VPC = "VPC";
const _VPCAANF = "VPCAssociationAuthorizationNotFound";
const _VPCANF = "VPCAssociationNotFound";
const _VPCI = "VPCId";
const _VPCR = "VPCRegion";
const _VPCs = "VPCs";
const _Ve = "Version";
const _W = "Weight";
const _c = "client";
const _co = "continentcode";
const _cou = "countrycode";
const _d = "dnsname";
const _de = "delegationsetid";
const _e = "error";
const _ed = "edns0clientsubnetip";
const _edn = "edns0clientsubnetmask";
const _h = "hostedzoneid";
const _hE = "httpError";
const _hH = "httpHeader";
const _hQ = "httpQuery";
const _ho = "hostedzonetype";
const _ht = "http";
const _i = "identifier";
const _id = "id";
const _l = "location";
const _m = "message";
const _ma = "maxresults";
const _mar = "marker";
const _max = "maxitems";
const _me = "messages";
const _n = "nexttoken";
const _na = "name";
const _r = "recordname";
const _re = "recordtype";
const _res = "resolverip";
const _s = "smithy.ts.sdk.synthetic.com.amazonaws.route53";
const _st = "startcontinentcode";
const _sta = "startcountrycode";
const _star = "startsubdivisioncode";
const _su = "subdivisioncode";
const _t = "type";
const _tr = "trafficpolicyid";
const _tra = "trafficpolicyinstancename";
const _traf = "trafficpolicyinstancetype";
const _traff = "trafficpolicyversion";
const _v = "vpcid";
const _ve = "version";
const _vp = "vpcregion";
const _xN = "xmlName";
const n0 = "com.amazonaws.route53";
const _s_registry = TypeRegistry.for(_s);
var Route53ServiceException$ = [-3, _s, "Route53ServiceException", 0, [], []];
_s_registry.registerError(Route53ServiceException$, Route53ServiceException);
const n0_registry = TypeRegistry.for(n0);
var CidrBlockInUseException$ = [-3, n0, _CBIUE,
    { [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
n0_registry.registerError(CidrBlockInUseException$, CidrBlockInUseException);
var CidrCollectionAlreadyExistsException$ = [-3, n0, _CCAEE,
    { [_e]: _c },
    [_M],
    [0]
];
n0_registry.registerError(CidrCollectionAlreadyExistsException$, CidrCollectionAlreadyExistsException);
var CidrCollectionInUseException$ = [-3, n0, _CCIUE,
    { [_e]: _c, [_hE]: 400 },
    [_M],
    [0]
];
n0_registry.registerError(CidrCollectionInUseException$, CidrCollectionInUseException);
var CidrCollectionVersionMismatchException$ = [-3, n0, _CCVME,
    { [_e]: _c, [_hE]: 409 },
    [_M],
    [0]
];
n0_registry.registerError(CidrCollectionVersionMismatchException$, CidrCollectionVersionMismatchException);
var ConcurrentModification$ = [-3, n0, _CM,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(ConcurrentModification$, ConcurrentModification);
var ConflictingDomainExists$ = [-3, n0, _CDE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(ConflictingDomainExists$, ConflictingDomainExists);
var ConflictingTypes$ = [-3, n0, _CT,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(ConflictingTypes$, ConflictingTypes);
var DelegationSetAlreadyCreated$ = [-3, n0, _DSAC,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(DelegationSetAlreadyCreated$, DelegationSetAlreadyCreated);
var DelegationSetAlreadyReusable$ = [-3, n0, _DSAR,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(DelegationSetAlreadyReusable$, DelegationSetAlreadyReusable);
var DelegationSetInUse$ = [-3, n0, _DSIU,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(DelegationSetInUse$, DelegationSetInUse);
var DelegationSetNotAvailable$ = [-3, n0, _DSNA,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(DelegationSetNotAvailable$, DelegationSetNotAvailable);
var DelegationSetNotReusable$ = [-3, n0, _DSNR,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(DelegationSetNotReusable$, DelegationSetNotReusable);
var DNSSECNotFound$ = [-3, n0, _DNSSECNF,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(DNSSECNotFound$, DNSSECNotFound);
var HealthCheckAlreadyExists$ = [-3, n0, _HCAE,
    { [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
n0_registry.registerError(HealthCheckAlreadyExists$, HealthCheckAlreadyExists);
var HealthCheckInUse$ = [-3, n0, _HCIU,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(HealthCheckInUse$, HealthCheckInUse);
var HealthCheckVersionMismatch$ = [-3, n0, _HCVM,
    { [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
n0_registry.registerError(HealthCheckVersionMismatch$, HealthCheckVersionMismatch);
var HostedZoneAlreadyExists$ = [-3, n0, _HZAE,
    { [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
n0_registry.registerError(HostedZoneAlreadyExists$, HostedZoneAlreadyExists);
var HostedZoneNotEmpty$ = [-3, n0, _HZNE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(HostedZoneNotEmpty$, HostedZoneNotEmpty);
var HostedZoneNotFound$ = [-3, n0, _HZNF,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(HostedZoneNotFound$, HostedZoneNotFound);
var HostedZoneNotPrivate$ = [-3, n0, _HZNP,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(HostedZoneNotPrivate$, HostedZoneNotPrivate);
var HostedZonePartiallyDelegated$ = [-3, n0, _HZPD,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(HostedZonePartiallyDelegated$, HostedZonePartiallyDelegated);
var IncompatibleVersion$ = [-3, n0, _IV,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(IncompatibleVersion$, IncompatibleVersion);
var InsufficientCloudWatchLogsResourcePolicy$ = [-3, n0, _ICWLRP,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(InsufficientCloudWatchLogsResourcePolicy$, InsufficientCloudWatchLogsResourcePolicy);
var InvalidArgument$ = [-3, n0, _IA,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(InvalidArgument$, InvalidArgument);
var InvalidChangeBatch$ = [-3, n0, _ICB,
    { [_e]: _c },
    [_me, _m],
    [[() => ErrorMessages, 0], 0]
];
n0_registry.registerError(InvalidChangeBatch$, InvalidChangeBatch);
var InvalidDomainName$ = [-3, n0, _IDN,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(InvalidDomainName$, InvalidDomainName);
var InvalidInput$ = [-3, n0, _II,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(InvalidInput$, InvalidInput);
var InvalidKeySigningKeyName$ = [-3, n0, _IKSKN,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(InvalidKeySigningKeyName$, InvalidKeySigningKeyName);
var InvalidKeySigningKeyStatus$ = [-3, n0, _IKSKS,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(InvalidKeySigningKeyStatus$, InvalidKeySigningKeyStatus);
var InvalidKMSArn$ = [-3, n0, _IKMSA,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(InvalidKMSArn$, InvalidKMSArn);
var InvalidPaginationToken$ = [-3, n0, _IPT,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(InvalidPaginationToken$, InvalidPaginationToken);
var InvalidSigningStatus$ = [-3, n0, _ISS,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(InvalidSigningStatus$, InvalidSigningStatus);
var InvalidTrafficPolicyDocument$ = [-3, n0, _ITPD,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(InvalidTrafficPolicyDocument$, InvalidTrafficPolicyDocument);
var InvalidVPCId$ = [-3, n0, _IVPCI,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(InvalidVPCId$, InvalidVPCId);
var KeySigningKeyAlreadyExists$ = [-3, n0, _KSKAE,
    { [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
n0_registry.registerError(KeySigningKeyAlreadyExists$, KeySigningKeyAlreadyExists);
var KeySigningKeyInParentDSRecord$ = [-3, n0, _KSKIPDSR,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(KeySigningKeyInParentDSRecord$, KeySigningKeyInParentDSRecord);
var KeySigningKeyInUse$ = [-3, n0, _KSKIU,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(KeySigningKeyInUse$, KeySigningKeyInUse);
var KeySigningKeyWithActiveStatusNotFound$ = [-3, n0, _KSKWASNF,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(KeySigningKeyWithActiveStatusNotFound$, KeySigningKeyWithActiveStatusNotFound);
var LastVPCAssociation$ = [-3, n0, _LVPCA,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(LastVPCAssociation$, LastVPCAssociation);
var LimitsExceeded$ = [-3, n0, _LE,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(LimitsExceeded$, LimitsExceeded);
var NoSuchChange$ = [-3, n0, _NSC,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(NoSuchChange$, NoSuchChange);
var NoSuchCidrCollectionException$ = [-3, n0, _NSCCE,
    { [_e]: _c, [_hE]: 404 },
    [_M],
    [0]
];
n0_registry.registerError(NoSuchCidrCollectionException$, NoSuchCidrCollectionException);
var NoSuchCidrLocationException$ = [-3, n0, _NSCLE,
    { [_e]: _c, [_hE]: 404 },
    [_M],
    [0]
];
n0_registry.registerError(NoSuchCidrLocationException$, NoSuchCidrLocationException);
var NoSuchCloudWatchLogsLogGroup$ = [-3, n0, _NSCWLLG,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(NoSuchCloudWatchLogsLogGroup$, NoSuchCloudWatchLogsLogGroup);
var NoSuchDelegationSet$ = [-3, n0, _NSDS,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(NoSuchDelegationSet$, NoSuchDelegationSet);
var NoSuchGeoLocation$ = [-3, n0, _NSGL,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(NoSuchGeoLocation$, NoSuchGeoLocation);
var NoSuchHealthCheck$ = [-3, n0, _NSHC,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(NoSuchHealthCheck$, NoSuchHealthCheck);
var NoSuchHostedZone$ = [-3, n0, _NSHZ,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(NoSuchHostedZone$, NoSuchHostedZone);
var NoSuchKeySigningKey$ = [-3, n0, _NSKSK,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(NoSuchKeySigningKey$, NoSuchKeySigningKey);
var NoSuchQueryLoggingConfig$ = [-3, n0, _NSQLC,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(NoSuchQueryLoggingConfig$, NoSuchQueryLoggingConfig);
var NoSuchTrafficPolicy$ = [-3, n0, _NSTP,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(NoSuchTrafficPolicy$, NoSuchTrafficPolicy);
var NoSuchTrafficPolicyInstance$ = [-3, n0, _NSTPI,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(NoSuchTrafficPolicyInstance$, NoSuchTrafficPolicyInstance);
var NotAuthorizedException$ = [-3, n0, _NAE,
    { [_e]: _c, [_hE]: 401 },
    [_m],
    [0]
];
n0_registry.registerError(NotAuthorizedException$, NotAuthorizedException);
var PriorRequestNotComplete$ = [-3, n0, _PRNC,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(PriorRequestNotComplete$, PriorRequestNotComplete);
var PublicZoneVPCAssociation$ = [-3, n0, _PZVPCA,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(PublicZoneVPCAssociation$, PublicZoneVPCAssociation);
var QueryLoggingConfigAlreadyExists$ = [-3, n0, _QLCAE,
    { [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
n0_registry.registerError(QueryLoggingConfigAlreadyExists$, QueryLoggingConfigAlreadyExists);
var ThrottlingException$ = [-3, n0, _TE,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(ThrottlingException$, ThrottlingException);
var TooManyHealthChecks$ = [-3, n0, _TMHC,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(TooManyHealthChecks$, TooManyHealthChecks);
var TooManyHostedZones$ = [-3, n0, _TMHZ,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(TooManyHostedZones$, TooManyHostedZones);
var TooManyKeySigningKeys$ = [-3, n0, _TMKSK,
    { [_e]: _c },
    [_m],
    [0]
];
n0_registry.registerError(TooManyKeySigningKeys$, TooManyKeySigningKeys);
var TooManyTrafficPolicies$ = [-3, n0, _TMTP,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(TooManyTrafficPolicies$, TooManyTrafficPolicies);
var TooManyTrafficPolicyInstances$ = [-3, n0, _TMTPI,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(TooManyTrafficPolicyInstances$, TooManyTrafficPolicyInstances);
var TooManyTrafficPolicyVersionsForCurrentPolicy$ = [-3, n0, _TMTPVFCP,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(TooManyTrafficPolicyVersionsForCurrentPolicy$, TooManyTrafficPolicyVersionsForCurrentPolicy);
var TooManyVPCAssociationAuthorizations$ = [-3, n0, _TMVPCAA,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(TooManyVPCAssociationAuthorizations$, TooManyVPCAssociationAuthorizations);
var TrafficPolicyAlreadyExists$ = [-3, n0, _TPAE,
    { [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
n0_registry.registerError(TrafficPolicyAlreadyExists$, TrafficPolicyAlreadyExists);
var TrafficPolicyInstanceAlreadyExists$ = [-3, n0, _TPIAE,
    { [_e]: _c, [_hE]: 409 },
    [_m],
    [0]
];
n0_registry.registerError(TrafficPolicyInstanceAlreadyExists$, TrafficPolicyInstanceAlreadyExists);
var TrafficPolicyInUse$ = [-3, n0, _TPIU,
    { [_e]: _c, [_hE]: 400 },
    [_m],
    [0]
];
n0_registry.registerError(TrafficPolicyInUse$, TrafficPolicyInUse);
var VPCAssociationAuthorizationNotFound$ = [-3, n0, _VPCAANF,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(VPCAssociationAuthorizationNotFound$, VPCAssociationAuthorizationNotFound);
var VPCAssociationNotFound$ = [-3, n0, _VPCANF,
    { [_e]: _c, [_hE]: 404 },
    [_m],
    [0]
];
n0_registry.registerError(VPCAssociationNotFound$, VPCAssociationNotFound);
const errorTypeRegistries = [
    _s_registry,
    n0_registry,
];
var AccountLimit$ = [3, n0, _AL,
    0,
    [_T, _V],
    [0, 1], 2
];
var ActivateKeySigningKeyRequest$ = [3, n0, _AKSKR,
    0,
    [_HZI, _N],
    [[0, 1], [0, 1]], 2
];
var ActivateKeySigningKeyResponse$ = [3, n0, _AKSKRc,
    0,
    [_CI],
    [() => ChangeInfo$], 1
];
var AlarmIdentifier$ = [3, n0, _AI,
    0,
    [_R, _N],
    [0, 0], 2
];
var AliasTarget$ = [3, n0, _AT,
    0,
    [_HZI, _DNSN, _ETH],
    [0, 0, 2], 3
];
var AssociateVPCWithHostedZoneRequest$ = [3, n0, _AVPCWHZR,
    0,
    [_HZI, _VPC, _C],
    [[0, 1], () => VPC$, 0], 2
];
var AssociateVPCWithHostedZoneResponse$ = [3, n0, _AVPCWHZRs,
    0,
    [_CI],
    [() => ChangeInfo$], 1
];
var Change$ = [3, n0, _Ch,
    0,
    [_A, _RRS],
    [0, [() => ResourceRecordSet$, 0]], 2
];
var ChangeBatch$ = [3, n0, _CB,
    0,
    [_Cha, _C],
    [[() => Changes, 0], 0], 1
];
var ChangeCidrCollectionRequest$ = [3, n0, _CCCR,
    0,
    [_I, _Cha, _CV],
    [[0, 1], [() => CidrCollectionChanges, 0], 1], 2
];
var ChangeCidrCollectionResponse$ = [3, n0, _CCCRh,
    0,
    [_I],
    [0], 1
];
var ChangeInfo$ = [3, n0, _CI,
    0,
    [_I, _S, _SA, _C],
    [0, 0, 4, 0], 3
];
var ChangeResourceRecordSetsRequest$ = [3, n0, _CRRSR,
    0,
    [_HZI, _CB],
    [[0, 1], [() => ChangeBatch$, 0]], 2
];
var ChangeResourceRecordSetsResponse$ = [3, n0, _CRRSRh,
    0,
    [_CI],
    [() => ChangeInfo$], 1
];
var ChangeTagsForResourceRequest$ = [3, n0, _CTFRR,
    0,
    [_RT, _RI, _ATd, _RTK],
    [[0, 1], [0, 1], [() => TagList, 0], [() => TagKeyList, 0]], 2
];
var ChangeTagsForResourceResponse$ = [3, n0, _CTFRRh,
    0,
    [],
    []
];
var CidrBlockSummary$ = [3, n0, _CBS,
    0,
    [_CBi, _LN],
    [0, 0]
];
var CidrCollection$ = [3, n0, _CC,
    0,
    [_Ar, _I, _N, _Ve],
    [0, 0, 0, 1]
];
var CidrCollectionChange$ = [3, n0, _CCC,
    0,
    [_LN, _A, _CL],
    [0, 0, [() => CidrList, 0]], 3
];
var CidrRoutingConfig$ = [3, n0, _CRC,
    0,
    [_CIo, _LN],
    [0, 0], 2
];
var CloudWatchAlarmConfiguration$ = [3, n0, _CWAC,
    0,
    [_EP, _Th, _CO, _P, _MN, _Na, _St, _D],
    [1, 1, 0, 1, 0, 0, 0, [() => DimensionList, 0]], 7
];
var CollectionSummary$ = [3, n0, _CS,
    0,
    [_Ar, _I, _N, _Ve],
    [0, 0, 0, 1]
];
var Coordinates$ = [3, n0, _Co,
    0,
    [_L, _Lo],
    [0, 0], 2
];
var CreateCidrCollectionRequest$ = [3, n0, _CCCRr,
    0,
    [_N, _CR],
    [0, 0], 2
];
var CreateCidrCollectionResponse$ = [3, n0, _CCCRre,
    0,
    [_Col, _Loc],
    [() => CidrCollection$, [0, { [_hH]: _Loc }]]
];
var CreateHealthCheckRequest$ = [3, n0, _CHCR,
    0,
    [_CR, _HCC],
    [0, [() => HealthCheckConfig$, 0]], 2
];
var CreateHealthCheckResponse$ = [3, n0, _CHCRr,
    0,
    [_HC, _Loc],
    [[() => HealthCheck$, 0], [0, { [_hH]: _Loc }]], 2
];
var CreateHostedZoneRequest$ = [3, n0, _CHZR,
    0,
    [_N, _CR, _VPC, _HZC, _DSI],
    [0, 0, () => VPC$, () => HostedZoneConfig$, 0], 2
];
var CreateHostedZoneResponse$ = [3, n0, _CHZRr,
    0,
    [_HZ, _CI, _DS, _Loc, _VPC],
    [() => HostedZone$, () => ChangeInfo$, [() => DelegationSet$, 0], [0, { [_hH]: _Loc }], () => VPC$], 4
];
var CreateKeySigningKeyRequest$ = [3, n0, _CKSKR,
    0,
    [_CR, _HZI, _KMSA, _N, _S],
    [0, 0, 0, 0, 0], 5
];
var CreateKeySigningKeyResponse$ = [3, n0, _CKSKRr,
    0,
    [_CI, _KSK, _Loc],
    [() => ChangeInfo$, () => KeySigningKey$, [0, { [_hH]: _Loc }]], 3
];
var CreateQueryLoggingConfigRequest$ = [3, n0, _CQLCR,
    0,
    [_HZI, _CWLLGA],
    [0, 0], 2
];
var CreateQueryLoggingConfigResponse$ = [3, n0, _CQLCRr,
    0,
    [_QLC, _Loc],
    [() => QueryLoggingConfig$, [0, { [_hH]: _Loc }]], 2
];
var CreateReusableDelegationSetRequest$ = [3, n0, _CRDSR,
    0,
    [_CR, _HZI],
    [0, 0], 1
];
var CreateReusableDelegationSetResponse$ = [3, n0, _CRDSRr,
    0,
    [_DS, _Loc],
    [[() => DelegationSet$, 0], [0, { [_hH]: _Loc }]], 2
];
var CreateTrafficPolicyInstanceRequest$ = [3, n0, _CTPIR,
    0,
    [_HZI, _N, _TTL, _TPI, _TPV],
    [0, 0, 1, 0, 1], 5
];
var CreateTrafficPolicyInstanceResponse$ = [3, n0, _CTPIRr,
    0,
    [_TPIr, _Loc],
    [() => TrafficPolicyInstance$, [0, { [_hH]: _Loc }]], 2
];
var CreateTrafficPolicyRequest$ = [3, n0, _CTPR,
    0,
    [_N, _Do, _C],
    [0, 0, 0], 2
];
var CreateTrafficPolicyResponse$ = [3, n0, _CTPRr,
    0,
    [_TP, _Loc],
    [() => TrafficPolicy$, [0, { [_hH]: _Loc }]], 2
];
var CreateTrafficPolicyVersionRequest$ = [3, n0, _CTPVR,
    0,
    [_I, _Do, _C],
    [[0, 1], 0, 0], 2
];
var CreateTrafficPolicyVersionResponse$ = [3, n0, _CTPVRr,
    0,
    [_TP, _Loc],
    [() => TrafficPolicy$, [0, { [_hH]: _Loc }]], 2
];
var CreateVPCAssociationAuthorizationRequest$ = [3, n0, _CVPCAAR,
    0,
    [_HZI, _VPC],
    [[0, 1], () => VPC$], 2
];
var CreateVPCAssociationAuthorizationResponse$ = [3, n0, _CVPCAARr,
    0,
    [_HZI, _VPC],
    [0, () => VPC$], 2
];
var DeactivateKeySigningKeyRequest$ = [3, n0, _DKSKR,
    0,
    [_HZI, _N],
    [[0, 1], [0, 1]], 2
];
var DeactivateKeySigningKeyResponse$ = [3, n0, _DKSKRe,
    0,
    [_CI],
    [() => ChangeInfo$], 1
];
var DelegationSet$ = [3, n0, _DS,
    0,
    [_NS, _I, _CR],
    [[() => DelegationSetNameServers, 0], 0, 0], 1
];
var DeleteCidrCollectionRequest$ = [3, n0, _DCCR,
    0,
    [_I],
    [[0, 1]], 1
];
var DeleteCidrCollectionResponse$ = [3, n0, _DCCRe,
    0,
    [],
    []
];
var DeleteHealthCheckRequest$ = [3, n0, _DHCR,
    0,
    [_HCI],
    [[0, 1]], 1
];
var DeleteHealthCheckResponse$ = [3, n0, _DHCRe,
    0,
    [],
    []
];
var DeleteHostedZoneRequest$ = [3, n0, _DHZR,
    0,
    [_I],
    [[0, 1]], 1
];
var DeleteHostedZoneResponse$ = [3, n0, _DHZRe,
    0,
    [_CI],
    [() => ChangeInfo$], 1
];
var DeleteKeySigningKeyRequest$ = [3, n0, _DKSKRel,
    0,
    [_HZI, _N],
    [[0, 1], [0, 1]], 2
];
var DeleteKeySigningKeyResponse$ = [3, n0, _DKSKRele,
    0,
    [_CI],
    [() => ChangeInfo$], 1
];
var DeleteQueryLoggingConfigRequest$ = [3, n0, _DQLCR,
    0,
    [_I],
    [[0, 1]], 1
];
var DeleteQueryLoggingConfigResponse$ = [3, n0, _DQLCRe,
    0,
    [],
    []
];
var DeleteReusableDelegationSetRequest$ = [3, n0, _DRDSR,
    0,
    [_I],
    [[0, 1]], 1
];
var DeleteReusableDelegationSetResponse$ = [3, n0, _DRDSRe,
    0,
    [],
    []
];
var DeleteTrafficPolicyInstanceRequest$ = [3, n0, _DTPIR,
    0,
    [_I],
    [[0, 1]], 1
];
var DeleteTrafficPolicyInstanceResponse$ = [3, n0, _DTPIRe,
    0,
    [],
    []
];
var DeleteTrafficPolicyRequest$ = [3, n0, _DTPR,
    0,
    [_I, _Ve],
    [[0, 1], [1, 1]], 2
];
var DeleteTrafficPolicyResponse$ = [3, n0, _DTPRe,
    0,
    [],
    []
];
var DeleteVPCAssociationAuthorizationRequest$ = [3, n0, _DVPCAAR,
    0,
    [_HZI, _VPC],
    [[0, 1], () => VPC$], 2
];
var DeleteVPCAssociationAuthorizationResponse$ = [3, n0, _DVPCAARe,
    0,
    [],
    []
];
var Dimension$ = [3, n0, _Di,
    0,
    [_N, _V],
    [0, 0], 2
];
var DisableHostedZoneDNSSECRequest$ = [3, n0, _DHZDNSSECR,
    0,
    [_HZI],
    [[0, 1]], 1
];
var DisableHostedZoneDNSSECResponse$ = [3, n0, _DHZDNSSECRi,
    0,
    [_CI],
    [() => ChangeInfo$], 1
];
var DisassociateVPCFromHostedZoneRequest$ = [3, n0, _DVPCFHZR,
    0,
    [_HZI, _VPC, _C],
    [[0, 1], () => VPC$, 0], 2
];
var DisassociateVPCFromHostedZoneResponse$ = [3, n0, _DVPCFHZRi,
    0,
    [_CI],
    [() => ChangeInfo$], 1
];
var DNSSECStatus$ = [3, n0, _DNSSECS,
    0,
    [_SS, _SM],
    [0, 0]
];
var EnableHostedZoneDNSSECRequest$ = [3, n0, _EHZDNSSECR,
    0,
    [_HZI],
    [[0, 1]], 1
];
var EnableHostedZoneDNSSECResponse$ = [3, n0, _EHZDNSSECRn,
    0,
    [_CI],
    [() => ChangeInfo$], 1
];
var GeoLocation$ = [3, n0, _GL,
    0,
    [_CCo, _CCou, _SC],
    [0, 0, 0]
];
var GeoLocationDetails$ = [3, n0, _GLD,
    0,
    [_CCo, _CN, _CCou, _CNo, _SC, _SN],
    [0, 0, 0, 0, 0, 0]
];
var GeoProximityLocation$ = [3, n0, _GPL,
    0,
    [_AWSR, _LZG, _Co, _B],
    [0, 0, () => Coordinates$, 1]
];
var GetAccountLimitRequest$ = [3, n0, _GALR,
    0,
    [_T],
    [[0, 1]], 1
];
var GetAccountLimitResponse$ = [3, n0, _GALRe,
    0,
    [_Li, _Cou],
    [() => AccountLimit$, 1], 2
];
var GetChangeRequest$ = [3, n0, _GCR,
    0,
    [_I],
    [[0, 1]], 1
];
var GetChangeResponse$ = [3, n0, _GCRe,
    0,
    [_CI],
    [() => ChangeInfo$], 1
];
var GetCheckerIpRangesRequest$ = [3, n0, _GCIRR,
    0,
    [],
    []
];
var GetCheckerIpRangesResponse$ = [3, n0, _GCIRRe,
    0,
    [_CIR],
    [64 | 0], 1
];
var GetDNSSECRequest$ = [3, n0, _GDNSSECR,
    0,
    [_HZI],
    [[0, 1]], 1
];
var GetDNSSECResponse$ = [3, n0, _GDNSSECRe,
    0,
    [_S, _KSKe],
    [() => DNSSECStatus$, () => KeySigningKeys], 2
];
var GetGeoLocationRequest$ = [3, n0, _GGLR,
    0,
    [_CCo, _CCou, _SC],
    [[0, { [_hQ]: _co }], [0, { [_hQ]: _cou }], [0, { [_hQ]: _su }]]
];
var GetGeoLocationResponse$ = [3, n0, _GGLRe,
    0,
    [_GLD],
    [() => GeoLocationDetails$], 1
];
var GetHealthCheckCountRequest$ = [3, n0, _GHCCR,
    0,
    [],
    []
];
var GetHealthCheckCountResponse$ = [3, n0, _GHCCRe,
    0,
    [_HCCe],
    [1], 1
];
var GetHealthCheckLastFailureReasonRequest$ = [3, n0, _GHCLFRR,
    0,
    [_HCI],
    [[0, 1]], 1
];
var GetHealthCheckLastFailureReasonResponse$ = [3, n0, _GHCLFRRe,
    0,
    [_HCO],
    [[() => HealthCheckObservations, 0]], 1
];
var GetHealthCheckRequest$ = [3, n0, _GHCR,
    0,
    [_HCI],
    [[0, 1]], 1
];
var GetHealthCheckResponse$ = [3, n0, _GHCRe,
    0,
    [_HC],
    [[() => HealthCheck$, 0]], 1
];
var GetHealthCheckStatusRequest$ = [3, n0, _GHCSR,
    0,
    [_HCI],
    [[0, 1]], 1
];
var GetHealthCheckStatusResponse$ = [3, n0, _GHCSRe,
    0,
    [_HCO],
    [[() => HealthCheckObservations, 0]], 1
];
var GetHostedZoneCountRequest$ = [3, n0, _GHZCR,
    0,
    [],
    []
];
var GetHostedZoneCountResponse$ = [3, n0, _GHZCRe,
    0,
    [_HZCo],
    [1], 1
];
var GetHostedZoneLimitRequest$ = [3, n0, _GHZLR,
    0,
    [_T, _HZI],
    [[0, 1], [0, 1]], 2
];
var GetHostedZoneLimitResponse$ = [3, n0, _GHZLRe,
    0,
    [_Li, _Cou],
    [() => HostedZoneLimit$, 1], 2
];
var GetHostedZoneRequest$ = [3, n0, _GHZR,
    0,
    [_I],
    [[0, 1]], 1
];
var GetHostedZoneResponse$ = [3, n0, _GHZRe,
    0,
    [_HZ, _DS, _VPCs],
    [() => HostedZone$, [() => DelegationSet$, 0], [() => VPCs, 0]], 1
];
var GetQueryLoggingConfigRequest$ = [3, n0, _GQLCR,
    0,
    [_I],
    [[0, 1]], 1
];
var GetQueryLoggingConfigResponse$ = [3, n0, _GQLCRe,
    0,
    [_QLC],
    [() => QueryLoggingConfig$], 1
];
var GetReusableDelegationSetLimitRequest$ = [3, n0, _GRDSLR,
    0,
    [_T, _DSI],
    [[0, 1], [0, 1]], 2
];
var GetReusableDelegationSetLimitResponse$ = [3, n0, _GRDSLRe,
    0,
    [_Li, _Cou],
    [() => ReusableDelegationSetLimit$, 1], 2
];
var GetReusableDelegationSetRequest$ = [3, n0, _GRDSR,
    0,
    [_I],
    [[0, 1]], 1
];
var GetReusableDelegationSetResponse$ = [3, n0, _GRDSRe,
    0,
    [_DS],
    [[() => DelegationSet$, 0]], 1
];
var GetTrafficPolicyInstanceCountRequest$ = [3, n0, _GTPICR,
    0,
    [],
    []
];
var GetTrafficPolicyInstanceCountResponse$ = [3, n0, _GTPICRe,
    0,
    [_TPIC],
    [1], 1
];
var GetTrafficPolicyInstanceRequest$ = [3, n0, _GTPIR,
    0,
    [_I],
    [[0, 1]], 1
];
var GetTrafficPolicyInstanceResponse$ = [3, n0, _GTPIRe,
    0,
    [_TPIr],
    [() => TrafficPolicyInstance$], 1
];
var GetTrafficPolicyRequest$ = [3, n0, _GTPR,
    0,
    [_I, _Ve],
    [[0, 1], [1, 1]], 2
];
var GetTrafficPolicyResponse$ = [3, n0, _GTPRe,
    0,
    [_TP],
    [() => TrafficPolicy$], 1
];
var HealthCheck$ = [3, n0, _HC,
    0,
    [_I, _CR, _HCC, _HCV, _LS, _CWAC],
    [0, 0, [() => HealthCheckConfig$, 0], 1, () => LinkedService$, [() => CloudWatchAlarmConfiguration$, 0]], 4
];
var HealthCheckConfig$ = [3, n0, _HCC,
    0,
    [_T, _IPA, _Po, _RP, _FQDN, _SSe, _RIe, _FT, _ML, _In, _Dis, _HT, _CHC, _ESNI, _Re, _AI, _IDHS, _RCA],
    [0, 0, 1, 0, 0, 0, 1, 1, 2, 2, 2, 1, [() => ChildHealthCheckList, 0], 2, [() => HealthCheckRegionList, 0], () => AlarmIdentifier$, 0, 0], 1
];
var HealthCheckObservation$ = [3, n0, _HCOe,
    0,
    [_R, _IPA, _SR],
    [0, 0, () => StatusReport$]
];
var HostedZone$ = [3, n0, _HZ,
    0,
    [_I, _N, _CR, _Con, _RRSC, _LS, _F],
    [0, 0, 0, () => HostedZoneConfig$, 1, () => LinkedService$, () => HostedZoneFeatures$], 3
];
var HostedZoneConfig$ = [3, n0, _HZC,
    0,
    [_C, _PZ],
    [0, 2]
];
var HostedZoneFailureReasons$ = [3, n0, _HZFR,
    0,
    [_AR],
    [0]
];
var HostedZoneFeatures$ = [3, n0, _HZF,
    0,
    [_ARS, _FR],
    [0, () => HostedZoneFailureReasons$]
];
var HostedZoneLimit$ = [3, n0, _HZL,
    0,
    [_T, _V],
    [0, 1], 2
];
var HostedZoneOwner$ = [3, n0, _HZO,
    0,
    [_OA, _OS],
    [0, 0]
];
var HostedZoneSummary$ = [3, n0, _HZS,
    0,
    [_HZI, _N, _O],
    [0, 0, () => HostedZoneOwner$], 3
];
var KeySigningKey$ = [3, n0, _KSK,
    0,
    [_N, _KA, _Fl, _SAM, _SAT, _DAM, _DAT, _KT, _DV, _PK, _DSR, _DNSKEYR, _S, _SM, _CD, _LMD],
    [0, 0, 1, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0, 0, 4, 4]
];
var LinkedService$ = [3, n0, _LS,
    0,
    [_SP, _De],
    [0, 0]
];
var ListCidrBlocksRequest$ = [3, n0, _LCBR,
    0,
    [_CIo, _LN, _NT, _MR],
    [[0, 1], [0, { [_hQ]: _l }], [0, { [_hQ]: _n }], [1, { [_hQ]: _ma }]], 1
];
var ListCidrBlocksResponse$ = [3, n0, _LCBRi,
    0,
    [_NT, _CBid],
    [0, () => CidrBlockSummaries]
];
var ListCidrCollectionsRequest$ = [3, n0, _LCCR,
    0,
    [_NT, _MR],
    [[0, { [_hQ]: _n }], [1, { [_hQ]: _ma }]]
];
var ListCidrCollectionsResponse$ = [3, n0, _LCCRi,
    0,
    [_NT, _CCi],
    [0, () => CollectionSummaries]
];
var ListCidrLocationsRequest$ = [3, n0, _LCLR,
    0,
    [_CIo, _NT, _MR],
    [[0, 1], [0, { [_hQ]: _n }], [1, { [_hQ]: _ma }]], 1
];
var ListCidrLocationsResponse$ = [3, n0, _LCLRi,
    0,
    [_NT, _CLi],
    [0, () => LocationSummaries]
];
var ListGeoLocationsRequest$ = [3, n0, _LGLR,
    0,
    [_SCC, _SCCt, _SSC, _MI],
    [[0, { [_hQ]: _st }], [0, { [_hQ]: _sta }], [0, { [_hQ]: _star }], [1, { [_hQ]: _max }]]
];
var ListGeoLocationsResponse$ = [3, n0, _LGLRi,
    0,
    [_GLDL, _IT, _MI, _NCC, _NCCe, _NSCe],
    [[() => GeoLocationDetailsList, 0], 2, 1, 0, 0, 0], 3
];
var ListHealthChecksRequest$ = [3, n0, _LHCR,
    0,
    [_Ma, _MI],
    [[0, { [_hQ]: _mar }], [1, { [_hQ]: _max }]]
];
var ListHealthChecksResponse$ = [3, n0, _LHCRi,
    0,
    [_HCe, _Ma, _IT, _MI, _NM],
    [[() => HealthChecks, 0], 0, 2, 1, 0], 4
];
var ListHostedZonesByNameRequest$ = [3, n0, _LHZBNR,
    0,
    [_DNSN, _HZI, _MI],
    [[0, { [_hQ]: _d }], [0, { [_hQ]: _h }], [1, { [_hQ]: _max }]]
];
var ListHostedZonesByNameResponse$ = [3, n0, _LHZBNRi,
    0,
    [_HZo, _IT, _MI, _DNSN, _HZI, _NDNSN, _NHZI],
    [[() => HostedZones, 0], 2, 1, 0, 0, 0, 0], 3
];
var ListHostedZonesByVPCRequest$ = [3, n0, _LHZBVPCR,
    0,
    [_VPCI, _VPCR, _MI, _NT],
    [[0, { [_hQ]: _v }], [0, { [_hQ]: _vp }], [1, { [_hQ]: _max }], [0, { [_hQ]: _n }]], 2
];
var ListHostedZonesByVPCResponse$ = [3, n0, _LHZBVPCRi,
    0,
    [_HZSo, _MI, _NT],
    [[() => HostedZoneSummaries, 0], 1, 0], 2
];
var ListHostedZonesRequest$ = [3, n0, _LHZR,
    0,
    [_Ma, _MI, _DSI, _HZT],
    [[0, { [_hQ]: _mar }], [1, { [_hQ]: _max }], [0, { [_hQ]: _de }], [0, { [_hQ]: _ho }]]
];
var ListHostedZonesResponse$ = [3, n0, _LHZRi,
    0,
    [_HZo, _Ma, _IT, _MI, _NM],
    [[() => HostedZones, 0], 0, 2, 1, 0], 4
];
var ListQueryLoggingConfigsRequest$ = [3, n0, _LQLCR,
    0,
    [_HZI, _NT, _MR],
    [[0, { [_hQ]: _h }], [0, { [_hQ]: _n }], [1, { [_hQ]: _ma }]]
];
var ListQueryLoggingConfigsResponse$ = [3, n0, _LQLCRi,
    0,
    [_QLCu, _NT],
    [[() => QueryLoggingConfigs, 0], 0], 1
];
var ListResourceRecordSetsRequest$ = [3, n0, _LRRSR,
    0,
    [_HZI, _SRN, _SRT, _SRI, _MI],
    [[0, 1], [0, { [_hQ]: _na }], [0, { [_hQ]: _t }], [0, { [_hQ]: _i }], [1, { [_hQ]: _max }]], 1
];
var ListResourceRecordSetsResponse$ = [3, n0, _LRRSRi,
    0,
    [_RRSe, _IT, _MI, _NRN, _NRT, _NRI],
    [[() => ResourceRecordSets, 0], 2, 1, 0, 0, 0], 3
];
var ListReusableDelegationSetsRequest$ = [3, n0, _LRDSR,
    0,
    [_Ma, _MI],
    [[0, { [_hQ]: _mar }], [1, { [_hQ]: _max }]]
];
var ListReusableDelegationSetsResponse$ = [3, n0, _LRDSRi,
    0,
    [_DSe, _Ma, _IT, _MI, _NM],
    [[() => DelegationSets, 0], 0, 2, 1, 0], 4
];
var ListTagsForResourceRequest$ = [3, n0, _LTFRR,
    0,
    [_RT, _RI],
    [[0, 1], [0, 1]], 2
];
var ListTagsForResourceResponse$ = [3, n0, _LTFRRi,
    0,
    [_RTS],
    [[() => ResourceTagSet$, 0]], 1
];
var ListTagsForResourcesRequest$ = [3, n0, _LTFRRis,
    0,
    [_RT, _RIes],
    [[0, 1], [() => TagResourceIdList, 0]], 2
];
var ListTagsForResourcesResponse$ = [3, n0, _LTFRRist,
    0,
    [_RTSe],
    [[() => ResourceTagSetList, 0]], 1
];
var ListTrafficPoliciesRequest$ = [3, n0, _LTPR,
    0,
    [_TPIM, _MI],
    [[0, { [_hQ]: _tr }], [1, { [_hQ]: _max }]]
];
var ListTrafficPoliciesResponse$ = [3, n0, _LTPRi,
    0,
    [_TPS, _IT, _TPIM, _MI],
    [[() => TrafficPolicySummaries, 0], 2, 0, 1], 4
];
var ListTrafficPolicyInstancesByHostedZoneRequest$ = [3, n0, _LTPIBHZR,
    0,
    [_HZI, _TPINM, _TPITM, _MI],
    [[0, { [_hQ]: _id }], [0, { [_hQ]: _tra }], [0, { [_hQ]: _traf }], [1, { [_hQ]: _max }]], 1
];
var ListTrafficPolicyInstancesByHostedZoneResponse$ = [3, n0, _LTPIBHZRi,
    0,
    [_TPIra, _IT, _MI, _TPINM, _TPITM],
    [[() => TrafficPolicyInstances, 0], 2, 1, 0, 0], 3
];
var ListTrafficPolicyInstancesByPolicyRequest$ = [3, n0, _LTPIBPR,
    0,
    [_TPI, _TPV, _HZIM, _TPINM, _TPITM, _MI],
    [[0, { [_hQ]: _id }], [1, { [_hQ]: _ve }], [0, { [_hQ]: _h }], [0, { [_hQ]: _tra }], [0, { [_hQ]: _traf }], [1, { [_hQ]: _max }]], 2
];
var ListTrafficPolicyInstancesByPolicyResponse$ = [3, n0, _LTPIBPRi,
    0,
    [_TPIra, _IT, _MI, _HZIM, _TPINM, _TPITM],
    [[() => TrafficPolicyInstances, 0], 2, 1, 0, 0, 0], 3
];
var ListTrafficPolicyInstancesRequest$ = [3, n0, _LTPIR,
    0,
    [_HZIM, _TPINM, _TPITM, _MI],
    [[0, { [_hQ]: _h }], [0, { [_hQ]: _tra }], [0, { [_hQ]: _traf }], [1, { [_hQ]: _max }]]
];
var ListTrafficPolicyInstancesResponse$ = [3, n0, _LTPIRi,
    0,
    [_TPIra, _IT, _MI, _HZIM, _TPINM, _TPITM],
    [[() => TrafficPolicyInstances, 0], 2, 1, 0, 0, 0], 3
];
var ListTrafficPolicyVersionsRequest$ = [3, n0, _LTPVR,
    0,
    [_I, _TPVM, _MI],
    [[0, 1], [0, { [_hQ]: _traff }], [1, { [_hQ]: _max }]], 1
];
var ListTrafficPolicyVersionsResponse$ = [3, n0, _LTPVRi,
    0,
    [_TPr, _IT, _TPVM, _MI],
    [[() => TrafficPolicies, 0], 2, 0, 1], 4
];
var ListVPCAssociationAuthorizationsRequest$ = [3, n0, _LVPCAAR,
    0,
    [_HZI, _NT, _MR],
    [[0, 1], [0, { [_hQ]: _n }], [1, { [_hQ]: _ma }]], 1
];
var ListVPCAssociationAuthorizationsResponse$ = [3, n0, _LVPCAARi,
    0,
    [_HZI, _VPCs, _NT],
    [0, [() => VPCs, 0], 0], 2
];
var LocationSummary$ = [3, n0, _LSo,
    0,
    [_LN],
    [0]
];
var QueryLoggingConfig$ = [3, n0, _QLC,
    0,
    [_I, _HZI, _CWLLGA],
    [0, 0, 0], 3
];
var ResourceRecord$ = [3, n0, _RR,
    0,
    [_V],
    [0], 1
];
var ResourceRecordSet$ = [3, n0, _RRS,
    0,
    [_N, _T, _SI, _W, _R, _GL, _Fa, _MVA, _TTL, _RRe, _AT, _HCI, _TPII, _CRC, _GPL],
    [0, 0, 0, 1, 0, () => GeoLocation$, 0, 2, 1, [() => ResourceRecords, 0], () => AliasTarget$, 0, 0, () => CidrRoutingConfig$, () => GeoProximityLocation$], 2
];
var ResourceTagSet$ = [3, n0, _RTS,
    0,
    [_RT, _RI, _Ta],
    [0, 0, [() => TagList, 0]]
];
var ReusableDelegationSetLimit$ = [3, n0, _RDSL,
    0,
    [_T, _V],
    [0, 1], 2
];
var StatusReport$ = [3, n0, _SR,
    0,
    [_S, _CTh],
    [0, 4]
];
var Tag$ = [3, n0, _Tag,
    0,
    [_K, _V],
    [0, 0]
];
var TestDNSAnswerRequest$ = [3, n0, _TDNSAR,
    0,
    [_HZI, _RN, _RTe, _RIP, _EDNSCSIP, _EDNSCSM],
    [[0, { [_hQ]: _h }], [0, { [_hQ]: _r }], [0, { [_hQ]: _re }], [0, { [_hQ]: _res }], [0, { [_hQ]: _ed }], [0, { [_hQ]: _edn }]], 3
];
var TestDNSAnswerResponse$ = [3, n0, _TDNSARe,
    0,
    [_Nam, _RN, _RTe, _RD, _RC, _Pr],
    [0, 0, 0, [() => RecordData, 0], 0, 0], 6
];
var TrafficPolicy$ = [3, n0, _TP,
    0,
    [_I, _Ve, _N, _T, _Do, _C],
    [0, 1, 0, 0, 0, 0], 5
];
var TrafficPolicyInstance$ = [3, n0, _TPIr,
    0,
    [_I, _HZI, _N, _TTL, _Sta, _M, _TPI, _TPV, _TPT],
    [0, 0, 0, 1, 0, 0, 0, 1, 0], 9
];
var TrafficPolicySummary$ = [3, n0, _TPSr,
    0,
    [_I, _N, _T, _LV, _TPC],
    [0, 0, 0, 1, 1], 5
];
var UpdateHealthCheckRequest$ = [3, n0, _UHCR,
    0,
    [_HCI, _HCV, _IPA, _Po, _RP, _FQDN, _SSe, _FT, _In, _Dis, _HT, _CHC, _ESNI, _Re, _AI, _IDHS, _RE],
    [[0, 1], 1, 0, 1, 0, 0, 0, 1, 2, 2, 1, [() => ChildHealthCheckList, 0], 2, [() => HealthCheckRegionList, 0], () => AlarmIdentifier$, 0, [() => ResettableElementNameList, 0]], 1
];
var UpdateHealthCheckResponse$ = [3, n0, _UHCRp,
    0,
    [_HC],
    [[() => HealthCheck$, 0]], 1
];
var UpdateHostedZoneCommentRequest$ = [3, n0, _UHZCR,
    0,
    [_I, _C],
    [[0, 1], 0], 1
];
var UpdateHostedZoneCommentResponse$ = [3, n0, _UHZCRp,
    0,
    [_HZ],
    [() => HostedZone$], 1
];
var UpdateHostedZoneFeaturesRequest$ = [3, n0, _UHZFR,
    0,
    [_HZI, _EAR],
    [[0, 1], 2], 1
];
var UpdateHostedZoneFeaturesResponse$ = [3, n0, _UHZFRp,
    0,
    [],
    []
];
var UpdateTrafficPolicyCommentRequest$ = [3, n0, _UTPCR,
    0,
    [_I, _Ve, _C],
    [[0, 1], [1, 1], 0], 3
];
var UpdateTrafficPolicyCommentResponse$ = [3, n0, _UTPCRp,
    0,
    [_TP],
    [() => TrafficPolicy$], 1
];
var UpdateTrafficPolicyInstanceRequest$ = [3, n0, _UTPIR,
    0,
    [_I, _TTL, _TPI, _TPV],
    [[0, 1], 1, 0, 1], 4
];
var UpdateTrafficPolicyInstanceResponse$ = [3, n0, _UTPIRp,
    0,
    [_TPIr],
    [() => TrafficPolicyInstance$], 1
];
var VPC$ = [3, n0, _VPC,
    0,
    [_VPCR, _VPCI],
    [0, 0]
];
var Changes = [1, n0, _Cha,
    0, [() => Change$,
        { [_xN]: _Ch }]
];
var ChildHealthCheckList = [1, n0, _CHCL,
    0, [0,
        { [_xN]: _CHCh }]
];
var CidrBlockSummaries = [1, n0, _CBSi,
    0, () => CidrBlockSummary$
];
var CidrCollectionChanges = [1, n0, _CCCi,
    0, [() => CidrCollectionChange$,
        0]
];
var CidrList = [1, n0, _CL,
    0, [0,
        { [_xN]: _Ci }]
];
var CollectionSummaries = [1, n0, _CSo,
    0, () => CollectionSummary$
];
var DelegationSetNameServers = [1, n0, _DSNS,
    0, [0,
        { [_xN]: _NSa }]
];
var DelegationSets = [1, n0, _DSe,
    0, [() => DelegationSet$,
        { [_xN]: _DS }]
];
var DimensionList = [1, n0, _DL,
    0, [() => Dimension$,
        { [_xN]: _Di }]
];
var ErrorMessages = [1, n0, _EM,
    0, [0,
        { [_xN]: _M }]
];
var GeoLocationDetailsList = [1, n0, _GLDL,
    0, [() => GeoLocationDetails$,
        { [_xN]: _GLD }]
];
var HealthCheckObservations = [1, n0, _HCO,
    0, [() => HealthCheckObservation$,
        { [_xN]: _HCOe }]
];
var HealthCheckRegionList = [1, n0, _HCRL,
    0, [0,
        { [_xN]: _R }]
];
var HealthChecks = [1, n0, _HCe,
    0, [() => HealthCheck$,
        { [_xN]: _HC }]
];
var HostedZones = [1, n0, _HZo,
    0, [() => HostedZone$,
        { [_xN]: _HZ }]
];
var HostedZoneSummaries = [1, n0, _HZSo,
    0, [() => HostedZoneSummary$,
        { [_xN]: _HZS }]
];
var KeySigningKeys = [1, n0, _KSKe,
    0, () => KeySigningKey$
];
var LocationSummaries = [1, n0, _LSoc,
    0, () => LocationSummary$
];
var QueryLoggingConfigs = [1, n0, _QLCu,
    0, [() => QueryLoggingConfig$,
        { [_xN]: _QLC }]
];
var RecordData = [1, n0, _RD,
    0, [0,
        { [_xN]: _RDE }]
];
var ResettableElementNameList = [1, n0, _RENL,
    0, [0,
        { [_xN]: _REN }]
];
var ResourceRecords = [1, n0, _RRe,
    0, [() => ResourceRecord$,
        { [_xN]: _RR }]
];
var ResourceRecordSets = [1, n0, _RRSe,
    0, [() => ResourceRecordSet$,
        { [_xN]: _RRS }]
];
var ResourceTagSetList = [1, n0, _RTSL,
    0, [() => ResourceTagSet$,
        { [_xN]: _RTS }]
];
var TagKeyList = [1, n0, _TKL,
    0, [0,
        { [_xN]: _K }]
];
var TagList = [1, n0, _TL,
    0, [() => Tag$,
        { [_xN]: _Tag }]
];
var TagResourceIdList = [1, n0, _TRIL,
    0, [0,
        { [_xN]: _RI }]
];
var TrafficPolicies = [1, n0, _TPr,
    0, [() => TrafficPolicy$,
        { [_xN]: _TP }]
];
var TrafficPolicyInstances = [1, n0, _TPIra,
    0, [() => TrafficPolicyInstance$,
        { [_xN]: _TPIr }]
];
var TrafficPolicySummaries = [1, n0, _TPS,
    0, [() => TrafficPolicySummary$,
        { [_xN]: _TPSr }]
];
var VPCs = [1, n0, _VPCs,
    0, [() => VPC$,
        { [_xN]: _VPC }]
];
var ActivateKeySigningKey$ = [9, n0, _AKSK,
    { [_ht]: ["POST", "/2013-04-01/keysigningkey/{HostedZoneId}/{Name}/activate", 200] }, () => ActivateKeySigningKeyRequest$, () => ActivateKeySigningKeyResponse$
];
var AssociateVPCWithHostedZone$ = [9, n0, _AVPCWHZ,
    { [_ht]: ["POST", "/2013-04-01/hostedzone/{HostedZoneId}/associatevpc", 200] }, () => AssociateVPCWithHostedZoneRequest$, () => AssociateVPCWithHostedZoneResponse$
];
var ChangeCidrCollection$ = [9, n0, _CCCh,
    { [_ht]: ["POST", "/2013-04-01/cidrcollection/{Id}", 200] }, () => ChangeCidrCollectionRequest$, () => ChangeCidrCollectionResponse$
];
var ChangeResourceRecordSets$ = [9, n0, _CRRS,
    { [_ht]: ["POST", "/2013-04-01/hostedzone/{HostedZoneId}/rrset", 200] }, () => ChangeResourceRecordSetsRequest$, () => ChangeResourceRecordSetsResponse$
];
var ChangeTagsForResource$ = [9, n0, _CTFR,
    { [_ht]: ["POST", "/2013-04-01/tags/{ResourceType}/{ResourceId}", 200] }, () => ChangeTagsForResourceRequest$, () => ChangeTagsForResourceResponse$
];
var CreateCidrCollection$ = [9, n0, _CCCr,
    { [_ht]: ["POST", "/2013-04-01/cidrcollection", 201] }, () => CreateCidrCollectionRequest$, () => CreateCidrCollectionResponse$
];
var CreateHealthCheck$ = [9, n0, _CHCr,
    { [_ht]: ["POST", "/2013-04-01/healthcheck", 201] }, () => CreateHealthCheckRequest$, () => CreateHealthCheckResponse$
];
var CreateHostedZone$ = [9, n0, _CHZ,
    { [_ht]: ["POST", "/2013-04-01/hostedzone", 201] }, () => CreateHostedZoneRequest$, () => CreateHostedZoneResponse$
];
var CreateKeySigningKey$ = [9, n0, _CKSK,
    { [_ht]: ["POST", "/2013-04-01/keysigningkey", 201] }, () => CreateKeySigningKeyRequest$, () => CreateKeySigningKeyResponse$
];
var CreateQueryLoggingConfig$ = [9, n0, _CQLC,
    { [_ht]: ["POST", "/2013-04-01/queryloggingconfig", 201] }, () => CreateQueryLoggingConfigRequest$, () => CreateQueryLoggingConfigResponse$
];
var CreateReusableDelegationSet$ = [9, n0, _CRDS,
    { [_ht]: ["POST", "/2013-04-01/delegationset", 201] }, () => CreateReusableDelegationSetRequest$, () => CreateReusableDelegationSetResponse$
];
var CreateTrafficPolicy$ = [9, n0, _CTP,
    { [_ht]: ["POST", "/2013-04-01/trafficpolicy", 201] }, () => CreateTrafficPolicyRequest$, () => CreateTrafficPolicyResponse$
];
var CreateTrafficPolicyInstance$ = [9, n0, _CTPI,
    { [_ht]: ["POST", "/2013-04-01/trafficpolicyinstance", 201] }, () => CreateTrafficPolicyInstanceRequest$, () => CreateTrafficPolicyInstanceResponse$
];
var CreateTrafficPolicyVersion$ = [9, n0, _CTPV,
    { [_ht]: ["POST", "/2013-04-01/trafficpolicy/{Id}", 201] }, () => CreateTrafficPolicyVersionRequest$, () => CreateTrafficPolicyVersionResponse$
];
var CreateVPCAssociationAuthorization$ = [9, n0, _CVPCAA,
    { [_ht]: ["POST", "/2013-04-01/hostedzone/{HostedZoneId}/authorizevpcassociation", 200] }, () => CreateVPCAssociationAuthorizationRequest$, () => CreateVPCAssociationAuthorizationResponse$
];
var DeactivateKeySigningKey$ = [9, n0, _DKSK,
    { [_ht]: ["POST", "/2013-04-01/keysigningkey/{HostedZoneId}/{Name}/deactivate", 200] }, () => DeactivateKeySigningKeyRequest$, () => DeactivateKeySigningKeyResponse$
];
var DeleteCidrCollection$ = [9, n0, _DCC,
    { [_ht]: ["DELETE", "/2013-04-01/cidrcollection/{Id}", 200] }, () => DeleteCidrCollectionRequest$, () => DeleteCidrCollectionResponse$
];
var DeleteHealthCheck$ = [9, n0, _DHC,
    { [_ht]: ["DELETE", "/2013-04-01/healthcheck/{HealthCheckId}", 200] }, () => DeleteHealthCheckRequest$, () => DeleteHealthCheckResponse$
];
var DeleteHostedZone$ = [9, n0, _DHZ,
    { [_ht]: ["DELETE", "/2013-04-01/hostedzone/{Id}", 200] }, () => DeleteHostedZoneRequest$, () => DeleteHostedZoneResponse$
];
var DeleteKeySigningKey$ = [9, n0, _DKSKe,
    { [_ht]: ["DELETE", "/2013-04-01/keysigningkey/{HostedZoneId}/{Name}", 200] }, () => DeleteKeySigningKeyRequest$, () => DeleteKeySigningKeyResponse$
];
var DeleteQueryLoggingConfig$ = [9, n0, _DQLC,
    { [_ht]: ["DELETE", "/2013-04-01/queryloggingconfig/{Id}", 200] }, () => DeleteQueryLoggingConfigRequest$, () => DeleteQueryLoggingConfigResponse$
];
var DeleteReusableDelegationSet$ = [9, n0, _DRDS,
    { [_ht]: ["DELETE", "/2013-04-01/delegationset/{Id}", 200] }, () => DeleteReusableDelegationSetRequest$, () => DeleteReusableDelegationSetResponse$
];
var DeleteTrafficPolicy$ = [9, n0, _DTP,
    { [_ht]: ["DELETE", "/2013-04-01/trafficpolicy/{Id}/{Version}", 200] }, () => DeleteTrafficPolicyRequest$, () => DeleteTrafficPolicyResponse$
];
var DeleteTrafficPolicyInstance$ = [9, n0, _DTPI,
    { [_ht]: ["DELETE", "/2013-04-01/trafficpolicyinstance/{Id}", 200] }, () => DeleteTrafficPolicyInstanceRequest$, () => DeleteTrafficPolicyInstanceResponse$
];
var DeleteVPCAssociationAuthorization$ = [9, n0, _DVPCAA,
    { [_ht]: ["POST", "/2013-04-01/hostedzone/{HostedZoneId}/deauthorizevpcassociation", 200] }, () => DeleteVPCAssociationAuthorizationRequest$, () => DeleteVPCAssociationAuthorizationResponse$
];
var DisableHostedZoneDNSSEC$ = [9, n0, _DHZDNSSEC,
    { [_ht]: ["POST", "/2013-04-01/hostedzone/{HostedZoneId}/disable-dnssec", 200] }, () => DisableHostedZoneDNSSECRequest$, () => DisableHostedZoneDNSSECResponse$
];
var DisassociateVPCFromHostedZone$ = [9, n0, _DVPCFHZ,
    { [_ht]: ["POST", "/2013-04-01/hostedzone/{HostedZoneId}/disassociatevpc", 200] }, () => DisassociateVPCFromHostedZoneRequest$, () => DisassociateVPCFromHostedZoneResponse$
];
var EnableHostedZoneDNSSEC$ = [9, n0, _EHZDNSSEC,
    { [_ht]: ["POST", "/2013-04-01/hostedzone/{HostedZoneId}/enable-dnssec", 200] }, () => EnableHostedZoneDNSSECRequest$, () => EnableHostedZoneDNSSECResponse$
];
var GetAccountLimit$ = [9, n0, _GAL,
    { [_ht]: ["GET", "/2013-04-01/accountlimit/{Type}", 200] }, () => GetAccountLimitRequest$, () => GetAccountLimitResponse$
];
var GetChange$ = [9, n0, _GC,
    { [_ht]: ["GET", "/2013-04-01/change/{Id}", 200] }, () => GetChangeRequest$, () => GetChangeResponse$
];
var GetCheckerIpRanges$ = [9, n0, _GCIR,
    { [_ht]: ["GET", "/2013-04-01/checkeripranges", 200] }, () => GetCheckerIpRangesRequest$, () => GetCheckerIpRangesResponse$
];
var GetDNSSEC$ = [9, n0, _GDNSSEC,
    { [_ht]: ["GET", "/2013-04-01/hostedzone/{HostedZoneId}/dnssec", 200] }, () => GetDNSSECRequest$, () => GetDNSSECResponse$
];
var GetGeoLocation$ = [9, n0, _GGL,
    { [_ht]: ["GET", "/2013-04-01/geolocation", 200] }, () => GetGeoLocationRequest$, () => GetGeoLocationResponse$
];
var GetHealthCheck$ = [9, n0, _GHC,
    { [_ht]: ["GET", "/2013-04-01/healthcheck/{HealthCheckId}", 200] }, () => GetHealthCheckRequest$, () => GetHealthCheckResponse$
];
var GetHealthCheckCount$ = [9, n0, _GHCC,
    { [_ht]: ["GET", "/2013-04-01/healthcheckcount", 200] }, () => GetHealthCheckCountRequest$, () => GetHealthCheckCountResponse$
];
var GetHealthCheckLastFailureReason$ = [9, n0, _GHCLFR,
    { [_ht]: ["GET", "/2013-04-01/healthcheck/{HealthCheckId}/lastfailurereason", 200] }, () => GetHealthCheckLastFailureReasonRequest$, () => GetHealthCheckLastFailureReasonResponse$
];
var GetHealthCheckStatus$ = [9, n0, _GHCS,
    { [_ht]: ["GET", "/2013-04-01/healthcheck/{HealthCheckId}/status", 200] }, () => GetHealthCheckStatusRequest$, () => GetHealthCheckStatusResponse$
];
var GetHostedZone$ = [9, n0, _GHZ,
    { [_ht]: ["GET", "/2013-04-01/hostedzone/{Id}", 200] }, () => GetHostedZoneRequest$, () => GetHostedZoneResponse$
];
var GetHostedZoneCount$ = [9, n0, _GHZC,
    { [_ht]: ["GET", "/2013-04-01/hostedzonecount", 200] }, () => GetHostedZoneCountRequest$, () => GetHostedZoneCountResponse$
];
var GetHostedZoneLimit$ = [9, n0, _GHZL,
    { [_ht]: ["GET", "/2013-04-01/hostedzonelimit/{HostedZoneId}/{Type}", 200] }, () => GetHostedZoneLimitRequest$, () => GetHostedZoneLimitResponse$
];
var GetQueryLoggingConfig$ = [9, n0, _GQLC,
    { [_ht]: ["GET", "/2013-04-01/queryloggingconfig/{Id}", 200] }, () => GetQueryLoggingConfigRequest$, () => GetQueryLoggingConfigResponse$
];
var GetReusableDelegationSet$ = [9, n0, _GRDS,
    { [_ht]: ["GET", "/2013-04-01/delegationset/{Id}", 200] }, () => GetReusableDelegationSetRequest$, () => GetReusableDelegationSetResponse$
];
var GetReusableDelegationSetLimit$ = [9, n0, _GRDSL,
    { [_ht]: ["GET", "/2013-04-01/reusabledelegationsetlimit/{DelegationSetId}/{Type}", 200] }, () => GetReusableDelegationSetLimitRequest$, () => GetReusableDelegationSetLimitResponse$
];
var GetTrafficPolicy$ = [9, n0, _GTP,
    { [_ht]: ["GET", "/2013-04-01/trafficpolicy/{Id}/{Version}", 200] }, () => GetTrafficPolicyRequest$, () => GetTrafficPolicyResponse$
];
var GetTrafficPolicyInstance$ = [9, n0, _GTPI,
    { [_ht]: ["GET", "/2013-04-01/trafficpolicyinstance/{Id}", 200] }, () => GetTrafficPolicyInstanceRequest$, () => GetTrafficPolicyInstanceResponse$
];
var GetTrafficPolicyInstanceCount$ = [9, n0, _GTPIC,
    { [_ht]: ["GET", "/2013-04-01/trafficpolicyinstancecount", 200] }, () => GetTrafficPolicyInstanceCountRequest$, () => GetTrafficPolicyInstanceCountResponse$
];
var ListCidrBlocks$ = [9, n0, _LCB,
    { [_ht]: ["GET", "/2013-04-01/cidrcollection/{CollectionId}/cidrblocks", 200] }, () => ListCidrBlocksRequest$, () => ListCidrBlocksResponse$
];
var ListCidrCollections$ = [9, n0, _LCC,
    { [_ht]: ["GET", "/2013-04-01/cidrcollection", 200] }, () => ListCidrCollectionsRequest$, () => ListCidrCollectionsResponse$
];
var ListCidrLocations$ = [9, n0, _LCL,
    { [_ht]: ["GET", "/2013-04-01/cidrcollection/{CollectionId}", 200] }, () => ListCidrLocationsRequest$, () => ListCidrLocationsResponse$
];
var ListGeoLocations$ = [9, n0, _LGL,
    { [_ht]: ["GET", "/2013-04-01/geolocations", 200] }, () => ListGeoLocationsRequest$, () => ListGeoLocationsResponse$
];
var ListHealthChecks$ = [9, n0, _LHC,
    { [_ht]: ["GET", "/2013-04-01/healthcheck", 200] }, () => ListHealthChecksRequest$, () => ListHealthChecksResponse$
];
var ListHostedZones$ = [9, n0, _LHZ,
    { [_ht]: ["GET", "/2013-04-01/hostedzone", 200] }, () => ListHostedZonesRequest$, () => ListHostedZonesResponse$
];
var ListHostedZonesByName$ = [9, n0, _LHZBN,
    { [_ht]: ["GET", "/2013-04-01/hostedzonesbyname", 200] }, () => ListHostedZonesByNameRequest$, () => ListHostedZonesByNameResponse$
];
var ListHostedZonesByVPC$ = [9, n0, _LHZBVPC,
    { [_ht]: ["GET", "/2013-04-01/hostedzonesbyvpc", 200] }, () => ListHostedZonesByVPCRequest$, () => ListHostedZonesByVPCResponse$
];
var ListQueryLoggingConfigs$ = [9, n0, _LQLC,
    { [_ht]: ["GET", "/2013-04-01/queryloggingconfig", 200] }, () => ListQueryLoggingConfigsRequest$, () => ListQueryLoggingConfigsResponse$
];
var ListResourceRecordSets$ = [9, n0, _LRRS,
    { [_ht]: ["GET", "/2013-04-01/hostedzone/{HostedZoneId}/rrset", 200] }, () => ListResourceRecordSetsRequest$, () => ListResourceRecordSetsResponse$
];
var ListReusableDelegationSets$ = [9, n0, _LRDS,
    { [_ht]: ["GET", "/2013-04-01/delegationset", 200] }, () => ListReusableDelegationSetsRequest$, () => ListReusableDelegationSetsResponse$
];
var ListTagsForResource$ = [9, n0, _LTFR,
    { [_ht]: ["GET", "/2013-04-01/tags/{ResourceType}/{ResourceId}", 200] }, () => ListTagsForResourceRequest$, () => ListTagsForResourceResponse$
];
var ListTagsForResources$ = [9, n0, _LTFRi,
    { [_ht]: ["POST", "/2013-04-01/tags/{ResourceType}", 200] }, () => ListTagsForResourcesRequest$, () => ListTagsForResourcesResponse$
];
var ListTrafficPolicies$ = [9, n0, _LTP,
    { [_ht]: ["GET", "/2013-04-01/trafficpolicies", 200] }, () => ListTrafficPoliciesRequest$, () => ListTrafficPoliciesResponse$
];
var ListTrafficPolicyInstances$ = [9, n0, _LTPI,
    { [_ht]: ["GET", "/2013-04-01/trafficpolicyinstances", 200] }, () => ListTrafficPolicyInstancesRequest$, () => ListTrafficPolicyInstancesResponse$
];
var ListTrafficPolicyInstancesByHostedZone$ = [9, n0, _LTPIBHZ,
    { [_ht]: ["GET", "/2013-04-01/trafficpolicyinstances/hostedzone", 200] }, () => ListTrafficPolicyInstancesByHostedZoneRequest$, () => ListTrafficPolicyInstancesByHostedZoneResponse$
];
var ListTrafficPolicyInstancesByPolicy$ = [9, n0, _LTPIBP,
    { [_ht]: ["GET", "/2013-04-01/trafficpolicyinstances/trafficpolicy", 200] }, () => ListTrafficPolicyInstancesByPolicyRequest$, () => ListTrafficPolicyInstancesByPolicyResponse$
];
var ListTrafficPolicyVersions$ = [9, n0, _LTPV,
    { [_ht]: ["GET", "/2013-04-01/trafficpolicies/{Id}/versions", 200] }, () => ListTrafficPolicyVersionsRequest$, () => ListTrafficPolicyVersionsResponse$
];
var ListVPCAssociationAuthorizations$ = [9, n0, _LVPCAA,
    { [_ht]: ["GET", "/2013-04-01/hostedzone/{HostedZoneId}/authorizevpcassociation", 200] }, () => ListVPCAssociationAuthorizationsRequest$, () => ListVPCAssociationAuthorizationsResponse$
];
var TestDNSAnswer$ = [9, n0, _TDNSA,
    { [_ht]: ["GET", "/2013-04-01/testdnsanswer", 200] }, () => TestDNSAnswerRequest$, () => TestDNSAnswerResponse$
];
var UpdateHealthCheck$ = [9, n0, _UHC,
    { [_ht]: ["POST", "/2013-04-01/healthcheck/{HealthCheckId}", 200] }, () => UpdateHealthCheckRequest$, () => UpdateHealthCheckResponse$
];
var UpdateHostedZoneComment$ = [9, n0, _UHZC,
    { [_ht]: ["POST", "/2013-04-01/hostedzone/{Id}", 200] }, () => UpdateHostedZoneCommentRequest$, () => UpdateHostedZoneCommentResponse$
];
var UpdateHostedZoneFeatures$ = [9, n0, _UHZF,
    { [_ht]: ["POST", "/2013-04-01/hostedzone/{HostedZoneId}/features", 200] }, () => UpdateHostedZoneFeaturesRequest$, () => UpdateHostedZoneFeaturesResponse$
];
var UpdateTrafficPolicyComment$ = [9, n0, _UTPC,
    { [_ht]: ["POST", "/2013-04-01/trafficpolicy/{Id}/{Version}", 200] }, () => UpdateTrafficPolicyCommentRequest$, () => UpdateTrafficPolicyCommentResponse$
];
var UpdateTrafficPolicyInstance$ = [9, n0, _UTPI,
    { [_ht]: ["POST", "/2013-04-01/trafficpolicyinstance/{Id}", 200] }, () => UpdateTrafficPolicyInstanceRequest$, () => UpdateTrafficPolicyInstanceResponse$
];

const getRuntimeConfig$1 = (config) => {
    return {
        apiVersion: "2013-04-01",
        base64Decoder: config?.base64Decoder ?? fromBase64,
        base64Encoder: config?.base64Encoder ?? toBase64,
        disableHostPrefix: config?.disableHostPrefix ?? false,
        endpointProvider: config?.endpointProvider ?? defaultEndpointResolver,
        extensions: config?.extensions ?? [],
        httpAuthSchemeProvider: config?.httpAuthSchemeProvider ?? defaultRoute53HttpAuthSchemeProvider,
        httpAuthSchemes: config?.httpAuthSchemes ?? [
            {
                schemeId: "aws.auth#sigv4",
                identityProvider: (ipc) => ipc.getIdentityProvider("aws.auth#sigv4"),
                signer: new AwsSdkSigV4Signer(),
            },
        ],
        logger: config?.logger ?? new NoOpLogger(),
        protocol: config?.protocol ?? AwsRestXmlProtocol,
        protocolSettings: config?.protocolSettings ?? {
            defaultNamespace: "com.amazonaws.route53",
            errorTypeRegistries,
            xmlNamespace: "https://route53.amazonaws.com/doc/2013-04-01/",
            version: "2013-04-01",
            serviceTarget: "AWSDnsV20130401",
        },
        serviceId: config?.serviceId ?? "Route 53",
        sha256: config?.sha256 ?? Sha256,
        urlParser: config?.urlParser ?? parseUrl,
        utf8Decoder: config?.utf8Decoder ?? fromUtf8,
        utf8Encoder: config?.utf8Encoder ?? toUtf8,
    };
};

const getRuntimeConfig = (config) => {
    emitWarningIfUnsupportedVersion(process.version);
    const defaultsMode = resolveDefaultsModeConfig(config);
    const defaultConfigProvider = () => defaultsMode().then(loadConfigsForDefaultMode);
    const clientSharedValues = getRuntimeConfig$1(config);
    emitWarningIfUnsupportedVersion$1(process.version);
    const loaderConfig = {
        profile: config?.profile,
        logger: clientSharedValues.logger,
    };
    return {
        ...clientSharedValues,
        ...config,
        runtime: "node",
        defaultsMode,
        authSchemePreference: config?.authSchemePreference ?? loadConfig(NODE_AUTH_SCHEME_PREFERENCE_OPTIONS, loaderConfig),
        bodyLengthChecker: config?.bodyLengthChecker ?? calculateBodyLength,
        credentialDefaultProvider: config?.credentialDefaultProvider ?? defaultProvider,
        defaultUserAgentProvider: config?.defaultUserAgentProvider ?? createDefaultUserAgentProvider({ serviceId: clientSharedValues.serviceId, clientVersion: packageInfo.version }),
        maxAttempts: config?.maxAttempts ?? loadConfig(NODE_MAX_ATTEMPT_CONFIG_OPTIONS, config),
        region: config?.region ?? loadConfig(NODE_REGION_CONFIG_OPTIONS, { ...NODE_REGION_CONFIG_FILE_OPTIONS, ...loaderConfig }),
        requestHandler: NodeHttpHandler.create(config?.requestHandler ?? defaultConfigProvider),
        retryMode: config?.retryMode ??
            loadConfig({
                ...NODE_RETRY_MODE_CONFIG_OPTIONS,
                default: async () => (await defaultConfigProvider()).retryMode || DEFAULT_RETRY_MODE,
            }, config),
        streamCollector: config?.streamCollector ?? streamCollector,
        useDualstackEndpoint: config?.useDualstackEndpoint ?? loadConfig(NODE_USE_DUALSTACK_ENDPOINT_CONFIG_OPTIONS, loaderConfig),
        useFipsEndpoint: config?.useFipsEndpoint ?? loadConfig(NODE_USE_FIPS_ENDPOINT_CONFIG_OPTIONS, loaderConfig),
        userAgentAppId: config?.userAgentAppId ?? loadConfig(NODE_APP_ID_CONFIG_OPTIONS, loaderConfig),
    };
};

const getHttpAuthExtensionConfiguration = (runtimeConfig) => {
    const _httpAuthSchemes = runtimeConfig.httpAuthSchemes;
    let _httpAuthSchemeProvider = runtimeConfig.httpAuthSchemeProvider;
    let _credentials = runtimeConfig.credentials;
    return {
        setHttpAuthScheme(httpAuthScheme) {
            const index = _httpAuthSchemes.findIndex((scheme) => scheme.schemeId === httpAuthScheme.schemeId);
            if (index === -1) {
                _httpAuthSchemes.push(httpAuthScheme);
            }
            else {
                _httpAuthSchemes.splice(index, 1, httpAuthScheme);
            }
        },
        httpAuthSchemes() {
            return _httpAuthSchemes;
        },
        setHttpAuthSchemeProvider(httpAuthSchemeProvider) {
            _httpAuthSchemeProvider = httpAuthSchemeProvider;
        },
        httpAuthSchemeProvider() {
            return _httpAuthSchemeProvider;
        },
        setCredentials(credentials) {
            _credentials = credentials;
        },
        credentials() {
            return _credentials;
        },
    };
};
const resolveHttpAuthRuntimeConfig = (config) => {
    return {
        httpAuthSchemes: config.httpAuthSchemes(),
        httpAuthSchemeProvider: config.httpAuthSchemeProvider(),
        credentials: config.credentials(),
    };
};

const resolveRuntimeExtensions = (runtimeConfig, extensions) => {
    const extensionConfiguration = Object.assign(getAwsRegionExtensionConfiguration(runtimeConfig), getDefaultExtensionConfiguration(runtimeConfig), getHttpHandlerExtensionConfiguration(runtimeConfig), getHttpAuthExtensionConfiguration(runtimeConfig));
    extensions.forEach((extension) => extension.configure(extensionConfiguration));
    return Object.assign(runtimeConfig, resolveAwsRegionExtensionConfiguration(extensionConfiguration), resolveDefaultRuntimeConfig(extensionConfiguration), resolveHttpHandlerRuntimeConfig(extensionConfiguration), resolveHttpAuthRuntimeConfig(extensionConfiguration));
};

class Route53Client extends Client {
    config;
    constructor(...[configuration]) {
        const _config_0 = getRuntimeConfig(configuration || {});
        super(_config_0);
        this.initConfig = _config_0;
        const _config_1 = resolveClientEndpointParameters(_config_0);
        const _config_2 = resolveUserAgentConfig(_config_1);
        const _config_3 = resolveRetryConfig(_config_2);
        const _config_4 = resolveRegionConfig(_config_3);
        const _config_5 = resolveHostHeaderConfig(_config_4);
        const _config_6 = resolveEndpointConfig(_config_5);
        const _config_7 = resolveHttpAuthSchemeConfig(_config_6);
        const _config_8 = resolveRuntimeExtensions(_config_7, configuration?.extensions || []);
        this.config = _config_8;
        this.middlewareStack.use(getSchemaSerdePlugin(this.config));
        this.middlewareStack.use(getUserAgentPlugin(this.config));
        this.middlewareStack.use(getRetryPlugin(this.config));
        this.middlewareStack.use(getContentLengthPlugin(this.config));
        this.middlewareStack.use(getHostHeaderPlugin(this.config));
        this.middlewareStack.use(getLoggerPlugin(this.config));
        this.middlewareStack.use(getRecursionDetectionPlugin(this.config));
        this.middlewareStack.use(getHttpAuthSchemeEndpointRuleSetPlugin(this.config, {
            httpAuthSchemeParametersProvider: defaultRoute53HttpAuthSchemeParametersProvider,
            identityProviderConfigProvider: async (config) => new DefaultIdentityProviderConfig({
                "aws.auth#sigv4": config.credentials,
            }),
        }));
        this.middlewareStack.use(getHttpSigningPlugin(this.config));
    }
    destroy() {
        super.destroy();
    }
}

const command = makeBuilder(commonParams, "AWSDnsV20130401", "Route53Client", getEndpointPlugin);
const _ep0 = {};
const _mw0 = (Command, cs, config, o) => [
    getIdNormalizerPlugin(config),
];
const _mw1 = (Command, cs, config, o) => [
    getChangeResourceRecordSetsPlugin(config),
    getIdNormalizerPlugin(config),
];
const _mw2 = (Command, cs, config, o) => [];

class ActivateKeySigningKeyCommand extends command(_ep0, _mw0, "ActivateKeySigningKey", ActivateKeySigningKey$) {
}

class AssociateVPCWithHostedZoneCommand extends command(_ep0, _mw0, "AssociateVPCWithHostedZone", AssociateVPCWithHostedZone$) {
}

class ChangeCidrCollectionCommand extends command(_ep0, _mw0, "ChangeCidrCollection", ChangeCidrCollection$) {
}

class ChangeResourceRecordSetsCommand extends command(_ep0, _mw1, "ChangeResourceRecordSets", ChangeResourceRecordSets$) {
}

class ChangeTagsForResourceCommand extends command(_ep0, _mw2, "ChangeTagsForResource", ChangeTagsForResource$) {
}

class CreateCidrCollectionCommand extends command(_ep0, _mw2, "CreateCidrCollection", CreateCidrCollection$) {
}

class CreateHealthCheckCommand extends command(_ep0, _mw2, "CreateHealthCheck", CreateHealthCheck$) {
}

class CreateHostedZoneCommand extends command(_ep0, _mw0, "CreateHostedZone", CreateHostedZone$) {
}

class CreateKeySigningKeyCommand extends command(_ep0, _mw0, "CreateKeySigningKey", CreateKeySigningKey$) {
}

class CreateQueryLoggingConfigCommand extends command(_ep0, _mw0, "CreateQueryLoggingConfig", CreateQueryLoggingConfig$) {
}

class CreateReusableDelegationSetCommand extends command(_ep0, _mw0, "CreateReusableDelegationSet", CreateReusableDelegationSet$) {
}

class CreateTrafficPolicyCommand extends command(_ep0, _mw2, "CreateTrafficPolicy", CreateTrafficPolicy$) {
}

class CreateTrafficPolicyInstanceCommand extends command(_ep0, _mw0, "CreateTrafficPolicyInstance", CreateTrafficPolicyInstance$) {
}

class CreateTrafficPolicyVersionCommand extends command(_ep0, _mw0, "CreateTrafficPolicyVersion", CreateTrafficPolicyVersion$) {
}

class CreateVPCAssociationAuthorizationCommand extends command(_ep0, _mw0, "CreateVPCAssociationAuthorization", CreateVPCAssociationAuthorization$) {
}

class DeactivateKeySigningKeyCommand extends command(_ep0, _mw0, "DeactivateKeySigningKey", DeactivateKeySigningKey$) {
}

class DeleteCidrCollectionCommand extends command(_ep0, _mw0, "DeleteCidrCollection", DeleteCidrCollection$) {
}

class DeleteHealthCheckCommand extends command(_ep0, _mw2, "DeleteHealthCheck", DeleteHealthCheck$) {
}

class DeleteHostedZoneCommand extends command(_ep0, _mw0, "DeleteHostedZone", DeleteHostedZone$) {
}

class DeleteKeySigningKeyCommand extends command(_ep0, _mw0, "DeleteKeySigningKey", DeleteKeySigningKey$) {
}

class DeleteQueryLoggingConfigCommand extends command(_ep0, _mw0, "DeleteQueryLoggingConfig", DeleteQueryLoggingConfig$) {
}

class DeleteReusableDelegationSetCommand extends command(_ep0, _mw0, "DeleteReusableDelegationSet", DeleteReusableDelegationSet$) {
}

class DeleteTrafficPolicyCommand extends command(_ep0, _mw0, "DeleteTrafficPolicy", DeleteTrafficPolicy$) {
}

class DeleteTrafficPolicyInstanceCommand extends command(_ep0, _mw0, "DeleteTrafficPolicyInstance", DeleteTrafficPolicyInstance$) {
}

class DeleteVPCAssociationAuthorizationCommand extends command(_ep0, _mw0, "DeleteVPCAssociationAuthorization", DeleteVPCAssociationAuthorization$) {
}

class DisableHostedZoneDNSSECCommand extends command(_ep0, _mw0, "DisableHostedZoneDNSSEC", DisableHostedZoneDNSSEC$) {
}

class DisassociateVPCFromHostedZoneCommand extends command(_ep0, _mw0, "DisassociateVPCFromHostedZone", DisassociateVPCFromHostedZone$) {
}

class EnableHostedZoneDNSSECCommand extends command(_ep0, _mw0, "EnableHostedZoneDNSSEC", EnableHostedZoneDNSSEC$) {
}

class GetAccountLimitCommand extends command(_ep0, _mw2, "GetAccountLimit", GetAccountLimit$) {
}

class GetChangeCommand extends command(_ep0, _mw0, "GetChange", GetChange$) {
}

class GetCheckerIpRangesCommand extends command(_ep0, _mw2, "GetCheckerIpRanges", GetCheckerIpRanges$) {
}

class GetDNSSECCommand extends command(_ep0, _mw0, "GetDNSSEC", GetDNSSEC$) {
}

class GetGeoLocationCommand extends command(_ep0, _mw2, "GetGeoLocation", GetGeoLocation$) {
}

class GetHealthCheckCommand extends command(_ep0, _mw2, "GetHealthCheck", GetHealthCheck$) {
}

class GetHealthCheckCountCommand extends command(_ep0, _mw2, "GetHealthCheckCount", GetHealthCheckCount$) {
}

class GetHealthCheckLastFailureReasonCommand extends command(_ep0, _mw2, "GetHealthCheckLastFailureReason", GetHealthCheckLastFailureReason$) {
}

class GetHealthCheckStatusCommand extends command(_ep0, _mw2, "GetHealthCheckStatus", GetHealthCheckStatus$) {
}

class GetHostedZoneCommand extends command(_ep0, _mw0, "GetHostedZone", GetHostedZone$) {
}

class GetHostedZoneCountCommand extends command(_ep0, _mw2, "GetHostedZoneCount", GetHostedZoneCount$) {
}

class GetHostedZoneLimitCommand extends command(_ep0, _mw0, "GetHostedZoneLimit", GetHostedZoneLimit$) {
}

class GetQueryLoggingConfigCommand extends command(_ep0, _mw0, "GetQueryLoggingConfig", GetQueryLoggingConfig$) {
}

class GetReusableDelegationSetCommand extends command(_ep0, _mw0, "GetReusableDelegationSet", GetReusableDelegationSet$) {
}

class GetReusableDelegationSetLimitCommand extends command(_ep0, _mw0, "GetReusableDelegationSetLimit", GetReusableDelegationSetLimit$) {
}

class GetTrafficPolicyCommand extends command(_ep0, _mw0, "GetTrafficPolicy", GetTrafficPolicy$) {
}

class GetTrafficPolicyInstanceCommand extends command(_ep0, _mw0, "GetTrafficPolicyInstance", GetTrafficPolicyInstance$) {
}

class GetTrafficPolicyInstanceCountCommand extends command(_ep0, _mw2, "GetTrafficPolicyInstanceCount", GetTrafficPolicyInstanceCount$) {
}

class ListCidrBlocksCommand extends command(_ep0, _mw2, "ListCidrBlocks", ListCidrBlocks$) {
}

class ListCidrCollectionsCommand extends command(_ep0, _mw2, "ListCidrCollections", ListCidrCollections$) {
}

class ListCidrLocationsCommand extends command(_ep0, _mw2, "ListCidrLocations", ListCidrLocations$) {
}

class ListGeoLocationsCommand extends command(_ep0, _mw2, "ListGeoLocations", ListGeoLocations$) {
}

class ListHealthChecksCommand extends command(_ep0, _mw2, "ListHealthChecks", ListHealthChecks$) {
}

class ListHostedZonesByNameCommand extends command(_ep0, _mw0, "ListHostedZonesByName", ListHostedZonesByName$) {
}

class ListHostedZonesByVPCCommand extends command(_ep0, _mw2, "ListHostedZonesByVPC", ListHostedZonesByVPC$) {
}

class ListHostedZonesCommand extends command(_ep0, _mw0, "ListHostedZones", ListHostedZones$) {
}

class ListQueryLoggingConfigsCommand extends command(_ep0, _mw0, "ListQueryLoggingConfigs", ListQueryLoggingConfigs$) {
}

class ListResourceRecordSetsCommand extends command(_ep0, _mw0, "ListResourceRecordSets", ListResourceRecordSets$) {
}

class ListReusableDelegationSetsCommand extends command(_ep0, _mw2, "ListReusableDelegationSets", ListReusableDelegationSets$) {
}

class ListTagsForResourceCommand extends command(_ep0, _mw2, "ListTagsForResource", ListTagsForResource$) {
}

class ListTagsForResourcesCommand extends command(_ep0, _mw2, "ListTagsForResources", ListTagsForResources$) {
}

class ListTrafficPoliciesCommand extends command(_ep0, _mw2, "ListTrafficPolicies", ListTrafficPolicies$) {
}

class ListTrafficPolicyInstancesByHostedZoneCommand extends command(_ep0, _mw0, "ListTrafficPolicyInstancesByHostedZone", ListTrafficPolicyInstancesByHostedZone$) {
}

class ListTrafficPolicyInstancesByPolicyCommand extends command(_ep0, _mw2, "ListTrafficPolicyInstancesByPolicy", ListTrafficPolicyInstancesByPolicy$) {
}

class ListTrafficPolicyInstancesCommand extends command(_ep0, _mw2, "ListTrafficPolicyInstances", ListTrafficPolicyInstances$) {
}

class ListTrafficPolicyVersionsCommand extends command(_ep0, _mw0, "ListTrafficPolicyVersions", ListTrafficPolicyVersions$) {
}

class ListVPCAssociationAuthorizationsCommand extends command(_ep0, _mw0, "ListVPCAssociationAuthorizations", ListVPCAssociationAuthorizations$) {
}

class TestDNSAnswerCommand extends command(_ep0, _mw0, "TestDNSAnswer", TestDNSAnswer$) {
}

class UpdateHealthCheckCommand extends command(_ep0, _mw2, "UpdateHealthCheck", UpdateHealthCheck$) {
}

class UpdateHostedZoneCommentCommand extends command(_ep0, _mw0, "UpdateHostedZoneComment", UpdateHostedZoneComment$) {
}

class UpdateHostedZoneFeaturesCommand extends command(_ep0, _mw0, "UpdateHostedZoneFeatures", UpdateHostedZoneFeatures$) {
}

class UpdateTrafficPolicyCommentCommand extends command(_ep0, _mw0, "UpdateTrafficPolicyComment", UpdateTrafficPolicyComment$) {
}

class UpdateTrafficPolicyInstanceCommand extends command(_ep0, _mw0, "UpdateTrafficPolicyInstance", UpdateTrafficPolicyInstance$) {
}

const paginateListCidrBlocks = createPaginator(Route53Client, ListCidrBlocksCommand, "NextToken", "NextToken", "MaxResults");

const paginateListCidrCollections = createPaginator(Route53Client, ListCidrCollectionsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListCidrLocations = createPaginator(Route53Client, ListCidrLocationsCommand, "NextToken", "NextToken", "MaxResults");

const paginateListHealthChecks = createPaginator(Route53Client, ListHealthChecksCommand, "Marker", "NextMarker", "MaxItems");

const paginateListHostedZones = createPaginator(Route53Client, ListHostedZonesCommand, "Marker", "NextMarker", "MaxItems");

const paginateListQueryLoggingConfigs = createPaginator(Route53Client, ListQueryLoggingConfigsCommand, "NextToken", "NextToken", "MaxResults");

const checkState = async (client, input) => {
    let reason;
    try {
        let result = await client.send(new GetChangeCommand(input));
        reason = result;
        try {
            const returnComparator = () => {
                return result.ChangeInfo.Status;
            };
            if (returnComparator() === "INSYNC") {
                return { state: WaiterState.SUCCESS, reason };
            }
        }
        catch (e) { }
    }
    catch (exception) {
        reason = exception;
    }
    return { state: WaiterState.RETRY, reason };
};
const waitForResourceRecordSetsChanged = async (params, input) => {
    const serviceDefaults = { minDelay: 30, maxDelay: 120 };
    return createWaiter({ ...serviceDefaults, ...params }, input, checkState);
};
const waitUntilResourceRecordSetsChanged = async (params, input) => {
    const serviceDefaults = { minDelay: 30, maxDelay: 120 };
    const result = await createWaiter({ ...serviceDefaults, ...params }, input, checkState);
    return checkExceptions(result);
};

const commands = {
    ActivateKeySigningKeyCommand,
    AssociateVPCWithHostedZoneCommand,
    ChangeCidrCollectionCommand,
    ChangeResourceRecordSetsCommand,
    ChangeTagsForResourceCommand,
    CreateCidrCollectionCommand,
    CreateHealthCheckCommand,
    CreateHostedZoneCommand,
    CreateKeySigningKeyCommand,
    CreateQueryLoggingConfigCommand,
    CreateReusableDelegationSetCommand,
    CreateTrafficPolicyCommand,
    CreateTrafficPolicyInstanceCommand,
    CreateTrafficPolicyVersionCommand,
    CreateVPCAssociationAuthorizationCommand,
    DeactivateKeySigningKeyCommand,
    DeleteCidrCollectionCommand,
    DeleteHealthCheckCommand,
    DeleteHostedZoneCommand,
    DeleteKeySigningKeyCommand,
    DeleteQueryLoggingConfigCommand,
    DeleteReusableDelegationSetCommand,
    DeleteTrafficPolicyCommand,
    DeleteTrafficPolicyInstanceCommand,
    DeleteVPCAssociationAuthorizationCommand,
    DisableHostedZoneDNSSECCommand,
    DisassociateVPCFromHostedZoneCommand,
    EnableHostedZoneDNSSECCommand,
    GetAccountLimitCommand,
    GetChangeCommand,
    GetCheckerIpRangesCommand,
    GetDNSSECCommand,
    GetGeoLocationCommand,
    GetHealthCheckCommand,
    GetHealthCheckCountCommand,
    GetHealthCheckLastFailureReasonCommand,
    GetHealthCheckStatusCommand,
    GetHostedZoneCommand,
    GetHostedZoneCountCommand,
    GetHostedZoneLimitCommand,
    GetQueryLoggingConfigCommand,
    GetReusableDelegationSetCommand,
    GetReusableDelegationSetLimitCommand,
    GetTrafficPolicyCommand,
    GetTrafficPolicyInstanceCommand,
    GetTrafficPolicyInstanceCountCommand,
    ListCidrBlocksCommand,
    ListCidrCollectionsCommand,
    ListCidrLocationsCommand,
    ListGeoLocationsCommand,
    ListHealthChecksCommand,
    ListHostedZonesCommand,
    ListHostedZonesByNameCommand,
    ListHostedZonesByVPCCommand,
    ListQueryLoggingConfigsCommand,
    ListResourceRecordSetsCommand,
    ListReusableDelegationSetsCommand,
    ListTagsForResourceCommand,
    ListTagsForResourcesCommand,
    ListTrafficPoliciesCommand,
    ListTrafficPolicyInstancesCommand,
    ListTrafficPolicyInstancesByHostedZoneCommand,
    ListTrafficPolicyInstancesByPolicyCommand,
    ListTrafficPolicyVersionsCommand,
    ListVPCAssociationAuthorizationsCommand,
    TestDNSAnswerCommand,
    UpdateHealthCheckCommand,
    UpdateHostedZoneCommentCommand,
    UpdateHostedZoneFeaturesCommand,
    UpdateTrafficPolicyCommentCommand,
    UpdateTrafficPolicyInstanceCommand,
};
const paginators = {
    paginateListCidrBlocks,
    paginateListCidrCollections,
    paginateListCidrLocations,
    paginateListHealthChecks,
    paginateListHostedZones,
    paginateListQueryLoggingConfigs,
};
const waiters = {
    waitUntilResourceRecordSetsChanged,
};
class Route53 extends Route53Client {
}
createAggregatedClient(commands, Route53, { paginators, waiters });

const AcceleratedRecoveryStatus = {
    DISABLED: "DISABLED",
    DISABLE_FAILED: "DISABLE_FAILED",
    DISABLING: "DISABLING",
    DISABLING_HOSTED_ZONE_LOCKED: "DISABLING_HOSTED_ZONE_LOCKED",
    ENABLED: "ENABLED",
    ENABLE_FAILED: "ENABLE_FAILED",
    ENABLING: "ENABLING",
    ENABLING_HOSTED_ZONE_LOCKED: "ENABLING_HOSTED_ZONE_LOCKED",
};
const AccountLimitType = {
    MAX_HEALTH_CHECKS_BY_OWNER: "MAX_HEALTH_CHECKS_BY_OWNER",
    MAX_HOSTED_ZONES_BY_OWNER: "MAX_HOSTED_ZONES_BY_OWNER",
    MAX_REUSABLE_DELEGATION_SETS_BY_OWNER: "MAX_REUSABLE_DELEGATION_SETS_BY_OWNER",
    MAX_TRAFFIC_POLICIES_BY_OWNER: "MAX_TRAFFIC_POLICIES_BY_OWNER",
    MAX_TRAFFIC_POLICY_INSTANCES_BY_OWNER: "MAX_TRAFFIC_POLICY_INSTANCES_BY_OWNER",
};
const ChangeStatus = {
    INSYNC: "INSYNC",
    PENDING: "PENDING",
};
const CloudWatchRegion = {
    af_south_1: "af-south-1",
    ap_east_1: "ap-east-1",
    ap_east_2: "ap-east-2",
    ap_northeast_1: "ap-northeast-1",
    ap_northeast_2: "ap-northeast-2",
    ap_northeast_3: "ap-northeast-3",
    ap_south_1: "ap-south-1",
    ap_south_2: "ap-south-2",
    ap_southeast_1: "ap-southeast-1",
    ap_southeast_2: "ap-southeast-2",
    ap_southeast_3: "ap-southeast-3",
    ap_southeast_4: "ap-southeast-4",
    ap_southeast_5: "ap-southeast-5",
    ap_southeast_6: "ap-southeast-6",
    ap_southeast_7: "ap-southeast-7",
    ca_central_1: "ca-central-1",
    ca_west_1: "ca-west-1",
    cn_north_1: "cn-north-1",
    cn_northwest_1: "cn-northwest-1",
    eu_central_1: "eu-central-1",
    eu_central_2: "eu-central-2",
    eu_isoe_west_1: "eu-isoe-west-1",
    eu_north_1: "eu-north-1",
    eu_south_1: "eu-south-1",
    eu_south_2: "eu-south-2",
    eu_west_1: "eu-west-1",
    eu_west_2: "eu-west-2",
    eu_west_3: "eu-west-3",
    eusc_de_east_1: "eusc-de-east-1",
    il_central_1: "il-central-1",
    me_central_1: "me-central-1",
    me_south_1: "me-south-1",
    mx_central_1: "mx-central-1",
    sa_east_1: "sa-east-1",
    us_east_1: "us-east-1",
    us_east_2: "us-east-2",
    us_gov_east_1: "us-gov-east-1",
    us_gov_west_1: "us-gov-west-1",
    us_iso_east_1: "us-iso-east-1",
    us_iso_west_1: "us-iso-west-1",
    us_isob_east_1: "us-isob-east-1",
    us_isob_west_1: "us-isob-west-1",
    us_isof_east_1: "us-isof-east-1",
    us_isof_south_1: "us-isof-south-1",
    us_west_1: "us-west-1",
    us_west_2: "us-west-2",
};
const VPCRegion = {
    af_south_1: "af-south-1",
    ap_east_1: "ap-east-1",
    ap_east_2: "ap-east-2",
    ap_northeast_1: "ap-northeast-1",
    ap_northeast_2: "ap-northeast-2",
    ap_northeast_3: "ap-northeast-3",
    ap_south_1: "ap-south-1",
    ap_south_2: "ap-south-2",
    ap_southeast_1: "ap-southeast-1",
    ap_southeast_2: "ap-southeast-2",
    ap_southeast_3: "ap-southeast-3",
    ap_southeast_4: "ap-southeast-4",
    ap_southeast_5: "ap-southeast-5",
    ap_southeast_6: "ap-southeast-6",
    ap_southeast_7: "ap-southeast-7",
    ca_central_1: "ca-central-1",
    ca_west_1: "ca-west-1",
    cn_north_1: "cn-north-1",
    cn_northwest_1: "cn-northwest-1",
    eu_central_1: "eu-central-1",
    eu_central_2: "eu-central-2",
    eu_isoe_west_1: "eu-isoe-west-1",
    eu_north_1: "eu-north-1",
    eu_south_1: "eu-south-1",
    eu_south_2: "eu-south-2",
    eu_west_1: "eu-west-1",
    eu_west_2: "eu-west-2",
    eu_west_3: "eu-west-3",
    eusc_de_east_1: "eusc-de-east-1",
    il_central_1: "il-central-1",
    me_central_1: "me-central-1",
    me_south_1: "me-south-1",
    mx_central_1: "mx-central-1",
    sa_east_1: "sa-east-1",
    us_east_1: "us-east-1",
    us_east_2: "us-east-2",
    us_gov_east_1: "us-gov-east-1",
    us_gov_west_1: "us-gov-west-1",
    us_iso_east_1: "us-iso-east-1",
    us_iso_west_1: "us-iso-west-1",
    us_isob_east_1: "us-isob-east-1",
    us_isob_west_1: "us-isob-west-1",
    us_isof_east_1: "us-isof-east-1",
    us_isof_south_1: "us-isof-south-1",
    us_west_1: "us-west-1",
    us_west_2: "us-west-2",
};
const CidrCollectionChangeAction = {
    DELETE_IF_EXISTS: "DELETE_IF_EXISTS",
    PUT: "PUT",
};
const ChangeAction = {
    CREATE: "CREATE",
    DELETE: "DELETE",
    UPSERT: "UPSERT",
};
const ResourceRecordSetFailover = {
    PRIMARY: "PRIMARY",
    SECONDARY: "SECONDARY",
};
const ResourceRecordSetRegion = {
    af_south_1: "af-south-1",
    ap_east_1: "ap-east-1",
    ap_east_2: "ap-east-2",
    ap_northeast_1: "ap-northeast-1",
    ap_northeast_2: "ap-northeast-2",
    ap_northeast_3: "ap-northeast-3",
    ap_south_1: "ap-south-1",
    ap_south_2: "ap-south-2",
    ap_southeast_1: "ap-southeast-1",
    ap_southeast_2: "ap-southeast-2",
    ap_southeast_3: "ap-southeast-3",
    ap_southeast_4: "ap-southeast-4",
    ap_southeast_5: "ap-southeast-5",
    ap_southeast_6: "ap-southeast-6",
    ap_southeast_7: "ap-southeast-7",
    ca_central_1: "ca-central-1",
    ca_west_1: "ca-west-1",
    cn_north_1: "cn-north-1",
    cn_northwest_1: "cn-northwest-1",
    eu_central_1: "eu-central-1",
    eu_central_2: "eu-central-2",
    eu_north_1: "eu-north-1",
    eu_south_1: "eu-south-1",
    eu_south_2: "eu-south-2",
    eu_west_1: "eu-west-1",
    eu_west_2: "eu-west-2",
    eu_west_3: "eu-west-3",
    eusc_de_east_1: "eusc-de-east-1",
    il_central_1: "il-central-1",
    me_central_1: "me-central-1",
    me_south_1: "me-south-1",
    mx_central_1: "mx-central-1",
    sa_east_1: "sa-east-1",
    us_east_1: "us-east-1",
    us_east_2: "us-east-2",
    us_gov_east_1: "us-gov-east-1",
    us_gov_west_1: "us-gov-west-1",
    us_west_1: "us-west-1",
    us_west_2: "us-west-2",
};
const RRType = {
    A: "A",
    AAAA: "AAAA",
    CAA: "CAA",
    CNAME: "CNAME",
    DS: "DS",
    HTTPS: "HTTPS",
    MX: "MX",
    NAPTR: "NAPTR",
    NS: "NS",
    PTR: "PTR",
    SOA: "SOA",
    SPF: "SPF",
    SRV: "SRV",
    SSHFP: "SSHFP",
    SVCB: "SVCB",
    TLSA: "TLSA",
    TXT: "TXT",
};
const TagResourceType = {
    healthcheck: "healthcheck",
    hostedzone: "hostedzone",
};
const InsufficientDataHealthStatus = {
    Healthy: "Healthy",
    LastKnownStatus: "LastKnownStatus",
    Unhealthy: "Unhealthy",
};
const HealthCheckRegion = {
    ap_northeast_1: "ap-northeast-1",
    ap_southeast_1: "ap-southeast-1",
    ap_southeast_2: "ap-southeast-2",
    eu_west_1: "eu-west-1",
    sa_east_1: "sa-east-1",
    us_east_1: "us-east-1",
    us_west_1: "us-west-1",
    us_west_2: "us-west-2",
};
const HealthCheckType = {
    CALCULATED: "CALCULATED",
    CLOUDWATCH_METRIC: "CLOUDWATCH_METRIC",
    HTTP: "HTTP",
    HTTPS: "HTTPS",
    HTTPS_STR_MATCH: "HTTPS_STR_MATCH",
    HTTP_STR_MATCH: "HTTP_STR_MATCH",
    RECOVERY_CONTROL: "RECOVERY_CONTROL",
    TCP: "TCP",
};
const ComparisonOperator = {
    GreaterThanOrEqualToThreshold: "GreaterThanOrEqualToThreshold",
    GreaterThanThreshold: "GreaterThanThreshold",
    LessThanOrEqualToThreshold: "LessThanOrEqualToThreshold",
    LessThanThreshold: "LessThanThreshold",
};
const Statistic = {
    Average: "Average",
    Maximum: "Maximum",
    Minimum: "Minimum",
    SampleCount: "SampleCount",
    Sum: "Sum",
};
const HostedZoneLimitType = {
    MAX_RRSETS_BY_ZONE: "MAX_RRSETS_BY_ZONE",
    MAX_VPCS_ASSOCIATED_BY_ZONE: "MAX_VPCS_ASSOCIATED_BY_ZONE",
};
const ReusableDelegationSetLimitType = {
    MAX_ZONES_BY_REUSABLE_DELEGATION_SET: "MAX_ZONES_BY_REUSABLE_DELEGATION_SET",
};
const HostedZoneType = {
    PRIVATE_HOSTED_ZONE: "PrivateHostedZone",
};
const ResettableElementName = {
    ChildHealthChecks: "ChildHealthChecks",
    FullyQualifiedDomainName: "FullyQualifiedDomainName",
    Regions: "Regions",
    ResourcePath: "ResourcePath",
};

exports.AcceleratedRecoveryStatus = AcceleratedRecoveryStatus;
exports.AccountLimit$ = AccountLimit$;
exports.AccountLimitType = AccountLimitType;
exports.ActivateKeySigningKey$ = ActivateKeySigningKey$;
exports.ActivateKeySigningKeyCommand = ActivateKeySigningKeyCommand;
exports.ActivateKeySigningKeyRequest$ = ActivateKeySigningKeyRequest$;
exports.ActivateKeySigningKeyResponse$ = ActivateKeySigningKeyResponse$;
exports.AlarmIdentifier$ = AlarmIdentifier$;
exports.AliasTarget$ = AliasTarget$;
exports.AssociateVPCWithHostedZone$ = AssociateVPCWithHostedZone$;
exports.AssociateVPCWithHostedZoneCommand = AssociateVPCWithHostedZoneCommand;
exports.AssociateVPCWithHostedZoneRequest$ = AssociateVPCWithHostedZoneRequest$;
exports.AssociateVPCWithHostedZoneResponse$ = AssociateVPCWithHostedZoneResponse$;
exports.Change$ = Change$;
exports.ChangeAction = ChangeAction;
exports.ChangeBatch$ = ChangeBatch$;
exports.ChangeCidrCollection$ = ChangeCidrCollection$;
exports.ChangeCidrCollectionCommand = ChangeCidrCollectionCommand;
exports.ChangeCidrCollectionRequest$ = ChangeCidrCollectionRequest$;
exports.ChangeCidrCollectionResponse$ = ChangeCidrCollectionResponse$;
exports.ChangeInfo$ = ChangeInfo$;
exports.ChangeResourceRecordSets$ = ChangeResourceRecordSets$;
exports.ChangeResourceRecordSetsCommand = ChangeResourceRecordSetsCommand;
exports.ChangeResourceRecordSetsRequest$ = ChangeResourceRecordSetsRequest$;
exports.ChangeResourceRecordSetsResponse$ = ChangeResourceRecordSetsResponse$;
exports.ChangeStatus = ChangeStatus;
exports.ChangeTagsForResource$ = ChangeTagsForResource$;
exports.ChangeTagsForResourceCommand = ChangeTagsForResourceCommand;
exports.ChangeTagsForResourceRequest$ = ChangeTagsForResourceRequest$;
exports.ChangeTagsForResourceResponse$ = ChangeTagsForResourceResponse$;
exports.CidrBlockInUseException = CidrBlockInUseException;
exports.CidrBlockInUseException$ = CidrBlockInUseException$;
exports.CidrBlockSummary$ = CidrBlockSummary$;
exports.CidrCollection$ = CidrCollection$;
exports.CidrCollectionAlreadyExistsException = CidrCollectionAlreadyExistsException;
exports.CidrCollectionAlreadyExistsException$ = CidrCollectionAlreadyExistsException$;
exports.CidrCollectionChange$ = CidrCollectionChange$;
exports.CidrCollectionChangeAction = CidrCollectionChangeAction;
exports.CidrCollectionInUseException = CidrCollectionInUseException;
exports.CidrCollectionInUseException$ = CidrCollectionInUseException$;
exports.CidrCollectionVersionMismatchException = CidrCollectionVersionMismatchException;
exports.CidrCollectionVersionMismatchException$ = CidrCollectionVersionMismatchException$;
exports.CidrRoutingConfig$ = CidrRoutingConfig$;
exports.CloudWatchAlarmConfiguration$ = CloudWatchAlarmConfiguration$;
exports.CloudWatchRegion = CloudWatchRegion;
exports.CollectionSummary$ = CollectionSummary$;
exports.ComparisonOperator = ComparisonOperator;
exports.ConcurrentModification = ConcurrentModification;
exports.ConcurrentModification$ = ConcurrentModification$;
exports.ConflictingDomainExists = ConflictingDomainExists;
exports.ConflictingDomainExists$ = ConflictingDomainExists$;
exports.ConflictingTypes = ConflictingTypes;
exports.ConflictingTypes$ = ConflictingTypes$;
exports.Coordinates$ = Coordinates$;
exports.CreateCidrCollection$ = CreateCidrCollection$;
exports.CreateCidrCollectionCommand = CreateCidrCollectionCommand;
exports.CreateCidrCollectionRequest$ = CreateCidrCollectionRequest$;
exports.CreateCidrCollectionResponse$ = CreateCidrCollectionResponse$;
exports.CreateHealthCheck$ = CreateHealthCheck$;
exports.CreateHealthCheckCommand = CreateHealthCheckCommand;
exports.CreateHealthCheckRequest$ = CreateHealthCheckRequest$;
exports.CreateHealthCheckResponse$ = CreateHealthCheckResponse$;
exports.CreateHostedZone$ = CreateHostedZone$;
exports.CreateHostedZoneCommand = CreateHostedZoneCommand;
exports.CreateHostedZoneRequest$ = CreateHostedZoneRequest$;
exports.CreateHostedZoneResponse$ = CreateHostedZoneResponse$;
exports.CreateKeySigningKey$ = CreateKeySigningKey$;
exports.CreateKeySigningKeyCommand = CreateKeySigningKeyCommand;
exports.CreateKeySigningKeyRequest$ = CreateKeySigningKeyRequest$;
exports.CreateKeySigningKeyResponse$ = CreateKeySigningKeyResponse$;
exports.CreateQueryLoggingConfig$ = CreateQueryLoggingConfig$;
exports.CreateQueryLoggingConfigCommand = CreateQueryLoggingConfigCommand;
exports.CreateQueryLoggingConfigRequest$ = CreateQueryLoggingConfigRequest$;
exports.CreateQueryLoggingConfigResponse$ = CreateQueryLoggingConfigResponse$;
exports.CreateReusableDelegationSet$ = CreateReusableDelegationSet$;
exports.CreateReusableDelegationSetCommand = CreateReusableDelegationSetCommand;
exports.CreateReusableDelegationSetRequest$ = CreateReusableDelegationSetRequest$;
exports.CreateReusableDelegationSetResponse$ = CreateReusableDelegationSetResponse$;
exports.CreateTrafficPolicy$ = CreateTrafficPolicy$;
exports.CreateTrafficPolicyCommand = CreateTrafficPolicyCommand;
exports.CreateTrafficPolicyInstance$ = CreateTrafficPolicyInstance$;
exports.CreateTrafficPolicyInstanceCommand = CreateTrafficPolicyInstanceCommand;
exports.CreateTrafficPolicyInstanceRequest$ = CreateTrafficPolicyInstanceRequest$;
exports.CreateTrafficPolicyInstanceResponse$ = CreateTrafficPolicyInstanceResponse$;
exports.CreateTrafficPolicyRequest$ = CreateTrafficPolicyRequest$;
exports.CreateTrafficPolicyResponse$ = CreateTrafficPolicyResponse$;
exports.CreateTrafficPolicyVersion$ = CreateTrafficPolicyVersion$;
exports.CreateTrafficPolicyVersionCommand = CreateTrafficPolicyVersionCommand;
exports.CreateTrafficPolicyVersionRequest$ = CreateTrafficPolicyVersionRequest$;
exports.CreateTrafficPolicyVersionResponse$ = CreateTrafficPolicyVersionResponse$;
exports.CreateVPCAssociationAuthorization$ = CreateVPCAssociationAuthorization$;
exports.CreateVPCAssociationAuthorizationCommand = CreateVPCAssociationAuthorizationCommand;
exports.CreateVPCAssociationAuthorizationRequest$ = CreateVPCAssociationAuthorizationRequest$;
exports.CreateVPCAssociationAuthorizationResponse$ = CreateVPCAssociationAuthorizationResponse$;
exports.DNSSECNotFound = DNSSECNotFound;
exports.DNSSECNotFound$ = DNSSECNotFound$;
exports.DNSSECStatus$ = DNSSECStatus$;
exports.DeactivateKeySigningKey$ = DeactivateKeySigningKey$;
exports.DeactivateKeySigningKeyCommand = DeactivateKeySigningKeyCommand;
exports.DeactivateKeySigningKeyRequest$ = DeactivateKeySigningKeyRequest$;
exports.DeactivateKeySigningKeyResponse$ = DeactivateKeySigningKeyResponse$;
exports.DelegationSet$ = DelegationSet$;
exports.DelegationSetAlreadyCreated = DelegationSetAlreadyCreated;
exports.DelegationSetAlreadyCreated$ = DelegationSetAlreadyCreated$;
exports.DelegationSetAlreadyReusable = DelegationSetAlreadyReusable;
exports.DelegationSetAlreadyReusable$ = DelegationSetAlreadyReusable$;
exports.DelegationSetInUse = DelegationSetInUse;
exports.DelegationSetInUse$ = DelegationSetInUse$;
exports.DelegationSetNotAvailable = DelegationSetNotAvailable;
exports.DelegationSetNotAvailable$ = DelegationSetNotAvailable$;
exports.DelegationSetNotReusable = DelegationSetNotReusable;
exports.DelegationSetNotReusable$ = DelegationSetNotReusable$;
exports.DeleteCidrCollection$ = DeleteCidrCollection$;
exports.DeleteCidrCollectionCommand = DeleteCidrCollectionCommand;
exports.DeleteCidrCollectionRequest$ = DeleteCidrCollectionRequest$;
exports.DeleteCidrCollectionResponse$ = DeleteCidrCollectionResponse$;
exports.DeleteHealthCheck$ = DeleteHealthCheck$;
exports.DeleteHealthCheckCommand = DeleteHealthCheckCommand;
exports.DeleteHealthCheckRequest$ = DeleteHealthCheckRequest$;
exports.DeleteHealthCheckResponse$ = DeleteHealthCheckResponse$;
exports.DeleteHostedZone$ = DeleteHostedZone$;
exports.DeleteHostedZoneCommand = DeleteHostedZoneCommand;
exports.DeleteHostedZoneRequest$ = DeleteHostedZoneRequest$;
exports.DeleteHostedZoneResponse$ = DeleteHostedZoneResponse$;
exports.DeleteKeySigningKey$ = DeleteKeySigningKey$;
exports.DeleteKeySigningKeyCommand = DeleteKeySigningKeyCommand;
exports.DeleteKeySigningKeyRequest$ = DeleteKeySigningKeyRequest$;
exports.DeleteKeySigningKeyResponse$ = DeleteKeySigningKeyResponse$;
exports.DeleteQueryLoggingConfig$ = DeleteQueryLoggingConfig$;
exports.DeleteQueryLoggingConfigCommand = DeleteQueryLoggingConfigCommand;
exports.DeleteQueryLoggingConfigRequest$ = DeleteQueryLoggingConfigRequest$;
exports.DeleteQueryLoggingConfigResponse$ = DeleteQueryLoggingConfigResponse$;
exports.DeleteReusableDelegationSet$ = DeleteReusableDelegationSet$;
exports.DeleteReusableDelegationSetCommand = DeleteReusableDelegationSetCommand;
exports.DeleteReusableDelegationSetRequest$ = DeleteReusableDelegationSetRequest$;
exports.DeleteReusableDelegationSetResponse$ = DeleteReusableDelegationSetResponse$;
exports.DeleteTrafficPolicy$ = DeleteTrafficPolicy$;
exports.DeleteTrafficPolicyCommand = DeleteTrafficPolicyCommand;
exports.DeleteTrafficPolicyInstance$ = DeleteTrafficPolicyInstance$;
exports.DeleteTrafficPolicyInstanceCommand = DeleteTrafficPolicyInstanceCommand;
exports.DeleteTrafficPolicyInstanceRequest$ = DeleteTrafficPolicyInstanceRequest$;
exports.DeleteTrafficPolicyInstanceResponse$ = DeleteTrafficPolicyInstanceResponse$;
exports.DeleteTrafficPolicyRequest$ = DeleteTrafficPolicyRequest$;
exports.DeleteTrafficPolicyResponse$ = DeleteTrafficPolicyResponse$;
exports.DeleteVPCAssociationAuthorization$ = DeleteVPCAssociationAuthorization$;
exports.DeleteVPCAssociationAuthorizationCommand = DeleteVPCAssociationAuthorizationCommand;
exports.DeleteVPCAssociationAuthorizationRequest$ = DeleteVPCAssociationAuthorizationRequest$;
exports.DeleteVPCAssociationAuthorizationResponse$ = DeleteVPCAssociationAuthorizationResponse$;
exports.Dimension$ = Dimension$;
exports.DisableHostedZoneDNSSEC$ = DisableHostedZoneDNSSEC$;
exports.DisableHostedZoneDNSSECCommand = DisableHostedZoneDNSSECCommand;
exports.DisableHostedZoneDNSSECRequest$ = DisableHostedZoneDNSSECRequest$;
exports.DisableHostedZoneDNSSECResponse$ = DisableHostedZoneDNSSECResponse$;
exports.DisassociateVPCFromHostedZone$ = DisassociateVPCFromHostedZone$;
exports.DisassociateVPCFromHostedZoneCommand = DisassociateVPCFromHostedZoneCommand;
exports.DisassociateVPCFromHostedZoneRequest$ = DisassociateVPCFromHostedZoneRequest$;
exports.DisassociateVPCFromHostedZoneResponse$ = DisassociateVPCFromHostedZoneResponse$;
exports.EnableHostedZoneDNSSEC$ = EnableHostedZoneDNSSEC$;
exports.EnableHostedZoneDNSSECCommand = EnableHostedZoneDNSSECCommand;
exports.EnableHostedZoneDNSSECRequest$ = EnableHostedZoneDNSSECRequest$;
exports.EnableHostedZoneDNSSECResponse$ = EnableHostedZoneDNSSECResponse$;
exports.GeoLocation$ = GeoLocation$;
exports.GeoLocationDetails$ = GeoLocationDetails$;
exports.GeoProximityLocation$ = GeoProximityLocation$;
exports.GetAccountLimit$ = GetAccountLimit$;
exports.GetAccountLimitCommand = GetAccountLimitCommand;
exports.GetAccountLimitRequest$ = GetAccountLimitRequest$;
exports.GetAccountLimitResponse$ = GetAccountLimitResponse$;
exports.GetChange$ = GetChange$;
exports.GetChangeCommand = GetChangeCommand;
exports.GetChangeRequest$ = GetChangeRequest$;
exports.GetChangeResponse$ = GetChangeResponse$;
exports.GetCheckerIpRanges$ = GetCheckerIpRanges$;
exports.GetCheckerIpRangesCommand = GetCheckerIpRangesCommand;
exports.GetCheckerIpRangesRequest$ = GetCheckerIpRangesRequest$;
exports.GetCheckerIpRangesResponse$ = GetCheckerIpRangesResponse$;
exports.GetDNSSEC$ = GetDNSSEC$;
exports.GetDNSSECCommand = GetDNSSECCommand;
exports.GetDNSSECRequest$ = GetDNSSECRequest$;
exports.GetDNSSECResponse$ = GetDNSSECResponse$;
exports.GetGeoLocation$ = GetGeoLocation$;
exports.GetGeoLocationCommand = GetGeoLocationCommand;
exports.GetGeoLocationRequest$ = GetGeoLocationRequest$;
exports.GetGeoLocationResponse$ = GetGeoLocationResponse$;
exports.GetHealthCheck$ = GetHealthCheck$;
exports.GetHealthCheckCommand = GetHealthCheckCommand;
exports.GetHealthCheckCount$ = GetHealthCheckCount$;
exports.GetHealthCheckCountCommand = GetHealthCheckCountCommand;
exports.GetHealthCheckCountRequest$ = GetHealthCheckCountRequest$;
exports.GetHealthCheckCountResponse$ = GetHealthCheckCountResponse$;
exports.GetHealthCheckLastFailureReason$ = GetHealthCheckLastFailureReason$;
exports.GetHealthCheckLastFailureReasonCommand = GetHealthCheckLastFailureReasonCommand;
exports.GetHealthCheckLastFailureReasonRequest$ = GetHealthCheckLastFailureReasonRequest$;
exports.GetHealthCheckLastFailureReasonResponse$ = GetHealthCheckLastFailureReasonResponse$;
exports.GetHealthCheckRequest$ = GetHealthCheckRequest$;
exports.GetHealthCheckResponse$ = GetHealthCheckResponse$;
exports.GetHealthCheckStatus$ = GetHealthCheckStatus$;
exports.GetHealthCheckStatusCommand = GetHealthCheckStatusCommand;
exports.GetHealthCheckStatusRequest$ = GetHealthCheckStatusRequest$;
exports.GetHealthCheckStatusResponse$ = GetHealthCheckStatusResponse$;
exports.GetHostedZone$ = GetHostedZone$;
exports.GetHostedZoneCommand = GetHostedZoneCommand;
exports.GetHostedZoneCount$ = GetHostedZoneCount$;
exports.GetHostedZoneCountCommand = GetHostedZoneCountCommand;
exports.GetHostedZoneCountRequest$ = GetHostedZoneCountRequest$;
exports.GetHostedZoneCountResponse$ = GetHostedZoneCountResponse$;
exports.GetHostedZoneLimit$ = GetHostedZoneLimit$;
exports.GetHostedZoneLimitCommand = GetHostedZoneLimitCommand;
exports.GetHostedZoneLimitRequest$ = GetHostedZoneLimitRequest$;
exports.GetHostedZoneLimitResponse$ = GetHostedZoneLimitResponse$;
exports.GetHostedZoneRequest$ = GetHostedZoneRequest$;
exports.GetHostedZoneResponse$ = GetHostedZoneResponse$;
exports.GetQueryLoggingConfig$ = GetQueryLoggingConfig$;
exports.GetQueryLoggingConfigCommand = GetQueryLoggingConfigCommand;
exports.GetQueryLoggingConfigRequest$ = GetQueryLoggingConfigRequest$;
exports.GetQueryLoggingConfigResponse$ = GetQueryLoggingConfigResponse$;
exports.GetReusableDelegationSet$ = GetReusableDelegationSet$;
exports.GetReusableDelegationSetCommand = GetReusableDelegationSetCommand;
exports.GetReusableDelegationSetLimit$ = GetReusableDelegationSetLimit$;
exports.GetReusableDelegationSetLimitCommand = GetReusableDelegationSetLimitCommand;
exports.GetReusableDelegationSetLimitRequest$ = GetReusableDelegationSetLimitRequest$;
exports.GetReusableDelegationSetLimitResponse$ = GetReusableDelegationSetLimitResponse$;
exports.GetReusableDelegationSetRequest$ = GetReusableDelegationSetRequest$;
exports.GetReusableDelegationSetResponse$ = GetReusableDelegationSetResponse$;
exports.GetTrafficPolicy$ = GetTrafficPolicy$;
exports.GetTrafficPolicyCommand = GetTrafficPolicyCommand;
exports.GetTrafficPolicyInstance$ = GetTrafficPolicyInstance$;
exports.GetTrafficPolicyInstanceCommand = GetTrafficPolicyInstanceCommand;
exports.GetTrafficPolicyInstanceCount$ = GetTrafficPolicyInstanceCount$;
exports.GetTrafficPolicyInstanceCountCommand = GetTrafficPolicyInstanceCountCommand;
exports.GetTrafficPolicyInstanceCountRequest$ = GetTrafficPolicyInstanceCountRequest$;
exports.GetTrafficPolicyInstanceCountResponse$ = GetTrafficPolicyInstanceCountResponse$;
exports.GetTrafficPolicyInstanceRequest$ = GetTrafficPolicyInstanceRequest$;
exports.GetTrafficPolicyInstanceResponse$ = GetTrafficPolicyInstanceResponse$;
exports.GetTrafficPolicyRequest$ = GetTrafficPolicyRequest$;
exports.GetTrafficPolicyResponse$ = GetTrafficPolicyResponse$;
exports.HealthCheck$ = HealthCheck$;
exports.HealthCheckAlreadyExists = HealthCheckAlreadyExists;
exports.HealthCheckAlreadyExists$ = HealthCheckAlreadyExists$;
exports.HealthCheckConfig$ = HealthCheckConfig$;
exports.HealthCheckInUse = HealthCheckInUse;
exports.HealthCheckInUse$ = HealthCheckInUse$;
exports.HealthCheckObservation$ = HealthCheckObservation$;
exports.HealthCheckRegion = HealthCheckRegion;
exports.HealthCheckType = HealthCheckType;
exports.HealthCheckVersionMismatch = HealthCheckVersionMismatch;
exports.HealthCheckVersionMismatch$ = HealthCheckVersionMismatch$;
exports.HostedZone$ = HostedZone$;
exports.HostedZoneAlreadyExists = HostedZoneAlreadyExists;
exports.HostedZoneAlreadyExists$ = HostedZoneAlreadyExists$;
exports.HostedZoneConfig$ = HostedZoneConfig$;
exports.HostedZoneFailureReasons$ = HostedZoneFailureReasons$;
exports.HostedZoneFeatures$ = HostedZoneFeatures$;
exports.HostedZoneLimit$ = HostedZoneLimit$;
exports.HostedZoneLimitType = HostedZoneLimitType;
exports.HostedZoneNotEmpty = HostedZoneNotEmpty;
exports.HostedZoneNotEmpty$ = HostedZoneNotEmpty$;
exports.HostedZoneNotFound = HostedZoneNotFound;
exports.HostedZoneNotFound$ = HostedZoneNotFound$;
exports.HostedZoneNotPrivate = HostedZoneNotPrivate;
exports.HostedZoneNotPrivate$ = HostedZoneNotPrivate$;
exports.HostedZoneOwner$ = HostedZoneOwner$;
exports.HostedZonePartiallyDelegated = HostedZonePartiallyDelegated;
exports.HostedZonePartiallyDelegated$ = HostedZonePartiallyDelegated$;
exports.HostedZoneSummary$ = HostedZoneSummary$;
exports.HostedZoneType = HostedZoneType;
exports.IncompatibleVersion = IncompatibleVersion;
exports.IncompatibleVersion$ = IncompatibleVersion$;
exports.InsufficientCloudWatchLogsResourcePolicy = InsufficientCloudWatchLogsResourcePolicy;
exports.InsufficientCloudWatchLogsResourcePolicy$ = InsufficientCloudWatchLogsResourcePolicy$;
exports.InsufficientDataHealthStatus = InsufficientDataHealthStatus;
exports.InvalidArgument = InvalidArgument;
exports.InvalidArgument$ = InvalidArgument$;
exports.InvalidChangeBatch = InvalidChangeBatch;
exports.InvalidChangeBatch$ = InvalidChangeBatch$;
exports.InvalidDomainName = InvalidDomainName;
exports.InvalidDomainName$ = InvalidDomainName$;
exports.InvalidInput = InvalidInput;
exports.InvalidInput$ = InvalidInput$;
exports.InvalidKMSArn = InvalidKMSArn;
exports.InvalidKMSArn$ = InvalidKMSArn$;
exports.InvalidKeySigningKeyName = InvalidKeySigningKeyName;
exports.InvalidKeySigningKeyName$ = InvalidKeySigningKeyName$;
exports.InvalidKeySigningKeyStatus = InvalidKeySigningKeyStatus;
exports.InvalidKeySigningKeyStatus$ = InvalidKeySigningKeyStatus$;
exports.InvalidPaginationToken = InvalidPaginationToken;
exports.InvalidPaginationToken$ = InvalidPaginationToken$;
exports.InvalidSigningStatus = InvalidSigningStatus;
exports.InvalidSigningStatus$ = InvalidSigningStatus$;
exports.InvalidTrafficPolicyDocument = InvalidTrafficPolicyDocument;
exports.InvalidTrafficPolicyDocument$ = InvalidTrafficPolicyDocument$;
exports.InvalidVPCId = InvalidVPCId;
exports.InvalidVPCId$ = InvalidVPCId$;
exports.KeySigningKey$ = KeySigningKey$;
exports.KeySigningKeyAlreadyExists = KeySigningKeyAlreadyExists;
exports.KeySigningKeyAlreadyExists$ = KeySigningKeyAlreadyExists$;
exports.KeySigningKeyInParentDSRecord = KeySigningKeyInParentDSRecord;
exports.KeySigningKeyInParentDSRecord$ = KeySigningKeyInParentDSRecord$;
exports.KeySigningKeyInUse = KeySigningKeyInUse;
exports.KeySigningKeyInUse$ = KeySigningKeyInUse$;
exports.KeySigningKeyWithActiveStatusNotFound = KeySigningKeyWithActiveStatusNotFound;
exports.KeySigningKeyWithActiveStatusNotFound$ = KeySigningKeyWithActiveStatusNotFound$;
exports.LastVPCAssociation = LastVPCAssociation;
exports.LastVPCAssociation$ = LastVPCAssociation$;
exports.LimitsExceeded = LimitsExceeded;
exports.LimitsExceeded$ = LimitsExceeded$;
exports.LinkedService$ = LinkedService$;
exports.ListCidrBlocks$ = ListCidrBlocks$;
exports.ListCidrBlocksCommand = ListCidrBlocksCommand;
exports.ListCidrBlocksRequest$ = ListCidrBlocksRequest$;
exports.ListCidrBlocksResponse$ = ListCidrBlocksResponse$;
exports.ListCidrCollections$ = ListCidrCollections$;
exports.ListCidrCollectionsCommand = ListCidrCollectionsCommand;
exports.ListCidrCollectionsRequest$ = ListCidrCollectionsRequest$;
exports.ListCidrCollectionsResponse$ = ListCidrCollectionsResponse$;
exports.ListCidrLocations$ = ListCidrLocations$;
exports.ListCidrLocationsCommand = ListCidrLocationsCommand;
exports.ListCidrLocationsRequest$ = ListCidrLocationsRequest$;
exports.ListCidrLocationsResponse$ = ListCidrLocationsResponse$;
exports.ListGeoLocations$ = ListGeoLocations$;
exports.ListGeoLocationsCommand = ListGeoLocationsCommand;
exports.ListGeoLocationsRequest$ = ListGeoLocationsRequest$;
exports.ListGeoLocationsResponse$ = ListGeoLocationsResponse$;
exports.ListHealthChecks$ = ListHealthChecks$;
exports.ListHealthChecksCommand = ListHealthChecksCommand;
exports.ListHealthChecksRequest$ = ListHealthChecksRequest$;
exports.ListHealthChecksResponse$ = ListHealthChecksResponse$;
exports.ListHostedZones$ = ListHostedZones$;
exports.ListHostedZonesByName$ = ListHostedZonesByName$;
exports.ListHostedZonesByNameCommand = ListHostedZonesByNameCommand;
exports.ListHostedZonesByNameRequest$ = ListHostedZonesByNameRequest$;
exports.ListHostedZonesByNameResponse$ = ListHostedZonesByNameResponse$;
exports.ListHostedZonesByVPC$ = ListHostedZonesByVPC$;
exports.ListHostedZonesByVPCCommand = ListHostedZonesByVPCCommand;
exports.ListHostedZonesByVPCRequest$ = ListHostedZonesByVPCRequest$;
exports.ListHostedZonesByVPCResponse$ = ListHostedZonesByVPCResponse$;
exports.ListHostedZonesCommand = ListHostedZonesCommand;
exports.ListHostedZonesRequest$ = ListHostedZonesRequest$;
exports.ListHostedZonesResponse$ = ListHostedZonesResponse$;
exports.ListQueryLoggingConfigs$ = ListQueryLoggingConfigs$;
exports.ListQueryLoggingConfigsCommand = ListQueryLoggingConfigsCommand;
exports.ListQueryLoggingConfigsRequest$ = ListQueryLoggingConfigsRequest$;
exports.ListQueryLoggingConfigsResponse$ = ListQueryLoggingConfigsResponse$;
exports.ListResourceRecordSets$ = ListResourceRecordSets$;
exports.ListResourceRecordSetsCommand = ListResourceRecordSetsCommand;
exports.ListResourceRecordSetsRequest$ = ListResourceRecordSetsRequest$;
exports.ListResourceRecordSetsResponse$ = ListResourceRecordSetsResponse$;
exports.ListReusableDelegationSets$ = ListReusableDelegationSets$;
exports.ListReusableDelegationSetsCommand = ListReusableDelegationSetsCommand;
exports.ListReusableDelegationSetsRequest$ = ListReusableDelegationSetsRequest$;
exports.ListReusableDelegationSetsResponse$ = ListReusableDelegationSetsResponse$;
exports.ListTagsForResource$ = ListTagsForResource$;
exports.ListTagsForResourceCommand = ListTagsForResourceCommand;
exports.ListTagsForResourceRequest$ = ListTagsForResourceRequest$;
exports.ListTagsForResourceResponse$ = ListTagsForResourceResponse$;
exports.ListTagsForResources$ = ListTagsForResources$;
exports.ListTagsForResourcesCommand = ListTagsForResourcesCommand;
exports.ListTagsForResourcesRequest$ = ListTagsForResourcesRequest$;
exports.ListTagsForResourcesResponse$ = ListTagsForResourcesResponse$;
exports.ListTrafficPolicies$ = ListTrafficPolicies$;
exports.ListTrafficPoliciesCommand = ListTrafficPoliciesCommand;
exports.ListTrafficPoliciesRequest$ = ListTrafficPoliciesRequest$;
exports.ListTrafficPoliciesResponse$ = ListTrafficPoliciesResponse$;
exports.ListTrafficPolicyInstances$ = ListTrafficPolicyInstances$;
exports.ListTrafficPolicyInstancesByHostedZone$ = ListTrafficPolicyInstancesByHostedZone$;
exports.ListTrafficPolicyInstancesByHostedZoneCommand = ListTrafficPolicyInstancesByHostedZoneCommand;
exports.ListTrafficPolicyInstancesByHostedZoneRequest$ = ListTrafficPolicyInstancesByHostedZoneRequest$;
exports.ListTrafficPolicyInstancesByHostedZoneResponse$ = ListTrafficPolicyInstancesByHostedZoneResponse$;
exports.ListTrafficPolicyInstancesByPolicy$ = ListTrafficPolicyInstancesByPolicy$;
exports.ListTrafficPolicyInstancesByPolicyCommand = ListTrafficPolicyInstancesByPolicyCommand;
exports.ListTrafficPolicyInstancesByPolicyRequest$ = ListTrafficPolicyInstancesByPolicyRequest$;
exports.ListTrafficPolicyInstancesByPolicyResponse$ = ListTrafficPolicyInstancesByPolicyResponse$;
exports.ListTrafficPolicyInstancesCommand = ListTrafficPolicyInstancesCommand;
exports.ListTrafficPolicyInstancesRequest$ = ListTrafficPolicyInstancesRequest$;
exports.ListTrafficPolicyInstancesResponse$ = ListTrafficPolicyInstancesResponse$;
exports.ListTrafficPolicyVersions$ = ListTrafficPolicyVersions$;
exports.ListTrafficPolicyVersionsCommand = ListTrafficPolicyVersionsCommand;
exports.ListTrafficPolicyVersionsRequest$ = ListTrafficPolicyVersionsRequest$;
exports.ListTrafficPolicyVersionsResponse$ = ListTrafficPolicyVersionsResponse$;
exports.ListVPCAssociationAuthorizations$ = ListVPCAssociationAuthorizations$;
exports.ListVPCAssociationAuthorizationsCommand = ListVPCAssociationAuthorizationsCommand;
exports.ListVPCAssociationAuthorizationsRequest$ = ListVPCAssociationAuthorizationsRequest$;
exports.ListVPCAssociationAuthorizationsResponse$ = ListVPCAssociationAuthorizationsResponse$;
exports.LocationSummary$ = LocationSummary$;
exports.NoSuchChange = NoSuchChange;
exports.NoSuchChange$ = NoSuchChange$;
exports.NoSuchCidrCollectionException = NoSuchCidrCollectionException;
exports.NoSuchCidrCollectionException$ = NoSuchCidrCollectionException$;
exports.NoSuchCidrLocationException = NoSuchCidrLocationException;
exports.NoSuchCidrLocationException$ = NoSuchCidrLocationException$;
exports.NoSuchCloudWatchLogsLogGroup = NoSuchCloudWatchLogsLogGroup;
exports.NoSuchCloudWatchLogsLogGroup$ = NoSuchCloudWatchLogsLogGroup$;
exports.NoSuchDelegationSet = NoSuchDelegationSet;
exports.NoSuchDelegationSet$ = NoSuchDelegationSet$;
exports.NoSuchGeoLocation = NoSuchGeoLocation;
exports.NoSuchGeoLocation$ = NoSuchGeoLocation$;
exports.NoSuchHealthCheck = NoSuchHealthCheck;
exports.NoSuchHealthCheck$ = NoSuchHealthCheck$;
exports.NoSuchHostedZone = NoSuchHostedZone;
exports.NoSuchHostedZone$ = NoSuchHostedZone$;
exports.NoSuchKeySigningKey = NoSuchKeySigningKey;
exports.NoSuchKeySigningKey$ = NoSuchKeySigningKey$;
exports.NoSuchQueryLoggingConfig = NoSuchQueryLoggingConfig;
exports.NoSuchQueryLoggingConfig$ = NoSuchQueryLoggingConfig$;
exports.NoSuchTrafficPolicy = NoSuchTrafficPolicy;
exports.NoSuchTrafficPolicy$ = NoSuchTrafficPolicy$;
exports.NoSuchTrafficPolicyInstance = NoSuchTrafficPolicyInstance;
exports.NoSuchTrafficPolicyInstance$ = NoSuchTrafficPolicyInstance$;
exports.NotAuthorizedException = NotAuthorizedException;
exports.NotAuthorizedException$ = NotAuthorizedException$;
exports.PriorRequestNotComplete = PriorRequestNotComplete;
exports.PriorRequestNotComplete$ = PriorRequestNotComplete$;
exports.PublicZoneVPCAssociation = PublicZoneVPCAssociation;
exports.PublicZoneVPCAssociation$ = PublicZoneVPCAssociation$;
exports.QueryLoggingConfig$ = QueryLoggingConfig$;
exports.QueryLoggingConfigAlreadyExists = QueryLoggingConfigAlreadyExists;
exports.QueryLoggingConfigAlreadyExists$ = QueryLoggingConfigAlreadyExists$;
exports.RRType = RRType;
exports.ResettableElementName = ResettableElementName;
exports.ResourceRecord$ = ResourceRecord$;
exports.ResourceRecordSet$ = ResourceRecordSet$;
exports.ResourceRecordSetFailover = ResourceRecordSetFailover;
exports.ResourceRecordSetRegion = ResourceRecordSetRegion;
exports.ResourceTagSet$ = ResourceTagSet$;
exports.ReusableDelegationSetLimit$ = ReusableDelegationSetLimit$;
exports.ReusableDelegationSetLimitType = ReusableDelegationSetLimitType;
exports.Route53 = Route53;
exports.Route53Client = Route53Client;
exports.Route53ServiceException = Route53ServiceException;
exports.Route53ServiceException$ = Route53ServiceException$;
exports.Statistic = Statistic;
exports.StatusReport$ = StatusReport$;
exports.Tag$ = Tag$;
exports.TagResourceType = TagResourceType;
exports.TestDNSAnswer$ = TestDNSAnswer$;
exports.TestDNSAnswerCommand = TestDNSAnswerCommand;
exports.TestDNSAnswerRequest$ = TestDNSAnswerRequest$;
exports.TestDNSAnswerResponse$ = TestDNSAnswerResponse$;
exports.ThrottlingException = ThrottlingException;
exports.ThrottlingException$ = ThrottlingException$;
exports.TooManyHealthChecks = TooManyHealthChecks;
exports.TooManyHealthChecks$ = TooManyHealthChecks$;
exports.TooManyHostedZones = TooManyHostedZones;
exports.TooManyHostedZones$ = TooManyHostedZones$;
exports.TooManyKeySigningKeys = TooManyKeySigningKeys;
exports.TooManyKeySigningKeys$ = TooManyKeySigningKeys$;
exports.TooManyTrafficPolicies = TooManyTrafficPolicies;
exports.TooManyTrafficPolicies$ = TooManyTrafficPolicies$;
exports.TooManyTrafficPolicyInstances = TooManyTrafficPolicyInstances;
exports.TooManyTrafficPolicyInstances$ = TooManyTrafficPolicyInstances$;
exports.TooManyTrafficPolicyVersionsForCurrentPolicy = TooManyTrafficPolicyVersionsForCurrentPolicy;
exports.TooManyTrafficPolicyVersionsForCurrentPolicy$ = TooManyTrafficPolicyVersionsForCurrentPolicy$;
exports.TooManyVPCAssociationAuthorizations = TooManyVPCAssociationAuthorizations;
exports.TooManyVPCAssociationAuthorizations$ = TooManyVPCAssociationAuthorizations$;
exports.TrafficPolicy$ = TrafficPolicy$;
exports.TrafficPolicyAlreadyExists = TrafficPolicyAlreadyExists;
exports.TrafficPolicyAlreadyExists$ = TrafficPolicyAlreadyExists$;
exports.TrafficPolicyInUse = TrafficPolicyInUse;
exports.TrafficPolicyInUse$ = TrafficPolicyInUse$;
exports.TrafficPolicyInstance$ = TrafficPolicyInstance$;
exports.TrafficPolicyInstanceAlreadyExists = TrafficPolicyInstanceAlreadyExists;
exports.TrafficPolicyInstanceAlreadyExists$ = TrafficPolicyInstanceAlreadyExists$;
exports.TrafficPolicySummary$ = TrafficPolicySummary$;
exports.UpdateHealthCheck$ = UpdateHealthCheck$;
exports.UpdateHealthCheckCommand = UpdateHealthCheckCommand;
exports.UpdateHealthCheckRequest$ = UpdateHealthCheckRequest$;
exports.UpdateHealthCheckResponse$ = UpdateHealthCheckResponse$;
exports.UpdateHostedZoneComment$ = UpdateHostedZoneComment$;
exports.UpdateHostedZoneCommentCommand = UpdateHostedZoneCommentCommand;
exports.UpdateHostedZoneCommentRequest$ = UpdateHostedZoneCommentRequest$;
exports.UpdateHostedZoneCommentResponse$ = UpdateHostedZoneCommentResponse$;
exports.UpdateHostedZoneFeatures$ = UpdateHostedZoneFeatures$;
exports.UpdateHostedZoneFeaturesCommand = UpdateHostedZoneFeaturesCommand;
exports.UpdateHostedZoneFeaturesRequest$ = UpdateHostedZoneFeaturesRequest$;
exports.UpdateHostedZoneFeaturesResponse$ = UpdateHostedZoneFeaturesResponse$;
exports.UpdateTrafficPolicyComment$ = UpdateTrafficPolicyComment$;
exports.UpdateTrafficPolicyCommentCommand = UpdateTrafficPolicyCommentCommand;
exports.UpdateTrafficPolicyCommentRequest$ = UpdateTrafficPolicyCommentRequest$;
exports.UpdateTrafficPolicyCommentResponse$ = UpdateTrafficPolicyCommentResponse$;
exports.UpdateTrafficPolicyInstance$ = UpdateTrafficPolicyInstance$;
exports.UpdateTrafficPolicyInstanceCommand = UpdateTrafficPolicyInstanceCommand;
exports.UpdateTrafficPolicyInstanceRequest$ = UpdateTrafficPolicyInstanceRequest$;
exports.UpdateTrafficPolicyInstanceResponse$ = UpdateTrafficPolicyInstanceResponse$;
exports.VPC$ = VPC$;
exports.VPCAssociationAuthorizationNotFound = VPCAssociationAuthorizationNotFound;
exports.VPCAssociationAuthorizationNotFound$ = VPCAssociationAuthorizationNotFound$;
exports.VPCAssociationNotFound = VPCAssociationNotFound;
exports.VPCAssociationNotFound$ = VPCAssociationNotFound$;
exports.VPCRegion = VPCRegion;
exports.errorTypeRegistries = errorTypeRegistries;
exports.paginateListCidrBlocks = paginateListCidrBlocks;
exports.paginateListCidrCollections = paginateListCidrCollections;
exports.paginateListCidrLocations = paginateListCidrLocations;
exports.paginateListHealthChecks = paginateListHealthChecks;
exports.paginateListHostedZones = paginateListHostedZones;
exports.paginateListQueryLoggingConfigs = paginateListQueryLoggingConfigs;
exports.waitForResourceRecordSetsChanged = waitForResourceRecordSetsChanged;
exports.waitUntilResourceRecordSetsChanged = waitUntilResourceRecordSetsChanged;
